{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Used in all sections for managing data and files\n",
    "import os\n",
    "import numpy as np\n",
    "from tqdm import tqdm\n",
    "import pandas as pd\n",
    "import pickle\n",
    "import re\n",
    "\n",
    "# NTLK is used for preprocessing text. You can find out more about each module using their documentation.\n",
    "import nltk\n",
    "from nltk.corpus import wordnet as wn\n",
    "from nltk.stem.wordnet import WordNetLemmatizer\n",
    "from nltk import word_tokenize, pos_tag\n",
    "from nltk.corpus import inaugural, stopwords\n",
    "\n",
    "\n",
    "# Scikit-Learn is used for feature extraction and training a logistic regression model\n",
    "from sklearn.feature_extraction.text import TfidfVectorizer\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.metrics import classification_report,f1_score \n",
    "from sklearn.model_selection import cross_val_score,train_test_split \n",
    "from sklearn.svm import SVC \n",
    "from sklearn.ensemble import RandomForestClassifier \n",
    "from sklearn.preprocessing import StandardScaler \n",
    "from sklearn.feature_selection import SelectKBest,f_classif \n",
    "from sklearn.naive_bayes import GaussianNB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "train = pd.read_csv(\"labelled_train_data.csv\") \n",
    "test  = pd.read_csv(\"data_only_test.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0    319\n",
       "1.0    304\n",
       "Name: ctrl, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train['ctrl'].value_counts()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'ctrl', 'pert',\n",
       "       'channel_count', 'characteristics_ch1', 'contact_address',\n",
       "       'contact_city', 'contact_country', 'contact_department',\n",
       "       'contact_email', 'contact_institute', 'contact_name', 'contact_state',\n",
       "       'data_processing', 'data_row_count', 'description',\n",
       "       'extract_protocol_ch1', 'growth_protocol_ch1', 'hyb_protocol',\n",
       "       'label_ch1', 'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
       "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
       "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
       "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
       "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
       "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
       "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
       "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
       "       'biomaterial_provider_ch2', 'growth_protocol_ch2'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "train.dtypes \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Germany', 'South Korea', 'France', 'Spain', 'USA', 'Russia',\n",
       "       'Singapore', 'Switzerland', 'China', 'Italy', 'Canada', 'Finland',\n",
       "       'Sweden', 'Austria', 'Israel', 'Taiwan', 'United Kingdom', 'Japan',\n",
       "       'Australia', 'Belgium'], dtype=object)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train['contact_country'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Italy' 'USA' 'South Korea' 'France' 'Taiwan' 'Germany' 'United Kingdom'\n",
      " 'Brazil' 'Japan' 'China' 'Canada' 'Spain' 'Switzerland' 'Israel'\n",
      " 'Belgium' 'Sweden' 'Austria' 'Netherlands' 'Singapore' nan 'Finland'\n",
      " 'Australia' 'India' 'Portugal' 'Argentina' 'Ireland' 'Norway' 'Poland'\n",
      " 'Mexico' 'Greece' 'Malaysia' 'Hungary' 'Thailand' 'Hong Kong'\n",
      " '----------' 'Russia']\n"
     ]
    }
   ],
   "source": [
    "train.groupby(['contact_country']).agg(['count','mean'])['ctrl'] \n",
    "print(test['contact_country'].unique()) \n",
    "p"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Basecalling was performed with the real time analysis (RTA) package within the Genome Analyzer Sequencing Control Software (SCS2.10).',\n",
       "       'All data normalization and selection of fold-changed genes were performed using GeneSpringGX 7.3 (Agilent Technology, USA). The averages of normalized ratios were calculated by dividing the average of normalized signal channel intensity by the average of normalized control channel intensity. Functional annotation of genes was performed according to Gene OntologyTM Consortium (http://www.geneontology.org/index.shtml) by GeneSpringGX 7.3. Gene classification was based on searches done by BioCarta (http://www.biocarta.com/), GenMAPP (http://www.genmapp.org/), DAVID (http://david.abcc.ncifcrf.gov/), and Medline databases (http://www.ncbi.nlm.nih.gov/).',\n",
       "       'Cell intensity files were generated with GeneChipOperating Software. Raw data were processed by the Robust Multichip Analysis (RMA) algorithm.',\n",
       "       'RNA-seq reads were mapped with STAR 2.5.3a',\n",
       "       'RNA-Seq reads were aligned to the mouse genome (UCSC Genome; Assembly NCBI37/mm9) using STAR',\n",
       "       'Full-length RNAseq (SMART-Seq2)', \"3' polyA RNA-seq\",\n",
       "       'Data were processed by variance stabilizing transform and quantile normalization',\n",
       "       'The data were normalised using preprocessCore quantile normalisation in R. Each culture (highly and low sensitive to the virus) was normalized separately',\n",
       "       'Raw data was processed by MAS.5 (Affymetrix) and analyzed using GeneSpring GX10',\n",
       "       'The raw data were exported from Illumina GenomeStudio (no background subtraction was performed).  The data were log2 transformed and normalized by quantiles using BRB-Array Tools.',\n",
       "       'Data were normalized by Robust Multichip Average (RMA) using Affymetrix Powertools',\n",
       "       'Gene expression levels were estimated from GeneChip® probe intensities using robust multi-array average (RMA) normalization and background-correction algorithms.',\n",
       "       \"CEL files were summarized using Affymetrix Expression Console (current version 1.1). Expression estimates were derived from the probe level hybridization intensities using the Robust Multi-Array Analysis (RMA) algorithm.  Differential expression between experimental and control samples was examined using Welch's T-test and a False Discovery Rate correction.\",\n",
       "       'Robust Multichip Averages', 'RNA-seq',\n",
       "       'Array hybridization, washing, scanning, data extraction and analyses were performed as previously described (Keita et al., 2013).',\n",
       "       'The data were normalised using quantile normalisation with Genome Studio',\n",
       "       'The data were analyzed with Expression Console, R and Bioconductor.',\n",
       "       'fastx_clipper from FASTX Toolkit v0.0.14 was employed to remove the adapter sequence from raw sequencing read, reads shorter than 18 bp were removed',\n",
       "       'Command Console Software (Affymetrix, Santa Clara, CA, US) was used to control the scanner and summarize probe cell intensity data (CEL file generation) with default settings. The raw data were normalized by Expression Console with MAS 5.0.',\n",
       "       'The scanned images were analyzed with Feature Extraction Software (Agilent) using default parameters to obtain background subtracted and spatially detrended Processed Signal intensities. Features flagged in Feature Extraction as Feature Non-uniform outliers were excluded. Signal intensities were normalized using combined rank consistency filtering with LOWESS intensity normalization in Agilent Feature Extraction software. Data was subsequently analyzed by GeneSpring (Agilent).',\n",
       "       'RMA (bioconductor)',\n",
       "       'Data was normalized using the GCRMA algorithm as implemented in Bioconductor/R.',\n",
       "       'BioConductor Limma package.',\n",
       "       'Data were evaluated using the R software and the limma package from BioConductor. Pathway analyzes were done using the ranks of the test statistics  based on the KEGG.db annotation package.',\n",
       "       \"The raw intensity data were preprocessed by Affymetrix's Expression Console (version 1.1)\",\n",
       "       \"Data processing cel files were normalized with 'rma' function of the Bioconductor environment (v2.9) in R.3.0.0\",\n",
       "       \"Affymetrix® microarray data were collected with Affymetrix® GeneChip® Command Console® Software. Probe values were corrected for background and normalized via quantile normalized by Robust Multichip Average (RMA) of the Bioconductor 'affy' package.In the detection column, calls include P for Present, A for Absent, and M for Marginal.\",\n",
       "       'For RNA-Seq data quality control was performed by Fastqc. Before downstream analysis, low quality reads and adaptor contamination were deleted. RNA-Seq data mapping was performed by Tophat2 and hg19 as reference genome. Gene assemble and expression quantification was performed by Cufflinks.',\n",
       "       'The data were analyzed with the Bioconductor package Affy using RMA method for normalizing and summarizing probe level intensity measurements.',\n",
       "       'RMA normalization log2 at the transcript level using the oligo package of Bioconductor.',\n",
       "       'CASAVA 1.8.4 - configureBclToFastq.pl --input-dir runs/150924_D00198_0427_BC7WFRACXX/Data/Intensities/BaseCalls/ --output-dir process/150924_D00198_0427_BC7WFRACXX/ --sample-sheet process/150924_D00198_0427_BC7WFRACXX/ --ignore-missing-bcl --ignore-missing-stats --use-bases-mask y51,i6n,y51',\n",
       "       'The data were normalised using quantile normalisation with IlluminaGUI in R',\n",
       "       'The microarray data were analyzed using the GeneSpring GX software (Agilent). The data were summarized, log2 transformed and normalized with the Robust Multi-array Analysis (RMA) algorithm. Only probe sets with signal intensities above the 20th percentile in at least one of the 6 microarrays performed were retained, whereas all other probe sets were excluded from further analyses. Subsequently, the signal intensities of the replicate microarrays performed for each mouse group (SPC-HAxTCR-HA, SPC-HA) were averaged and the fold changes were calculated in reference to the data from the SPC-HA group. Only probe sets with a fold change greater than or equal to ±2 were retained. All other probe sets were excluded from further analyses. Finally, 378 transcripts were found to be differentially expressed in SPC-HAxTCR-HA vs. SPC-HA lungs. 326 of those were up-regulated and 52 were down-regulated.',\n",
       "       'Illumina bcl2fastq-1.8.4 used for base calling',\n",
       "       'Agilent Feature Extraction Software (v 9.5.3.1) was used for background subtraction and linear Lowess normalization.',\n",
       "       'RNAseq reads were filtered for average QC score>30. Reads were mapped to hg19 genome.',\n",
       "       'Data from both channels was extracted using the Agilent Feature Extractor Software, Version 7.5.  Due to potential variation of the Cy-5 dye we prefer to analyze the green channel alone and use single channel data.  Results for control probes were first removed, multiple repeats of any probe were averaged, then each array was z-normalized.  Differently expressed genes were identified between the 3 sample groups and pathway enrichment analysis was conducted using Diane 6.0, an in-house microarray analysis program.',\n",
       "       'Basecalls performed using GAPipeline or CASAVA',\n",
       "       'RMA normalization, done by the SCIBLU Micoarray Resourse Center (MARC), Lund University, Sweden',\n",
       "       'Affymetrix Expression Console version 1.1 was used to generate the .cel files. Partek Genomics Suite 6.5 (Partek Inc., St. Louis, MO) was used to perform data analysis. Robust multi-chip analysis (RMA) normalization was done on the entire data set.',\n",
       "       'Affymetrics probeset summaries were computed using RMA algorithm.\\xa0 Data were then imported to BRB-ArrayTools and subsequent data filtering, median normalization and statistical analyses were performed using this software.',\n",
       "       'Raw scan data were normalized with the use of cubic spline algorithm implemented in Illumina Normalizer module of GenePattern software (www.broadinstitute.org/genepattern).',\n",
       "       'Data was processed using GeneSpring GX 10 Software. Data was imported using the enhanced Agilent FE Import setting. Spot information in Feature Extraction was used to flag data, and normalization was performed using \"per spot and per chip: intensity dependent (Lowess) normalization\".  Data obtained for the Cy5-labelled GLN sample were divided by the Cy3-labelled PDLN sample to obtain ratios shown in the \"PRE_VALUE\" column of processed data. Raw data show log2 of ratio.',\n",
       "       'Statistical analysis was done with the statistical computing environment R. Additional software packages (affy, geneplotter, multtest, and vsn) were taken from the Bioconductor project.  Microarray preprocessing: Probe level normalization was conducted using the variance stabilization method (vsn) from Huber et al..',\n",
       "       'Fastq files containing passed-QC reads were uploaded to BaseSpace. Filtering, mapping and abundance measurements (FPKM) were performed with the app RNAexpress (Illumina) within BaseSpace.',\n",
       "       'The data were analyzed with Affymetrix Expression Console software (v1.4.1) using RMA-Sketch algorithm for data normalization. FF, FFPE_3h and FFPE_10h datasets were normalized together. Normalization was done to gene level as well as to exon level.',\n",
       "       'The raw data were normalized by the log scale robust multi-array analysis (RMA) method with the Expression Console software version 1.1 (Affymetrix).',\n",
       "       '1. Mapping: Reads were mapped with Tophat/2.0.4 to the Human genome assembly, build GRCh37.',\n",
       "       'Using a HiSeq SBS sequencing kit version 4, Fastq files were generated and demultiplexed with with the bcl2fastq v2.17.1.14 Conversion Software (Illumina).',\n",
       "       'GCOS scaling to target intensity of 100',\n",
       "       'Data processing of arrays were performed using Illumina BeadStudio software. Gene expression data was normalized using Quantile normalization and log2 transformation. Sample Gene Profile option of Illumina BeadStudio was used to export to a data matrix.',\n",
       "       'Background correction and normalization was performed using the RMA-Sketch implementation in Affymetrix Expression Console',\n",
       "       'GeneSpring software GX (Agilent Technologies), RMA normalization',\n",
       "       'Data was normalized in R (v.2.14.) with the bioconductor package affy(v.1.32.0) and the function RMA.',\n",
       "       'Microarray data were analyzed using the Oligo package available at the Bioconductor website (www.bioconductor.org). The raw data were first background-corrected by the Robust Multichip Average (RMA) method and then normalized by an invariant set method. Probesets were considered expressed if at least one sample had a RMA expression value greater than the mean of the negative control probesets, 18,494 probesets were considered expressed.  Differential gene expression between the control and mutant mice was analyzed by the Limma package available at the Bioconductor website. P-values obtained from the multiple comparison tests were corrected by false discovery rates. Heatmap displays were created using the freely available MeV package ( http://www.tm4.org/mev/).',\n",
       "       'Affymetrix Expression Console RMA',\n",
       "       'Paired-end sequencing (150 bp for each read) was performed using HiSeq X with the depth of 24 million reads for each sample. Quality of sequenced reads were assessed using FastQC (https://www.bioinformatics.babraham.ac.uk/projects/fastqc/). PolyA/T tails were trimmed using PRINSEQ (Schmieder and Edwards, 2011). Adaptor sequences were trimmed and shorter reads than 24 bp were dropped using Trimmomatic (Bolger et al., 2014). Reads were mapped to the mouse reference genome (mm10) using Hisat2 (Kim et al., 2015) with default setting. Duplicate reads were removed using the markdup option of SAMtools (Li et al., 2009). Fragments numbers were counted using the featureCounts option of SUBREAD (Liao et al., 2014).',\n",
       "       'The Cy5 fluorescent intensity of each spot was analyzed by Genepix 4.1 software (Molecular Devices, Sunnyvale, CA). The signal intensity of each spot was corrected by subtracting background signals in the surrounding. We filtered out spots that signal-to-noise ratio was less than 0 or control probes. Spots that passed aforementioned criteria were normalized by the R program in the limma package using quantile normalization (Smyth, 2005).',\n",
       "       'Raw signal intensities were initially screened by using the MaxD/View-Program to remove false positive and false negative signals (using a cut off of at least one P value for each transcript <0.2). The data were then normalized by using the global geometric mean (GGM) normalization option within the MaxD/View-Program',\n",
       "       'The data were generated using the Affymetrix Scanner according to Affymetrix protocol. The intensities are from MAS5.0',\n",
       "       'Probes with detection P-values above 0.05 in all samples were excluded from the dataset by the Illumina GenomeStudio software. Absolute signal intensities of the filtered probes were quantile normalized using EXPANDER 5.2 (Ulitsky I et al. 2010, Nat Protoc. 2010;5:303-22).',\n",
       "       'CEL files were exported to Affymetrix Expression Console Version 1.1 and RMA normalization was carried out. [The data were also  analyzed with Microarray Suite version 5.0 (MAS 5.0) using GeneChip Operating Software (GCOS, Affymetrix, CA, USA). For absolute analysis, each chip was normalized to a target intensity of 500, and probe sets were assigned a signal intensity and detection call of Present, Marginal or Absent (See MAS5_GCOS_analyzed.txt on GSE29885)]',\n",
       "       'The scanned images were analyzed using the GeneChip Analysis Suite 3.3 (Affymetrix) to identify genes differentially expressed among the RNA samples. Data were normalized using the RMA algorithm.',\n",
       "       'Data processing was done using Partek Genomics Suite 6.5 software. RMA was used to summarize and normalize the probe data into probe set expression values. The list of utilized probe sets was defined by custom .ps and .mps files. The RMA procedure used included the options adjust for GC content, probe sequence, RMA background correction, Quantile Normalization and median polish probe set summarization. Expression values for probesets were summarized to a single value per gene based on the custom file MoEx-1_0-st-v1._core+miRNA+RfSq+UCSC_.SH.2916.mps. The outlier excluded mean method was used. This procedure excludes probe set values that are more than 3 standard deviations above the mean or more and 2 standard deviations below the mean.',\n",
       "       'CEL files were imported into the Partek Genomics Suite v2 (www.partek.com) using option \"RMA\" under default settings. RMA background correction and quantile normalization was performed upon import; generating 53,981 log2 median polished gene probe expression values per CEL file. Probe values representing spike-in and hybridization controls were next filter-removed and the remaining probe values (n=53,617) exported for interrogation and analysis in R (https://cran.r-project.org/).',\n",
       "       'The Agilent Feature Extraction Software (FES) was used to read out and process the microarray image files. The software determines feature intensities (including background subtraction), rejects outliers and calculates statistical confidences. For the normalised data, the signal intensities from the single-experiment raw data lists are normalized by dividing the intensity values by their median.',\n",
       "       'Agilent Feature Extraction Software (v 10.7) was used for background subtraction.',\n",
       "       'Data extraction was performed with the ‘Feature Extraction Software V10.7.3.1’ using the extraction protocol file ‘GE1_107_Sep09.xml’. Extracted microarray raw data were imported into Omics Explorer software v3.0 (Qlucore) under default import settings for Agilent One Color mRNA Microarrays. Accordingly, data processing steps were: 1) removal of control measurements, 2) log base 2 transformation, 3) normalization by shifting to 75 Percentile, 4) averaging of values from on-chip replicates, 5) baseline transformation to the median. Note, that six additional samples have been initially imported into one Qlucore Omics Explorer project. These 6 additional samples have not been utilized in the context of this study. However, baseline transformation in Qlucore Omics Explorer software (see step 5) is based on all samples (including the 6 samples that were finally removed and are not contained in this very GEO series).',\n",
       "       'For data processing, a novel algorithm that distinguished between changes in gene expression and splicing variation was developed.',\n",
       "       'This information is not available since the experiments were done more than 10 years ago.',\n",
       "       'High-level and low-level transcriptomics analyses were carried out using GeneSpring GX12.0 (Agilent) (www.genomics.agilent.com). ExonPLIER16 summarization algorithm, quantile normalization, baseline transformation to all samples.',\n",
       "       'Cel files were imported into Genespring for preprocessing using RMA, and normalization using a global per-chip and per-gene normalization strategy',\n",
       "       'The data was analyzed by the statistical core facility at Dan L. Duncan Cancer Center at Baylor College of Mediaine. CEL files were processed in Bioconductor using Quantile method.',\n",
       "       'Image analysis was performed in real time by the HiSeq Control Software (HCS) v2.2.68 and Real Time Analysis (RTA) v1.18.66.3, running on the instrument computer. RTA performs real-time base calling on the HiSeq instrument computer. Then the Illumina bcl2fastq 1.8.4 pipeline was used to generate the sequence data.',\n",
       "       \"RRBS: Reads were trimmed with Trim Galore! (v0.4.2, http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/) for adapter contamination and 3' ends with base quality below 20 in RRBS mode.\",\n",
       "       'The data was analyzed with GeneSpring version 7.3 using RMA normalization with a custom CDF file(Mm430_Mm_REFSEQ_8), then normalized to the REF colon repl2 and the mean of the 3 WT colon samples',\n",
       "       'The data was analyzed with GeneSpring version 7.3 using RMA normalization with a custom CDF file(Mm430_Mm_REFSEQ_8), then normalized to the REF colon repl1 and the mean of the 3 WT colon samples',\n",
       "       'Agilent Feature Extraction V11.5.1.1 was used for background subtraction and LOWESS normalization. Control spots are removed, replicate oligos are averaged.',\n",
       "       'MAS5',\n",
       "       'The data were analyzed with R software and Bioconductor with MAS5.0 algorithm using global scaling as normalization method. Trimmed mean target intensity of each array was determined as 210. R was used for additional quality control and computing statistics with corrected variances.',\n",
       "       'The data were analyzed with GCOS using MAS5.0 algorithm with Affymetrix default analysis settings and global scaling as normalization method. The trimmed mean target intensity of each array was arbitrarily set to 218.',\n",
       "       'The data were analyzed with GCOS using MAS5.0 algorithm with Affymetrix default analysis settings and global scaling as normalization method. The trimmed mean target intensity of each array was arbitrarily set to 107.',\n",
       "       'Aligned to hg19 using tophat v2.0.13 and Gencode V19 transcript models',\n",
       "       'The data were normalized using quantile normalization withIllumina Genome Studio',\n",
       "       'Raw data were summarised and normalised using Robust Multi-array Average (RMA)',\n",
       "       'The raw data was checked by FastQC.',\n",
       "       'The data were analyzed and normalized by Robust Multi-array Average (RMA) algorithm using affylmGUI (1.14.0) running limma version 2.9.16 within the language and environment program R (2.4.1).',\n",
       "       'Grooming of de-multiplexed reads was performed with FASTQ groomer 1.04.',\n",
       "       'A single weighted mean expression level for each gene was derived using MICROARRAY SUITE 5.0 (MAS 5.0) software . The MAS 5.0 software also generated a detection P value [P(detection)] using a one-sided Wilcoxon sign-ranked test, which indicated whether the transcript was reliably detected. The trimmed mean target intensity of each array was arbitrarily set to 100.',\n",
       "       'The raw signal intensity data were background corrected (normexp method) and quantile normalised using limma package of Bioconductor in R program.',\n",
       "       'Background correction, summarization (median polish) and normalization (quantile) were done with Robust Multi-array Analysis using RMAExpress v1.0.5',\n",
       "       'The data were quantile normalized without background subtraction using Illumina GenomeStudio V1.8',\n",
       "       'The data were analyzed with Expression Console using Affymetrix default analysis settings and RMA normalization method.',\n",
       "       'AGCC Scan Control software',\n",
       "       'Expression values were subjected to RMA normalization using Affymetrix CDF probeset definition and GeneSpringGX v7.3.1.',\n",
       "       'The images and raw data from each chip were transferred automatically to the microarray database using one of the specified microarray core servers',\n",
       "       'Sequencing, data quality, reads repartition (e.g., for potential ribosomal contamination), and insert size estimation were performed using FastQC, Picard-Tools, Samtools and rseqc',\n",
       "       'bcl2fastq/2.15.0.4',\n",
       "       'The resulting .gpr files were uploaded to Spotfire DecisionSite for Microarray Analysis running the Splus suite, and normalization performed using the LOWESS method.  The resulting log2 fold changes were calculated and used for uploading to Database for Annotation, Visualization and Integrated Discovery (DAVID) (NCBI).Two-fold up or down regulated genes were uploaded to DAVID. Pathways/terms which were enriched in a statistically significant manner (P <0.05; FDR corrected) were used for further analysis and display.',\n",
       "       'Data were processed with GCOS 1.4 (TGT = 150) and subsequently imported and further analyzed in the online database (www.bioretis.de).',\n",
       "       'Illumina Casava1.7.1a5 software used for basecalling.'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train['data_processing'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([nan,\n",
       "       'The hybridized images were scanned using Agilent’s DNA microarray scanner and quantified with Feature Extraction Software (Agilent Technology, Palo Alto, CA).',\n",
       "       'Affymetrix Gene ChIP Scanner 3000 7G',\n",
       "       'Standard Illumina scanning protocol',\n",
       "       'Subsequent to hybridization, the arrays were washed and stained with streptavidin-phycoerythrin, then scanned in an Affymetrix GeneChip® Scanner 3000 (Santa Clara, CA). All control parameters were confirmed to be within normal ranges before normalization and data reduction was initiated.',\n",
       "       'We incubated each BeadChip with Cy3-Streptavidin for 10 min, dried, and then scanned on an Illumina BeadArray Scanner.',\n",
       "       'GeneChips were scanned using the Affymetrix Genechip Scanner 3000.',\n",
       "       'GeneChips were scanned using the Affymetrix GeneChip Scanner 3000.',\n",
       "       'Affymetrix GeneArray Scanner 3000 7G Plus',\n",
       "       'standard Affymetrix procedures',\n",
       "       'Array hybridization, washing, scanning, data extraction and analyses were performed as previously described (Keita et al., 2013).',\n",
       "       'Standard Illumina hybridization protocol',\n",
       "       'GeneChips were scanned using the Affymetrix GeneChip Scanner 3000 7G.',\n",
       "       'Arrays were scanned by Affymetrix GeneChip® Scanner 3000 (Cat#00-00213, Affymetrix, Santa Clara, CA, US).',\n",
       "       'Slides were scanned immediately after washing on the Agilent DNA Microarray Scanner using one color scan setting for 4x44k array slides (Scan resolution 10um, Dye channel is set to Green and Green PMT is set to 100%).',\n",
       "       'Johns Hopkins Microarray Core Facility',\n",
       "       'Arrays were scanned on an Affymetrix Scanner 3000-7G scanner with GCOS software. Scanning was peformed according to the protocol described in the Affymetrix GeneChip® Expression Analysis Technical Manual, November 2004 Edition.',\n",
       "       'Microarrays were washed and scanned with Agilent Scanner  using manufacturer provided protocols. Feature Extraction software was used for image analysis.',\n",
       "       'Slides were scanned using the GenePix 4100A scanner (Axon Instruments, Downingtown, PA) at 5 µm resolution.',\n",
       "       'Arrays were scanned using an Affymetrix GeneChip Scanner 3000',\n",
       "       'Standard Affymetrix protocol using a Gene chip scanner 3000',\n",
       "       'GeneChips were scanned using the GeneChip® Scanner 3000 .',\n",
       "       'GeneChips were scanned using the Affymetrix Gene Array Scanner 3000 through the University of Washington Functional Genomics Laboratory in the Center for Ecogenetics and Environmental Health (http://depts.washington.edu/ceeh/members/core-services.html).',\n",
       "       'Microarrays were scanned with an Affymetrix GCS 3000 scanner running with GCOS v1.1.1 software.',\n",
       "       'Scanned on an Agilent G2565BA scanner.',\n",
       "       'Arrays were scanned using an Agilent Scanner G2565AA .',\n",
       "       'Done by the SCIBLU Micoarray Resourse Center (MARC), Lund University, Sweden',\n",
       "       'GeneChips were scanned with Affymetrix GeneChip® Scanner 7G and AGCC software version 1.1',\n",
       "       'GeneChips were scanned using GeneArray Scanner 3000_7Gplus',\n",
       "       'Refer to the Illumina System Manual',\n",
       "       'Microarray chips were scanned using the DNA microarray scanner (Agilent) with the following parameters: 2 color scan; scan resolution 5 um, single pass, red and green dye channels.  Data was extracted using Feature Extraction software.',\n",
       "       'GeneChips were scanned using Affymetrix GeneChip Scanner 3000 7G (with AGCC v3.1.1 software)',\n",
       "       'The arrays were scanned by the GeneChip Scanner 3000 (Affymetrix).',\n",
       "       'Affymetrix standard protocol',\n",
       "       \"Arrays were scanned using an Illumina Bead array Reader (BeadStation 500GXDW; Illumina, Inc., San Diego, CA) following the manufacturer's procedure.\",\n",
       "       'Biotinylated cRNA was hybridized on GeneChip® Mouse Gene 2.0 ST Arrays , that were stained, washed and scanned following standard procedures.',\n",
       "       'standard Affymetrix protocol', 'Affymetrix scanner 3000 7g.',\n",
       "       'A confocal scanner was used to collect fluorescence signal after excitation at 570 nm.',\n",
       "       'GeneChip scanner 3000 7G, GeneChip Operating Software v1.4',\n",
       "       'The slides were scanned by an Axon 4000 scanner (Molecular Devices, Sunnyvale, CA).',\n",
       "       'PICR Standard 11uM feature-1 protocol - using the Affymetrix GeneChip Scanner 3000 system with Autoloader',\n",
       "       'All samples were subjected to gene expression analysis via the Affymetrix Human Genome U133 Plus 2.0 high-density oligonucleotide array which currently queries over 47,000 transcripts.  Hybridization, staining and washing of all arrays were performed in the Affymetrix fluidics module as per the manufacturer’s protocol.  Streptavidin phycroerythrin stain (SAPE, Molecular Probes) was the fluorescent conjugate used to detect hybridized target sequences. The detection and quantitation of target hybridization was performed with a GeneArray Scanner 3000 (Affymetrix) All arrays referred to in this study were assessed for “array performance” prior to data analysis.',\n",
       "       \"The arrays were stained according to the manufacturer's protocols and then scanned with the Genechip scanner (Affymetrix, CA, USA).\",\n",
       "       'Specifically bound probe was detected by incubating the arrays with SAPE (Molecular Probes) and scanning the chips with a Hewlett-Packard GeneArray scanner (Affymetrix). A second scan was performed after signal enhancement with biotinylated anti-streptavidin antibody (Vector Laboratories).',\n",
       "       'Affymetrix Gene ChIP Scanner 3000 7G; GeneChip® Command Console.',\n",
       "       'The Affymetrix GeneChip Scanner 3000 was used to scan each Affymetrix GeneChip Human 2.0 ST Array. After, gene probe intensities were generated using the Affymetrix AGCC software (Affymetrix, Inc).',\n",
       "       'Performed by Milteny Biotech following standard procedures',\n",
       "       'Array was scanned with an Agilent G2565BA Microarray Scanner.',\n",
       "       'Slides were scanned on the Agilent Micro Array Scanner G2565CA (pixel resolution 5 µm, bit depth 20).',\n",
       "       'Raw data were acquired using an Agilent DNA Microarray Scanner and Agilent Feature Extraction Software (v.9.1).',\n",
       "       'This information is not available since the experiments were done more than 10 years ago.',\n",
       "       'Affymetrix GeneChip Scanner',\n",
       "       'The microarrays were scanned with a genchip scanner 3000 7G',\n",
       "       'The microarray was scanned with an Affymetix GeneChip 7 G scanner (Affymetix, Santa Clara, CA).',\n",
       "       'Scanned on an Agilent G2505C scanner.',\n",
       "       'The arrays were then stained with phycoerythrein-conjugated streptavidin (Molecular Probes) and the fluorescence intensities were determined using the GCS 3000 high-resolution confocal laser scanner (Affymetrix).  The scanned images were analyzed using programs resident in GeneChip Operating System (GCOS; Affymetrix). Sample loading and variations in staining were standardized by scaling the average of the fluorescent intensities of all genes on an array to a constant target intensity of 250 for all arrays used. ',\n",
       "       'GeneChips were scanned using Affymetrix Scanner 3000.',\n",
       "       'As per Affimetrix specifications',\n",
       "       'Arrays were scanned on an Affymetrix Gene Chip Scanner 3000, using Command Console Software.',\n",
       "       'Confocal laser scanning (Agilent) performed to detect the streptavidin-labeled fluor.',\n",
       "       'Chips were scanned on an Agilent G2505B scanner. Images were quantified using Agilent Feature Extraction Software (version 10.7.3.1).',\n",
       "       'The chips were scanned with an Affymetrix Gene-Chip Scanner-3000-7G and the quality control matrices were confirmed with the Affymetrix GCOS software following the manufacturer’s guidelines.',\n",
       "       'Affymetrix GeneChip Scanner 3000 7G and basic data processing was done on the AGCC Scan Control software.',\n",
       "       'Affymetrix Expression Console',\n",
       "       'Images were obtained on an Illumina Beadchip scanner.',\n",
       "       'Arrays were scanned using a GenePix 4000B dual laser scanner using GenePix Pro 6.3 software.',\n",
       "       'GeneChips were scanned using the Affymetrix GeneChip Scanner 3000'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train['scan_protocol'].unique()  \n",
    "##test['label_ch1'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Unnamed: 0                    0\n",
       "geo_accession                 0\n",
       "gse_id                        0\n",
       "ctrl                          0\n",
       "pert                          0\n",
       "channel_count                 0\n",
       "characteristics_ch1           0\n",
       "contact_address               0\n",
       "contact_city                  0\n",
       "contact_country               0\n",
       "contact_department          120\n",
       "contact_email               229\n",
       "contact_institute             0\n",
       "contact_name                  0\n",
       "contact_state               174\n",
       "data_processing               0\n",
       "data_row_count                0\n",
       "description                 147\n",
       "extract_protocol_ch1          0\n",
       "growth_protocol_ch1         341\n",
       "hyb_protocol                110\n",
       "label_ch1                   110\n",
       "label_protocol_ch1          110\n",
       "last_update_date              0\n",
       "molecule_ch1                  0\n",
       "organism_ch1                  0\n",
       "platform_id                   0\n",
       "scan_protocol               110\n",
       "source_name_ch1               0\n",
       "status                        0\n",
       "submission_date               0\n",
       "supplementary_file          110\n",
       "taxid_ch1                     0\n",
       "title                         0\n",
       "treatment_protocol_ch1      325\n",
       "type                          0\n",
       "contact_phone               463\n",
       "contact_laboratory          279\n",
       "relation                    486\n",
       "contact_fax                 608\n",
       "biomaterial_provider_ch1    609\n",
       "contact_web_link            614\n",
       "characteristics_ch2         597\n",
       "extract_protocol_ch2        597\n",
       "label_ch2                   597\n",
       "label_protocol_ch2          597\n",
       "molecule_ch2                597\n",
       "organism_ch2                597\n",
       "source_name_ch2             597\n",
       "taxid_ch2                   597\n",
       "treatment_protocol_ch2      608\n",
       "biomaterial_provider_ch2    619\n",
       "growth_protocol_ch2         614\n",
       "dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'ctrl', 'pert',\n",
       "       'channel_count', 'characteristics_ch1', 'contact_address',\n",
       "       'contact_city', 'contact_country', 'contact_institute', 'contact_name',\n",
       "       'data_processing', 'data_row_count', 'description',\n",
       "       'extract_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
       "       'organism_ch1', 'platform_id', 'source_name_ch1', 'status',\n",
       "       'submission_date', 'taxid_ch1', 'title', 'type'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.columns[train.isnull().sum() == 0 ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'channel_count', 'data_row_count', 'taxid_ch1'], dtype='object')"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.columns[train.dtypes == np.int64]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(623, 53)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['rep1_cd44low_dox' 'rep1_imecs_myc_dox' 'rep2_cd44low_etoh'\n",
      " 'rep1_cd44high_dox' 'lipopolysaccharide treated sample replication 2'\n",
      " 'untreated control replication 1' 'untreated control replication 2'\n",
      " 'lipopolysaccharide treated sample replication 1'\n",
      " 'Lovo cells infected with mutant DLntA Listeria, biological rep1'\n",
      " 'Lovo cells infected with LntA-V5+ Listeria, biological rep3'\n",
      " 'Lovo cells infected with LntA-V5+ Listeria, biological rep1'\n",
      " 'Lovo cells infected with LntA-V5+ Listeria, biological rep2'\n",
      " 'Lovo cells infected with mutant DLntA Listeria, biological rep2'\n",
      " 'EBC1Crizo2' 'H1573IFN2' 'H1993B3' 'H596CrizoH2' 'alpha7G Smoked Male'\n",
      " 'alpha7 Control Female' 'alpha7G Control Male'\n",
      " 'alpha7E260A:G Smoked Male' 'alpha7E260A:G Control Male'\n",
      " 'Trachea epithelial single cell 18'\n",
      " 'Tp0_Rep2_Tom: Pulse-seq T=0 days, Rep2, basal lineage-negative tdTomato+'\n",
      " 'Trachea epithelial single cell 90' 'Trachea epithelial single cell 12'\n",
      " 'Trachea epithelial single cell 301' 'Control #195' 'Emphysema #440'\n",
      " 'Control #444' 'Control #450' 'Emphysema #442'\n",
      " 'Infected Highly Sensitive 3 hours without infection'\n",
      " 'Infected Highly Sensitive 3 hours after infection' 'Norm Low Sensitive'\n",
      " 'Infected Low Sensitive 3 hours after infection' 'Norm Highly Sensitive'\n",
      " '104T' '105N' '106T' '114T' '122T' '123N' '130T' '136T' '138T' '140T'\n",
      " '141N' '151N' '158T' '162T' '201N' '207N' '112T' '117N' '129N' '139N'\n",
      " '147N' '152T' '153N' '155N' '204T' '103N' '108T' '124T' '125N' '131N'\n",
      " '150T' '159N' '163N' '208T' '209N' '107N' '109N' '110T' '113N' '116T'\n",
      " '128T' '137N' '157N' '160T' '202T' '203N' '206T' '102T' '111N' '115N'\n",
      " '120T' '121N' '146T' '154T' '156T' '161N' '200T' '205N' '211N' '220T'\n",
      " '221N' '224T' '225N' 'HRE93T' '307N' '332T' '341N' '342T' '344T' '347N'\n",
      " '356T' '357N' '359N' '414T' '430T' 'HRE 101T' 'HRE 105T' 'HRE 242N'\n",
      " 'HKT 2T' 'HKT 7N' '219N' '228T' 'HRE115T' 'HRE221N' '306T' '319N' '323N'\n",
      " '337N' '345N' '349N' '358T' '408T' '412T' '431N' '433N' '438T' 'HRE 243T'\n",
      " 'HKT 7T' 'HKT 28T' '226T' '227N' '230T' 'HRE148N' 'HRE149T' 'HRE222T'\n",
      " '318T' '340T' '343N' '346T' '355N' '410T' '411N' '413N' 'HRE 84T'\n",
      " 'HRE 87T' '210T' '216T' '217N' '223N' '231N' 'HRE92N' '333N' '336T'\n",
      " '348T' '409N' '432T' '439N' 'HRE83N' 'HKT 2N' '218T' '222T' '229N' '232T'\n",
      " '233N' 'HRE114N' '322T' '324T' '325N' '354T' '415N' 'HRE 86N' 'HRE 101N'\n",
      " 'HRE 104N' 'HKT 28N' 'PATIENT01' 'PATIENT04' 'PATIENT03' 'PATIENT05'\n",
      " 'PATIENT02' 'MCF10A Vector Control pQCXI-replicate1'\n",
      " 'MCF10A Vector Control pQCXI-replicate2' 'MCF10A YAP-5SA-replicate2'\n",
      " 'MCF10A RUNX3-replicate1' 'MCF10A YAP-5SA-replicate1'\n",
      " 'Ets2 null fibroblasts with no oncogene present in epithelial cells'\n",
      " 'Ets2 present in fibroblasts with the ErbB2 oncogene present in epithelial cells'\n",
      " 'Ets2 present in fibroblasts with no oncogene present in epithelial cells'\n",
      " 'Ets2 null fibroblasts with the ErbB oncogene present in epithelial cells'\n",
      " 'Grhl2 downregulation in MLE 15 cells 1'\n",
      " 'Grhl2 downregulation in MLE 15 cells 3' 'Control 1'\n",
      " 'Grhl2 downregulation in MLE 15 cells 2' 'Control 2'\n",
      " 'MCF10A cells cocultured with breast cancer associated fibroblasts that extracted from breast cancer sample No 4'\n",
      " 'MCF10A cells cocultured with breast cancer associated fibroblasts that extracted from breast cancer sample No 12'\n",
      " 'MCF10A cells cocultured with breast cancer associated fibroblasts that extracted from breast cancer sample No 15'\n",
      " 'MCF10A cells cocultured with normal breast fibroblasts that extracted from normal breast tissue sample No 16'\n",
      " 'MCF10A cells cocultured with normal breast fibroblasts that extracted from normal breast tissue sample No 23'\n",
      " 'RNA-seq, poly(A) selected, overexpression control GFP'\n",
      " 'RNA-seq, poly(A) selected, knockdown control'\n",
      " 'RNA-seq, poly(A) selected, knockdown Hdlbp/Vigilin'\n",
      " 'RNA-seq, poly(A) selected, overexpression human HDLBP/Vigilin'\n",
      " 'T10_A549' 'T10_A549Si' 'T10_H322' 'T10_H322_OE' 'DDC_only_2'\n",
      " 'PBS_only_1' 'PBS_only_2' 'DDC_only_1' 'DDC_only_3' 'Untransduced_36h_C'\n",
      " 'Untransduced_a144h_A' 'GFP_48h_B' 'GFP_36h_B' 'GFP_72h_A'\n",
      " 'SC_SNI_mRNA_1' 'DRG_Sham_mRNA_1' 'DRG_Sham_RFP_2' 'DRG_SNI_RFP_1'\n",
      " 'DRG_SNI_RFP_2' 'N4_liver_normal diet_4 weeks'\n",
      " 'M24_liver_high fat diet_4 weeks' 'M36_liver_high fat diet_4 weeks'\n",
      " 'N2_liver_normal diet_4 weeks' 'N3_liver_normal diet_4 weeks'\n",
      " 'Direct -PMN 1' 'Direct +PMN 1' 'Direct +PMN 2' 'Direct -PMN 3'\n",
      " 'Direct -PMN 2' 'Errb mutant stria vascularis replicate 2'\n",
      " 'wild type stria vascularis replicate 2' 'WT rep 1' 'WT rep 2'\n",
      " 'Gas KO rep 2' 'WT rep 3' 'Gas KO rep 1' 'p3143_rep1' 'MEDIUM_rep1'\n",
      " 'PT_rep1' 'MEDIUM_rep2' 'PTG_rep1' 'C12.1h repl. 1' 'C12.4h repl. 1'\n",
      " 'C12.4h repl. 3' 'C12.1h repl. 2' 'C12.4h repl. 2'\n",
      " 'Bmi1 wild-type lung cells, biological replicate 3'\n",
      " 'Bmi1 mutant lung cells, biological replicate 2'\n",
      " 'Bmi1 wild-type lung cells, biological replicate 2'\n",
      " 'Bmi1 wild-type lung cells, biological replicate 1'\n",
      " 'Bmi1 mutant lung cells, biological replicate 1'\n",
      " 'Liver Transcriptome from CTROLS_613'\n",
      " 'Liver Transcriptome from CTROLS_313'\n",
      " 'Liver Transcriptome from CTROLS_1613'\n",
      " 'Liver Transcriptome from CTROLS_1713'\n",
      " 'Liver Transcriptome from CTROLS_1113'\n",
      " 'Lung of Pneumocystis infected mice,biological rep1'\n",
      " 'Lung of Pneumocystis infected mice,biological rep3'\n",
      " 'Lung of Pneumocystis infected mice,biological rep4'\n",
      " 'Lung of Pneumocystis infected mice,biological rep2'\n",
      " 'Lung of Pneumocystis infected mice,biological rep5'\n",
      " 'HMC_L.T.R_siKDM3A_RNASeq' 'HMC_L.T.R_RNASeq' 'HMC_L.T_shKDM3A_RNASeq'\n",
      " 'HMC_P6_RNASeq' 'HMC_L_RNASeq' 'Control, biological replicate 2'\n",
      " 'Control, biological replicate 1' 'Piz6, biological replicate2'\n",
      " 'Control, biological replicate 3' 'Piz6, biological replicate1'\n",
      " '8HR_novirus_DMSO_3' '24HR_novirus_DMSO_2' '4HR_novirus_Silymarin_1'\n",
      " '24HR_novirus_DMSO_1' '24HR_novirus_Silymarin_2' 'Total_4' 'Total_1'\n",
      " 'Total_2' 'Total_3' 'IPF Rapidly lung fibroblst 1-2'\n",
      " 'IPF Rapidly lung fibroblst 1-1' 'IPF Rapidly lung fibroblst 1-4'\n",
      " 'IPF Slowly lung fibroblast 2-1' 'IPF Rapidly lung fibroblst 1-3'\n",
      " 'SxT_lung_2' 'SPC-HA_lung_2' 'SxT_lung_1' 'SPC-HA_lung_1' 'SPC-HA_lung_3'\n",
      " 'WTd3_B' 'WTd0_C' 'WTd0_B' 'WTd1_A' 'WTd1_C'\n",
      " '4TO7 pMSCV-miR-23b/27b/24 Replicate 1'\n",
      " '4TO7 pMSCV-puro Vector Replicate 1' '4TO7 pMSCV-puro Vector Replicate 2'\n",
      " '4TO7 pMSCV-miR-23b/27b/24 Replicate 2' 'Jeko NR4A1 sgRNA#1'\n",
      " 'Rec-1 NR4A1 sgRNA#2' 'Jeko control' 'Jeko NR4A1 sgRNA#2' 'Rec-1 control'\n",
      " 'Rec-1 NR4A1 sgRNA#1' 'liver Control Standard Diet 1'\n",
      " 'liver Control Standard Diet 5' 'liver Control Standard Diet 2'\n",
      " 'liver Control Standard Diet 3' 'liver Control Standard Diet 4'\n",
      " 'EB_d6_shLuc_ctrl (RNA-Seq)' 'EB_24h_shSmyd5 (RNA-Seq)'\n",
      " 'EB_d10_shSmyd5 (RNA-Seq)' 'mES_shLuc_ctrl (RNA-Seq)'\n",
      " 'EB_d14_shLuc_ctrl (RNA-Seq)' 'FoxF1 transfected cells, rep 1'\n",
      " 'FoxF1 transfected cells, rep 3' 'NFI-C2 transfected cells, rep 1'\n",
      " 'NFI-C2 transfected cells, rep 2' 'FoxF1 transfected cells, rep 2'\n",
      " 'control epithelium, biological rep2'\n",
      " 'control epithelium, biological rep1'\n",
      " 'control epithelium, biological rep3' '48hpi epithelium, biological rep2'\n",
      " '48hpi epithelium, biological rep1' 'HepG2b' 'Huh75b' 'HepG2a' 'Huh75a'\n",
      " 'CD_Liver_2L' 'CD_Liver_1L' 'CD_Liver_1R' 'CD_Liver_2N' 'CD_Liver_3N'\n",
      " 'Gene expression in GLN vs PDLN (NOD mouse #2)'\n",
      " 'Gene expression in GLN vs PDLN (NOD mouse #1)'\n",
      " 'Gene expression in GLN vs PDLN (NOD mouse #4)'\n",
      " 'Gene expression in GLN vs PDLN (NOD mouse #3)'\n",
      " 'Huh-7.5 cells treated with mock (0.1% ethanol) 2'\n",
      " 'Huh-7.5 cells treated with calcitriol (10 nM) 2'\n",
      " 'Huh-7.5 cells treated with calcitriol (10 nM) 1'\n",
      " 'Huh-7.5 cells treated with mock (0.1% ethanol) 1'\n",
      " 'Huh-7.5 cells treated with calcipotriol (10 nM) 2' 'WT-1' 'WT-2' 'WT-3'\n",
      " 'p62KO-2' 'p62KO-1' 'FF, patient 2' 'FF, patient 3'\n",
      " 'FFPE, patient 1, 3 hours tissue lysis' 'FF, patient 1'\n",
      " 'FFPE, patient 2, 3 hours tissue lysis'\n",
      " 'U87 cell as normal control, biological rep3'\n",
      " 'U87 cell as normal control, biological rep1'\n",
      " 'PHF20 gene was koncked down, biological rep2'\n",
      " 'PHF20 gene was koncked down, biological rep1'\n",
      " 'U87 cell as normal control, biological rep2' 'Control replicate A'\n",
      " 'Control replicate B' 'Control replicate C' 'miR-372 replicate C'\n",
      " 'miR-372 replicate B' 'AAV-TBG-WT FXR #1' 'AAV-TBG-WT FXR #2'\n",
      " 'AAV-TBG-Y67F FXR #2' 'AAV-TBG-WT FXR #3' 'AAV-TBG-Y67F FXR #1'\n",
      " 'Dap12-ko microglia 2_8' 'Dap12-ko microglia 3_11' 'Dap12-wt microglia 6'\n",
      " 'Dap12-wt microglia 7_12' 'Dap12-wt microglia 4_5' 'ActA0_3' 'ActA0_1'\n",
      " 'ActA10_1' 'ActA0_2' 'ActA10_2'\n",
      " 'Splenocytes_18h_post_VSVinPBS (A912b_02)'\n",
      " 'Splenocytes_18h_post_PBS (A912b_08)'\n",
      " 'Splenocytes_18h_post_VSVinPBS (A912b_04)' 'Wild Type 1' 'Wild Type 2'\n",
      " 'Wild Type 3' 'Scnn1b-Transgenic 2' 'Scnn1b-Transgenic 1'\n",
      " 'CD4+ cells from peripheral blood from child I074'\n",
      " 'CD4+ cells from peripheral blood from child I082'\n",
      " 'CD4+ cells from peripheral blood from child I093'\n",
      " 'CD4+ cells from peripheral blood from child I097'\n",
      " 'CD4+ cells from peripheral blood from child I095' 'P4 SP-C+ AEC2s_1'\n",
      " 'P4 SP-C+ AEC2s_4' 'P4 SP-C+ AEC2s_2' 'P4 SP-C+ AEC2s_3'\n",
      " 'P4 SP-C+ AEC2s_5' 'CSES12_WT_Teratoma' 'CSES13_T21_Teratoma'\n",
      " 'CSES22p16_T12_Teratoma' 'CSES45p30_T12_Teratoma' 'CSES20_T21_Teratoma'\n",
      " 'Control Rep #1' 'Control Rep #2' 'Control Rep #3' 'caBmpr1a Rep #1'\n",
      " 'caBmpr1a Rep #2' 'Mock_Liver' 'MCSE_Adipose tissue' 'MCSE_Muscle'\n",
      " 'Mock_Muscle' 'Mock_Adipose tissue' 'D4fl_APCfl_r2' 'd3fl_PAF776'\n",
      " 'D4fl_APCfl_r1' 'd3fl_BCAB107' 'd3fl_CAB1757'\n",
      " 'MDA_MB_231-not_transfected-3' 'MDA_MB_231-transfected_with_wtCopFry-1'\n",
      " 'MDA_MB_231-not_transfected-2' 'MDA_MB_231-transfected_with_wtCopFry-2'\n",
      " 'MDA_MB_231-not_transfected-1' 'ZnD-6d_PAX_pooled' 'ZnD-6d_GRP_pooled'\n",
      " 'Baseline_PAX_pooled' 'ZnD-10d_PAX_pooled' 'Baseline_GRP_pooled'\n",
      " 'Ramified microglia from 4-week rat brain biological rep1'\n",
      " 'Amoeboid microglia from 5-day rat brain biological rep2'\n",
      " 'Amoeboid microglia from 5-day rat brain biological rep3'\n",
      " 'mouse jejunum_hnf1alphamutant_rep2'\n",
      " 'mouse jejunum_hnf1alphacontrol_rep1'\n",
      " 'mouse jejunum_hnf1alphamutant_rep1' 'mouse jejunum_hnf1alphamutant_rep3'\n",
      " 'mouse jejunum_hnf1alphacontrol_rep2' '373T1' '393T1' '373N1'\n",
      " 'NCRM-5 Replicate 1' 'iMG (NCRM-5) Replicate 1' 'NCRM-5 Replicate 3'\n",
      " 'NCRM-5 Replicate 2' 'iMG (NCRM-5) Replicate 2'\n",
      " 'IntestineEC_MHCIIlo_rep1' 'IntestineEC_MHCIIhi_rep1'\n",
      " 'IntestineEC_MHCIIlo_rep2' 'IntestineEC_MHCIIhi_rep2' 'siCont'\n",
      " 'siAKR1B10+ADR' 'siCont+ADR' 'EPEC_WT_rep2' 'EPEC_TTSSKO_rep2'\n",
      " 'EPEC_WT_rep1' 'EPEC_TTSSKO_rep1' 'Control_rep1' '1785 vs 1754'\n",
      " '1803 vs 1802' '304 vs 303' 'TBE cells with IL-6' 'TBE cells with IL-17'\n",
      " 'TBE cells No RA' 'TBE cells with RA'\n",
      " 'Lipid-free apolipoprotein A-I, biological rep 3'\n",
      " 'control, biological rep 2' 'control, biological rep 1'\n",
      " 'Lipid-free apolipoprotein A-I, biological rep 1'\n",
      " 'Lipid-free apolipoprotein A-I, biological rep 2'\n",
      " 'A2B5/OTMP+ chipA biological replicate-4'\n",
      " 'A2B5/OTMP+ chipA biological replicate-1'\n",
      " 'A2B5/OTMP+ chipA biological replicate-2'\n",
      " 'A2B5/OTMP+ chipA biological replicate-3' 'MCF10A 2' 'MDA-468 2'\n",
      " 'MCF10A 1' 'MDA-468 1' 'MCF10A 3' 'Basal 2' 'Basal 3' 'Non-Basal 1'\n",
      " 'Non-Basal 2' 'Basal 1' '6967_IZ_RNA' '7012_IZ_RNA' '7172_CV_RNA'\n",
      " '7194_IZ_RNA' '6610_CV_RNA' 'REF colon repl2' 'STAT5b-/- colon repl2'\n",
      " 'STAT5b-/- colon repl1' 'REF colon repl1' 'STAT5b-/- colon repl3' 'ANS2'\n",
      " 'WTHD2r' 'WTHD1l' 'Baboon brain lcpufa control 1'\n",
      " 'Baboon brain lcpufa control 2' 'Baboon brain lcpufa L 1'\n",
      " 'Baboon brain lcpufa control 3' 'Baboon brain lcpufa control 4'\n",
      " 'E13_Testis_Control_biological rep2' 'E14_Testis_Control_biological rep2'\n",
      " 'E13-Cultured_Testis_Vinclozolin_biological rep2'\n",
      " 'E14_Testis_Vinclozolin_biological rep1'\n",
      " 'E14_Testis_Vinclozolin_biological rep2'\n",
      " 'E16_Testis_Control_biological rep1'\n",
      " 'E13-Cultured_Testis_Control_2_biological rep1'\n",
      " 'E13-Cultured_Testis_Control_2_biological rep2'\n",
      " 'E13_Testis_Vinclozolin_biological rep2'\n",
      " 'E14_Testis_Control_biological rep1'\n",
      " 'E13-Cultured_Testis_Control_biological rep1'\n",
      " 'E13-Cultured_Testis_Control_biological rep2'\n",
      " 'E13-Cultured_Testis_Vinclozolin_biological rep1'\n",
      " 'E13_Testis_Vinclozolin_biological rep1'\n",
      " 'E16_Testis_Control_biological rep2'\n",
      " 'E16_Testis_Vinclozolin_biological rep1'\n",
      " 'E16_Testis_Vinclozolin_biological rep2' 'RCC022 VHLwt' 'RCC407 VHLmut'\n",
      " 'RCC407 Tumor' 'RCC234 Tumor' 'RCC323 Tumor'\n",
      " 'NSCLC tumor, biological rep 1' 'NSCLC tumor, biological rep 2'\n",
      " 'NSCLC tumor, biological rep 3' 'NSCLC tumor, biological rep 4'\n",
      " 'NSCLC tumor, biological rep 5' '6853: D1_IFNg+IL10+'\n",
      " '6851: D1_IFNg-IL10-' '6854: D1_IFNg-IL10+' '6852: D1_IFNg+IL10-'\n",
      " '6855: D2_IFNg-IL10-' 'ΔIEC-adjacent 3' 'flox-adjacent 3'\n",
      " 'flox-adjacent 2' 'Mouse mammary gland Ccnd1+/+ E2 Rep1'\n",
      " 'Mouse mammary gland Ccnd1+/+ E2 Rep3'\n",
      " 'Mouse mammary gland Ccnd1-/- Vehicle Rep1'\n",
      " 'Mouse mammary gland Ccnd1-/- E2 Rep2'\n",
      " 'Mouse mammary gland Ccnd1-/- E2 Rep3'\n",
      " 'Mouse mammary gland Ccnd1-/- Vehicle Rep3'\n",
      " 'Mouse mammary gland Ccnd1+/+ Vehicle Rep3'\n",
      " 'Mouse mammary gland Ccnd1-/- E2 Rep1'\n",
      " 'Mouse mammary gland Ccnd1+/+ Vehicle Rep1'\n",
      " 'Mouse mammary gland Ccnd1+/+ Vehicle Rep2'\n",
      " 'Mouse mammary gland Ccnd1+/+ E2 Rep2'\n",
      " 'Mouse mammary gland Ccnd1-/- Vehicle Rep2' 'P15 Control 3' 'E16 T2MG 3'\n",
      " 'P15 T2MG 1' 'P15 T2MG 2' 'P15 Control 1' 'Nose14 (Current Smoker)'\n",
      " 'Nose10 (Never Smoker)' 'Nose11 (Current Smoker)' 'Nose12 (Never Smoker)'\n",
      " 'Nose13 (Never Smoker)' 'HCC827-RB1.1-rep1' 'HCC827-RB1-rep1'\n",
      " 'HCC827-rep1' 'HCC4006-rep1' 'HCC827-RA1-rep1'\n",
      " 'hSKP-HPC control condition, biological rep3'\n",
      " 'hSKP-HPC + Na-VPA, biological rep1'\n",
      " 'hSKP-HPC control condition, biological rep2'\n",
      " 'hSKP-HPC control condition, biological rep1'\n",
      " 'hSKP-HPC + Na-VPA, biological rep2' 'primaryAstro_unstimulated_rep1'\n",
      " 'primaryAstro_unstimulated_rep2' 'primaryAstro_IFNg_12hr_rep2'\n",
      " 'primaryAstro_IFNa_12hr_rep1' 'primaryAstro_IFNg_12hr_rep1'\n",
      " 'H1993 KD replicate 1' 'H1993 PLKO CONTROL replicate 1'\n",
      " 'H1693 PLKO CONTROL replicate 1' 'H1693 KD replicate 1' 'control, rep 1'\n",
      " 'control, rep 2' 'control rep 3' 'mutant p53, rep1' 'mutant p53, rep2'\n",
      " 'Uninfected spleen, day 0, rep3' 'Uninfected spleen, day 0, rep4'\n",
      " 'Uninfected spleen, day 0, rep2' 'Uninfected spleen, day 0, rep1'\n",
      " 'LCMV-Armstrong spleen day 5, rep1' 'cells_control_rep5'\n",
      " 'cells_control_rep3' 'cells_control_rep2' 'cells_control_rep4'\n",
      " 'cells_control_rep1' '(R)-Criz + CDDP_4' 'Ctrl_3' '(R)-Criz + CDDP_3'\n",
      " 'Ctrl_1' 'Ctrl_2' 'AB398 (Sample 4)' 'AB399 (Sample 5)'\n",
      " 'AB336 (Sample 2)' 'AB397 (Sample 3)' 'AB335 (Sample 1)' '1805Q2'\n",
      " '1805Q3' '1806Q3' '1805Q1' '1805Q4' 'BM-MSC culture rep 1'\n",
      " 'CD235a+ sorted rep 1' 'BM-MSC culture rep 2' 'CD235a+ sorted rep 2'\n",
      " 'BM-MSC culture rep 3' 'RNA-seq_598-SKD_3dDox_rep3'\n",
      " 'RNA-seq_598-SKD_0dDox_rep1' 'RNA-seq_598-SKD_0dDox_rep3'\n",
      " 'RNA-seq_598-SKD_3dDox_rep2' 'RNA-seq_598-SKD_0dDox_rep2']\n",
      "5      1.0\n",
      "6      1.0\n",
      "18     1.0\n",
      "19     1.0\n",
      "21     1.0\n",
      "      ... \n",
      "596    1.0\n",
      "597    1.0\n",
      "599    1.0\n",
      "601    1.0\n",
      "602    1.0\n",
      "Name: ctrl, Length: 87, dtype: float64\n",
      "87\n",
      "87.0\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "print(train['title'].unique()) \n",
    "newsr = train['title'].map(lambda x : 1 if (('control' in x.lower())  or ('baseline' in  x.lower())  or ('ctrl' in x.lower()) or ('ctrols' in x.lower()) or ('mock' in x.lower())) else 0) \n",
    "print(train[newsr == 1]['ctrl'])  \n",
    "print(newsr.sum())\n",
    "print(train[newsr == 1]['ctrl'].sum())  \n",
    "print('control' in 'control, rep 1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "type\n",
      "RNA    0.481481\n",
      "SRA    0.518182\n",
      "Name: ctrl, dtype: float64\n"
     ]
    }
   ],
   "source": [
    "train['type'].unique() \n",
    "print(train.groupby(['type']).mean()['ctrl'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Public on Oct 28 2015' 'Public on Dec 31 2015' 'Public on Feb 03 2011'\n",
      " 'Public on Jul 25 2018' 'Public on Feb 01 2018' 'Public on Jul 30 2018'\n",
      " 'Public on Jan 01 2011' 'Public on Nov 25 2015' 'Public on Jan 03 2014'\n",
      " 'Public on Jul 12 2013' 'Public on Feb 26 2018' 'Public on Dec 01 2013'\n",
      " 'Public on Sep 10 2012' 'Public on Dec 01 2008' 'Public on Jan 01 2017'\n",
      " 'Public on Mar 27 2019' 'Public on Jul 01 2016' 'Public on Feb 19 2015'\n",
      " 'Public on Dec 18 2018' 'Public on Oct 01 2016' 'Public on Oct 31 2012'\n",
      " 'Public on Aug 01 2007' 'Public on Apr 03 2016' 'Public on Mar 21 2013'\n",
      " 'Public on Apr 20 2014' 'Public on Sep 02 2011' 'Public on Dec 22 2016'\n",
      " 'Public on May 03 2016' 'Public on Apr 08 2016' 'Public on Mar 23 2017'\n",
      " 'Public on Jul 01 2015' 'Public on Aug 25 2017' 'Public on Oct 31 2015'\n",
      " 'Public on Dec 31 2016' 'Public on Sep 26 2018' 'Public on Dec 23 2014'\n",
      " 'Public on Oct 25 2017' 'Public on Nov 03 2006' 'Public on Jun 12 2017'\n",
      " 'Public on Feb 04 2016' 'Public on Jan 01 2016' 'Public on Mar 01 2018'\n",
      " 'Public on Jul 25 2016' 'Public on Apr 23 2009' 'Public on Jun 14 2018'\n",
      " 'Public on Nov 23 2016' 'Public on Apr 13 2018' 'Public on Oct 31 2017'\n",
      " 'Public on Jun 21 2016' 'Public on Apr 27 2018' 'Public on Aug 24 2011'\n",
      " 'Public on Sep 02 2015' 'Public on Apr 20 2016' 'Public on Jul 24 2014'\n",
      " 'Public on Oct 01 2014' 'Public on Nov 30 2016' 'Public on Jun 10 2015'\n",
      " 'Public on May 01 2018' 'Public on Jan 31 2015' 'Public on Dec 31 2014'\n",
      " 'Public on Nov 15 2011' 'Public on Jun 11 2011' 'Public on Jan 27 2011'\n",
      " 'Public on Mar 02 2017' 'Public on Oct 31 2013' 'Public on Oct 06 2009'\n",
      " 'Public on Mar 28 2015' 'Public on Feb 27 2014' 'Public on Jun 01 2008'\n",
      " 'Public on Mar 21 2019' 'Public on Jan 31 2017' 'Public on Sep 13 2018'\n",
      " 'Public on Aug 15 2008' 'Public on Jun 17 2018' 'Public on Mar 10 2007'\n",
      " 'Public on Feb 14 2010' 'Public on Jun 01 2016' 'Public on Dec 01 2014'\n",
      " 'Public on Jan 30 2019' 'Public on Jan 02 2019' 'Public on Jul 16 2013'\n",
      " 'Public on Jan 10 2019' 'Public on Jul 03 2008' 'Public on Sep 23 2017'\n",
      " 'Public on Jan 30 2015' 'Public on Oct 20 2016' 'Public on Jun 22 2015'\n",
      " 'Public on Oct 25 2014' 'Public on Feb 15 2013' 'Public on Dec 09 2010'\n",
      " 'Public on Feb 25 2019' 'Public on Jul 13 2016' 'Public on Oct 02 2015'\n",
      " 'Public on Dec 15 2007' 'Public on Aug 01 2014']\n",
      "            mean  count\n",
      "newser                 \n",
      "2006    1.000000      5\n",
      "2007    0.416667     12\n",
      "2008    0.368421     19\n",
      "2009    0.000000      7\n",
      "2010    0.636364     22\n",
      "2011    0.361111     36\n",
      "2012    0.500000     10\n",
      "2013    0.617647     34\n",
      "2014    0.505319    188\n",
      "2015    0.363636     66\n",
      "2016    0.579545     88\n",
      "2017    0.500000     44\n",
      "2018    0.492308     65\n",
      "2019    0.370370     27\n",
      "[2015 2015 2015 2015 2015 2015 2015 2015 2011 2011 2011 2011 2011 2018\n",
      " 2018 2018 2018 2018 2018 2018 2018 2018 2018 2018 2018 2018 2018 2011\n",
      " 2011 2011 2011 2011 2015 2015 2015 2015 2015 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2014 2013\n",
      " 2013 2013 2013 2013 2018 2018 2018 2018 2018 2013 2013 2013 2013 2012\n",
      " 2012 2012 2012 2012 2008 2008 2008 2008 2008 2017 2017 2017 2017 2019\n",
      " 2019 2019 2019 2016 2016 2016 2016 2016 2015 2015 2015 2015 2015 2018\n",
      " 2018 2018 2018 2018 2016 2016 2016 2016 2016 2012 2012 2012 2012 2012\n",
      " 2007 2007 2016 2016 2016 2016 2016 2013 2013 2013 2013 2013 2014 2014\n",
      " 2014 2014 2014 2011 2011 2011 2011 2011 2016 2016 2016 2016 2016 2016\n",
      " 2016 2016 2016 2016 2016 2016 2016 2016 2016 2017 2017 2017 2017 2017\n",
      " 2015 2015 2015 2015 2015 2017 2017 2017 2017 2015 2015 2015 2015 2015\n",
      " 2016 2016 2016 2016 2016 2018 2018 2018 2018 2018 2014 2014 2014 2014\n",
      " 2017 2017 2017 2017 2017 2017 2006 2006 2006 2006 2006 2017 2017 2017\n",
      " 2017 2017 2016 2016 2016 2016 2016 2016 2016 2016 2016 2016 2018 2018\n",
      " 2018 2018 2016 2016 2016 2016 2016 2009 2009 2009 2009 2018 2018 2018\n",
      " 2018 2018 2016 2016 2016 2016 2016 2018 2018 2018 2018 2018 2017 2017\n",
      " 2017 2017 2017 2016 2016 2016 2016 2016 2018 2018 2018 2018 2018 2011\n",
      " 2011 2011 2011 2011 2015 2015 2015 2015 2015 2016 2016 2016 2014 2014\n",
      " 2014 2014 2014 2014 2014 2014 2014 2014 2016 2016 2016 2016 2016 2015\n",
      " 2015 2015 2015 2015 2018 2018 2018 2018 2018 2015 2015 2015 2015 2015\n",
      " 2015 2015 2015 2015 2015 2014 2014 2014 2014 2014 2011 2011 2011 2011\n",
      " 2011 2011 2011 2011 2011 2011 2011 2011 2011 2011 2011 2011 2017 2017\n",
      " 2017 2017 2017 2018 2018 2018 2018 2013 2013 2013 2016 2016 2016 2016\n",
      " 2016 2009 2009 2009 2015 2015 2015 2015 2014 2014 2014 2014 2014 2008\n",
      " 2008 2008 2008 2019 2019 2019 2019 2019 2017 2017 2017 2017 2017 2018\n",
      " 2018 2018 2018 2018 2008 2008 2008 2008 2008 2018 2018 2018 2007 2007\n",
      " 2007 2007 2007 2010 2010 2010 2010 2010 2010 2010 2010 2010 2010 2010\n",
      " 2010 2010 2010 2010 2010 2010 2016 2016 2016 2016 2016 2014 2014 2014\n",
      " 2014 2014 2019 2019 2019 2019 2019 2019 2019 2019 2013 2013 2013 2013\n",
      " 2013 2013 2013 2013 2013 2013 2013 2013 2019 2019 2019 2019 2019 2008\n",
      " 2008 2008 2008 2008 2017 2017 2017 2017 2017 2015 2015 2015 2015 2015\n",
      " 2016 2016 2016 2016 2016 2015 2015 2015 2015 2014 2014 2014 2014 2014\n",
      " 2013 2013 2013 2013 2013 2010 2010 2010 2010 2010 2019 2019 2019 2019\n",
      " 2019 2016 2016 2016 2016 2016 2015 2015 2015 2015 2015 2007 2007 2007\n",
      " 2007 2007 2014 2014 2014 2014 2014]\n"
     ]
    }
   ],
   "source": [
    "print(train['status'].unique()) \n",
    "newser = train['status'].map(lambda x: int(x[-4:])) \n",
    "t1 = train.copy() \n",
    "t1['newser'] = newser\n",
    "print(t1.groupby('newser').agg(['mean','count'])['ctrl'])\n",
    "print(newser.values)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Unnamed: 0                     0\n",
       "geo_accession                  0\n",
       "gse_id                         0\n",
       "channel_count                  0\n",
       "characteristics_ch1           47\n",
       "contact_address               14\n",
       "contact_city                  14\n",
       "contact_country              368\n",
       "contact_department          2363\n",
       "contact_email               1251\n",
       "contact_institute            236\n",
       "contact_name                  14\n",
       "contact_state               1491\n",
       "data_processing               61\n",
       "data_row_count               368\n",
       "description                 1441\n",
       "extract_protocol_ch1         337\n",
       "growth_protocol_ch1         3357\n",
       "hyb_protocol                1518\n",
       "label_ch1                   1236\n",
       "label_protocol_ch1          1523\n",
       "last_update_date               0\n",
       "molecule_ch1                   5\n",
       "organism_ch1                   0\n",
       "platform_id                   14\n",
       "scan_protocol               1523\n",
       "source_name_ch1                0\n",
       "status                         0\n",
       "submission_date                0\n",
       "supplementary_file          1548\n",
       "taxid_ch1                      9\n",
       "title                          0\n",
       "treatment_protocol_ch1      3036\n",
       "type                           0\n",
       "contact_phone               4050\n",
       "contact_laboratory          3749\n",
       "relation                    4758\n",
       "contact_fax                 5648\n",
       "biomaterial_provider_ch1    5987\n",
       "contact_web_link            5970\n",
       "characteristics_ch2         5532\n",
       "extract_protocol_ch2        5787\n",
       "label_ch2                   5532\n",
       "label_protocol_ch2          5787\n",
       "molecule_ch2                5527\n",
       "organism_ch2                5527\n",
       "source_name_ch2             5527\n",
       "taxid_ch2                   5527\n",
       "treatment_protocol_ch2      6012\n",
       "biomaterial_provider_ch2    6056\n",
       "growth_protocol_ch2         5999\n",
       "dtype: int64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test.isnull().sum()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['polyA RNA' 'total RNA']\n",
      "['total RNA' 'polyA RNA' nan 'genomic DNA']\n"
     ]
    }
   ],
   "source": [
    "print(train['molecule_ch1'].unique()) \n",
    "train.groupby(['molecule_ch1']).mean()['ctrl'] \n",
    "print(test['molecule_ch1'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Homo sapiens' 'Mus musculus' 'Rattus norvegicus' 'Papio anubis'\n",
      " 'Macaca mulatta']\n",
      "['Mus musculus' 'Homo sapiens' 'Rattus norvegicus' 'Macaca mulatta'\n",
      " 'Microcebus murinus' 'Macaca fascicularis' 'Pan troglodytes']\n"
     ]
    }
   ],
   "source": [
    "print(train['organism_ch1'].unique()) \n",
    "train.groupby(['organism_ch1']).agg(['mean','count'])['ctrl']  \n",
    "print(test['organism_ch1'].unique())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1 2]\n",
      "[1. 2.]\n"
     ]
    }
   ],
   "source": [
    "print(train['channel_count'].unique()) \n",
    "train.groupby(['channel_count']).agg(['mean','count'])['ctrl']  \n",
    "print(test['channel_count'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['facs sorting: CD44low/CD24high' 'facs sorting: Unsorted'\n",
      " 'facs sorting: CD44high/CD24low' 'tissue: human nasal polyp'\n",
      " 'infection: Mutant DlntA Listeria' 'infection: LntA-V5+ Listeria'\n",
      " 'disease state: Squamous cell carcinoma'\n",
      " 'disease state: Lung adenocarcinoma'\n",
      " 'protocol: exposed to side-stream cigarette smoke' 'protocol: Control'\n",
      " 'strain: C57BL/6' 'gender: Male' 'gender: Female' 'cell line: HELF-977'\n",
      " 'cell line: HAF-1608' 'tissue: clear cell renal cell carcinoma'\n",
      " 'tissue: normal kidney' 'individual: PATIENT01' 'individual: PATIENT04'\n",
      " 'individual: PATIENT03' 'individual: PATIENT05' 'individual: PATIENT02'\n",
      " 'cell line: MCF10A' 'strain: FVB/N'\n",
      " 'cell line: Lung Epithelial Cell Line MLE15'\n",
      " 'cell line, normal mammary epithelium, immortalized' 'cell line: A549'\n",
      " 'tissue: liver' 'vector: none' 'vector: SINLV.PGK.GFP'\n",
      " 'tissue: Dorsal section of lumbar spinal cord'\n",
      " 'tissue: Lumbar(L3 to L5) dorsal root ganglion' 'strain: Sprague-Dawley'\n",
      " 'cell line: T84'\n",
      " 'postnatal 3~5 days, mixed gender, C57BL6/129 mixed background'\n",
      " 'gender: male' 'cell line: CaCo2' 'treatment: C12 irradiated'\n",
      " 'treatment: C12 control' 'source tissue: lung' 'condition: Control'\n",
      " 'tissue: left lung' 'cell line: HMC-LTR' 'cell line: HMC-LT'\n",
      " 'cell line: HMC-P6' 'cell line: HMC-L' 'cell line: Huh7.5.1'\n",
      " 'cell type: colon' 'age: 62' 'age: 72' 'age: 68' 'age: 50'\n",
      " 'mouse strain: SPC-HAxTCR-HA' 'mouse strain: SPC-HA'\n",
      " 'cell type: primary airway epithelial cells' 'tumor type: mammary tumor'\n",
      " 'treatment: Jeko cell line transduced with NR4A1 sgRNA#1, and induced with Doxycyclin for 48H'\n",
      " 'treatment: Rec-1 cell line transduced with NR4A1 sgRNA#2, and induced with Doxycyclin for 48H'\n",
      " 'treatment: Jeko cell line transduced with the CRISPR/cas9 empty vector as a control, and induced with Doxycyclin for 48H'\n",
      " 'treatment: Jeko cell line transduced with NR4A1 sgRNA#2, and induced with Doxycyclin for 48H'\n",
      " 'treatment: Rec-1 cell line transduced  with the CRISPR/cas9 empty vector as a control, and induced with Doxycyclin for 48H'\n",
      " 'treatment: Rec-1 cell line transduced  with NR4A1 sgRNA#1, and induced with Doxycyclin for 48H'\n",
      " 'C57BL/6NIA Mice, 18months of age, Standard Diet, Ad Lib, Liver'\n",
      " 'cell line: R1 ES cell' 'cell line: HC11' 'strain: C57Bl/6'\n",
      " 'cell line: HepG2' 'cell line: Huh75' 'diet: CD'\n",
      " 'tissue: gastric lymph node (left pancreatic lymph node)'\n",
      " 'cell type: hepatoma' 'tissue: Liver'\n",
      " 'diagnosis: epileptic patient suffering from Rasmussen encephalitis'\n",
      " 'cell line: U87 cell' 'cell line: U1810' 'C57BL6 Dap12-deficient'\n",
      " 'C57BL6 wt' 'treatment: None' 'treatment: 10 ng/ml ActA'\n",
      " 'genotype: wild type' 'genotype: Scnn1b-Transgenic'\n",
      " 'group: healthy control (GK)' 'tissue: lung' 'cell line: CSES12'\n",
      " 'cell line: CSES13' 'cell line: CSES22' 'cell line: CSES45'\n",
      " 'cell line: CSES20' 'genotype: Sftpc-CreERT2/+; R26-tdTomato/+'\n",
      " 'genotype: Sftpc-CreERT2/+; R26-tdTomato/R26-caBmpr1a'\n",
      " 'strain: ob/ob mice (B6.V-Lepob/J)'\n",
      " 'genotype: AhCre-positive APCflox hom mice' 'cell line: MDA-MB-231'\n",
      " 'cell line: 231wCFry' 'age: Postnatal day 28' 'age: Postnatal day 5'\n",
      " 'strain: mixed background'\n",
      " 'cell type: lung adenocarcinoma primary tumor derived cell line'\n",
      " 'cell type: lung adenocarcinoma metastasis derived cell line (from lymph node metastasis)'\n",
      " 'cell type: NCRM-5' 'cell type: iMG (NCRM-5)'\n",
      " 'strain background: C57BL/6' 'cell line: HCT116' 'tissue: normal lung'\n",
      " 'tissue: trachea' 'cell type: Human Coronary Artery Endothelial Cells'\n",
      " 'Fluorescent activated cell sorted A2B5/OTMP+ oligodendrocytes isolated from post-natal day 7 rat whole brain'\n",
      " 'cell line: MDA-468' 'genotype: STAT5b-/-' 'tissue: kidney'\n",
      " 'Male, Age: 83 Days at Necropsy, Tissue: Brain'\n",
      " 'Female, Age: 85 Days at Necropsy, Tissue: Brain'\n",
      " 'Female, Age: 84 Days at Necropsy, Tissue: Brain'\n",
      " 'Male, Age: 85 Days at Necropsy, Tissue: Brain' 'tissue: Testis'\n",
      " 'cell type: Cell line' 'cell type: Tumor' 'tnm 7th ed stage: IIA'\n",
      " 'tnm 7th ed stage: IB' 'tnm 7th ed stage: IIB' 'subject: Donor 1'\n",
      " 'subject: Donor 2' 'strain: C57Bl6; FVB' 'Current Smoker' 'Never Smoker'\n",
      " 'cell type: NSCLC' 'treatment: untreated'\n",
      " 'treatment: IFNg stimulated for 12 hrs'\n",
      " 'treatment: IFNa stimulated for 12 hrs' 'cell line: H1993'\n",
      " 'cell line: H1693' 'tissue: oviductal epithelium' 'strain: C57BL6'\n",
      " 'cell type: lung adenocarcinoma cells' 'strain: CX3CR1-GFP+ C57BL/6J'\n",
      " 'cell type: Bronchoalveolar lavage cells'\n",
      " 'cell type: bone marrow-derived mesenchymal stem cells'\n",
      " 'cell type: erythropoietic cells sorted by CD235a labeling of BM cells immediately after BM aspiration'\n",
      " 'cell line: MCF7']\n"
     ]
    }
   ],
   "source": [
    "print(train['characteristics_ch1'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[nan 'Cy3' 'biotin' 'Biotin' 'Cy5' 'Cy-3' 'biotinylated UTP & CTP']\n",
      "                            mean  count\n",
      "label_ch1                              \n",
      "Biotin                  0.480000     50\n",
      "Cy-3                    1.000000      5\n",
      "Cy3                     0.442623     61\n",
      "Cy5                     0.555556     18\n",
      "biotin                  0.473262    374\n",
      "biotinylated UTP & CTP  0.800000      5\n",
      "[nan 'Cy3' 'biotin' 'phycoerythrin' 'Cy5' 'cy3' 'cy5' 'Biotin'\n",
      " 'A549 siE2F3_2' 'A549_Neg2' 'A549 siCT_2' 'A549 210_2' 'Cy-3'\n",
      " 'biotin, cy3' 'Phycoerythrin' 'Biotin, Cy3'\n",
      " 'Applied Biosystems Expression Array System' 'Cyanine 3 (Cy3)' 'Biotin '\n",
      " 'Biotin Allonamide Triphosphate' 'cy 3' 'Lowinput Quick Amp Labeling Kit'\n",
      " 'Biotin is incorporated during the amplification.  This binds a streptavidin-conjugated phycoerythrin during the post-hybridization staining process that provides the measurable signal.'\n",
      " 'Cy3-dCTP' 'enzo biotin label' 'Streptavidin-Cy3' 'total RNA'\n",
      " 'indocarbocyanine']\n"
     ]
    }
   ],
   "source": [
    "print(train['label_ch1'].unique()) \n",
    "print(train.groupby('label_ch1').agg(['mean','count'])['ctrl']) \n",
    "print(test['label_ch1'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['cd44high_dox_vs_etoh.txt and cd44high_etoh_vs_cd44low_etoh.txt'\n",
      " 'imecs_myc_dox_vs_vector_dox.txt'\n",
      " 'Gene expression of LPS-stimulated nasal fibroblasts for 12hr'\n",
      " 'Gene expression of untreated-nasal fibroblasts' nan 'AE1165' 'AE1172'\n",
      " 'AE1179' 'AE1155'\n",
      " 'This sample represents a single cell processed with the SMART-Seq2 protocol.'\n",
      " \"3' RNAseq reads aligned to mm10\" 'healthy control 1' 'patient 1'\n",
      " 'healthy control 2' 'healthy control 3' 'patient 2'\n",
      " 'Cop050HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop051HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop052HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop060HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop066HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop067HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop072HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop074HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop076HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop078HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop079HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop083HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop090HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop094HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop099HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop105HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop058HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop063HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop071HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop077HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop081HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop084HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop085HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop087HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop102HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop049HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop054HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop068HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop069HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop073HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop082HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop091HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop095HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop106HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop107HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop053HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop055HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop056HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop059HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop062HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop070HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop075HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop089HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop092HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop100HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop101HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop104HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop048HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop057HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop061HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop064HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop065HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop080HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop086HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop088HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop093HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop098HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop103HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop109HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop114HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop115HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop118HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop119HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop129HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop139HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop146HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop151HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop152HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop154HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop157HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop162HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop163HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop165HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop172HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop174HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop185HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop187HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop188HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop193HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop196HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop113HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop122HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop131HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop134HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop138HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop141HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop143HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop149HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop155HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop159HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop164HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop166HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop170HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop175HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop177HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop178HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop189HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop197HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop199HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop120HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop121HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop124HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop132HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop133HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop135HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop140HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop150HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop153HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop156HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop161HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop168HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop169HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop171HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop181HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop183HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop108HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop110HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop111HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop117HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop125HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop128HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop147HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop148HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop158HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop167HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop176HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop179HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop180HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop192HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop112HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop116HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop123HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop126HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop127HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop130HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop142HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop144HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop145HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop160HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop173HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop182HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop184HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop186HG-U133_Plus_2.RCCStg1To4.mas5-Signal'\n",
      " 'Cop198HG-U133_Plus_2.RCCStg1To4.mas5-Signal' 'PATIENT01' 'PATIENT04'\n",
      " 'PATIENT03' 'PATIENT05' 'PATIENT02'\n",
      " 'Gene expression from MCF10A Vector Control, replicate 1'\n",
      " 'Gene expression from MCF10A Vector Control, replicate 2'\n",
      " 'Gene expression from MCF10A YAP-5SA, replicate 2'\n",
      " 'Gene expression data from MCF10A RUNX3, replicate 1'\n",
      " 'Gene expression from MCF10A YAP-5SA, replicate 1'\n",
      " 'Gene expression data from mammary fibroblasts with Ets2 deletion and no oncogene present in epithelial cells'\n",
      " 'Gene expression data from mammary fibroblasts with Ets2 and the ErbB2 oncogene present in epithelial cells'\n",
      " 'Gene expression data from mammary fibroblasts with Ets2 and no oncogene present in epithelial cells'\n",
      " 'Gene expression data from mammary fibroblasts with Ets2 deletion and the ErbB2 oncogene present in epithelial cells'\n",
      " 'Grhl2-shRNA 1' 'Grhl2-shRNA 3' 'Non-silencing 1' 'Grhl2-shRNA 2'\n",
      " 'Non-silencing 2'\n",
      " 'gene expression data from MCF10A cells cocultured with breast fibroblasts'\n",
      " 'adenoviral protein overexpression, RNA-seq'\n",
      " 'adenoviral shRNA knockdown, RNA-seq' 'Gene expression in A549 cell line'\n",
      " 'Gene expression of GALNT10 knockdown in A549 cell line'\n",
      " 'Gene expression in H322 cell line'\n",
      " 'Gene expression of GALNT10 over-expression in H322 cell line'\n",
      " 'human primary hepatocytes' 'Ribosome footprint'\n",
      " 'Gene expression data from liver'\n",
      " 'Gene expression 2hrs after incubation with no neutrophils'\n",
      " 'Gene expression 2hrs after neutrophil transmigration'\n",
      " 'contact Jichao Chen (jchen35@stanford.edu) for details'\n",
      " 'Gene expression of colonic mucosa of 6 week old mice'\n",
      " 'Gene expression data from wild-type lung cells'\n",
      " 'Gene expression data from mutant lung cells'\n",
      " 'Liver Transcriptome from CTROLS_613'\n",
      " 'Liver Transcriptome from CTROLS_313'\n",
      " 'Liver Transcriptome from CTROLS_1613'\n",
      " 'Liver Transcriptome from CTROLS_1713'\n",
      " 'Liver Transcriptome from CTROLS_1113'\n",
      " 'Gene expression data from lung of wild-type mice 14d post Pneumocystis infection.'\n",
      " 'HMC' 'Gene expression data from liver of wild-type mouse'\n",
      " 'Gene expression data from liver of transgenic PiZ mouse' 'fibroblst 1-2'\n",
      " 'fibroblst 1-1' 'fibroblst 1-4' 'fibroblast 2-1' 'fibroblst 1-3'\n",
      " '12 week old female mouse' 'See Agilent Cat#740100 for product details'\n",
      " 'liver  Control Standard Diet 1' 'liver  Control Standard Diet 5'\n",
      " 'liver  Control Standard Diet 2' 'liver  Control Standard Diet 3'\n",
      " 'liver  Control Standard Diet 4'\n",
      " 'HC11 cell line a clone of the COMMA-1D mouse (Balb/c) mammary epithelial cell line'\n",
      " 'Gene expression data of normal mouse tracheal epithelium replicate 2'\n",
      " 'Gene expression data of normal mouse tracheal epithelium replicate 1'\n",
      " 'Gene expression data of normal mouse tracheal epithelium replicate 3'\n",
      " 'Gene expression data of 48hpi mouse tracheal epithelium replicate 2'\n",
      " 'Gene expression data of 48hpi mouse tracheal epithelium replicate 1'\n",
      " 'No additional information.'\n",
      " 'Gene expression data from human hippocampus (fresh frozen)'\n",
      " 'Gene expression data from human hippocampus (formalin-fixed paraffin-embedded)'\n",
      " 'CTRLA' 'CTRLB' 'CTRLC' 'mir372C' 'mir372B' 'Dap12-ko' 'wt'\n",
      " 'Sample Gene Profile option of Illumina BeadStudio software was used to export to gene expression data based on unique gene.'\n",
      " 'Splenocytes from VSV injected mice'\n",
      " 'Splenocytes from PBS injected mice (control for VSV infection)'\n",
      " 'S1844_074_CD4+.CEL' 'S1845_082_CD4+.CEL' 'S1846_093_CD4+.CEL'\n",
      " 'S1848_097_CD4+.CEL' 'S1847_095_CD4+.CEL'\n",
      " 'Gene expression data from P3 and adult type 2 alveolar epithelial cells isolated by cell sorting using YFP from SftpcCreERT2-R26REYFP mice and TdTomato+, EpCAM+, CD31-, CD45- from Axin2CreERT2-TdTomato mice'\n",
      " 'MDA-MB-231 breast tumor epithelial cells'\n",
      " 'MDA-MB-231 breast tumor epithelial cells stably transfected with the wild-type Cop Fry allele'\n",
      " 'Pooled'\n",
      " 'Gene expression data from ramified microglia isolated from corpus callosum of 4-week rat brain'\n",
      " 'Gene expression data from amoedboid microglia isolated from corpus callosum of 5-day rat brain'\n",
      " 'Monitor the effect of Hnf1alpha loss in the mouse jejunum epithelium.'\n",
      " 'A8147_373T1' 'A8176_393T1' 'A8150_373N1' 'SAMPLE 1' 'SAMPLE 4'\n",
      " 'SAMPLE 3' 'SAMPLE 2' 'SAMPLE 5' 'SAMPLE 10' 'SAMPLE 7' 'SAMPLE 11'\n",
      " 'SAMPLE 8'\n",
      " 'Gene expression after the transfection of siControl without adriamycin treatment'\n",
      " 'Gene expression after the transfection of siAKR1B10 with adriamycin treatment'\n",
      " 'Gene expression after the transfection of siControl with adriamycin treatment'\n",
      " 'IECs isolated 8 days post-infection from neonate mouse orally infected at birth with WT EPEC. Replicate 2'\n",
      " 'IECs isolated 8 days post-infection from neonate mouse orally infected at birth with ΔescV EPEC. Replicate 2'\n",
      " 'IECs isolated 8 days post-infection from neonate mouse orally infected at birth with WT EPEC. Replicate 1'\n",
      " 'IECs isolated 8 days post-infection from neonate mouse orally infected at birth with ΔescV EPEC. Replicate 1'\n",
      " 'IECs isolated from a healthy non-infected 9 days old neonate mouse. Replicate 1'\n",
      " 'The general reproducibility of the hybridization platform (labeling procedure, hybridization and detection) was assessed by means of self to self hybridization.'\n",
      " 'RWX101025_Remaley_Vickers_HDL_AI3A' 'RWX101025_Remaley_Vickers_HDL_C2'\n",
      " 'RWX101025_Remaley_Vickers_HDL_C1' 'RWX101025_Remaley_Vickers_HDL_AI1B'\n",
      " 'RWX101025_Remaley_Vickers_HDL_AI2B'\n",
      " 'Flow Cytometry was used to isolate A2B5/OTMP+ oligodendrocytes from postnatal day 7 rat brain. 4 bioloical replicates were obtained. This data can be compared to the previously submitted oligodendrocyte expression data (GSE5940).'\n",
      " 'MCF10A' 'MDA-468' 'gene expression data from adult mouse colon'\n",
      " 'tumor tissue of patient ANS2'\n",
      " 'tumor tissue of right kidney of patient WTHD2'\n",
      " 'tumor tissue of left kidney of patient WTHD1'\n",
      " 'The expression data were analyzed by MAS5 statistical algorithms resident in GCOS; in short, the signal intensity for each gene was calculated as the average intensity difference, represented by [Σ(PM - MM)/(number of probe pairs)], where PM and MM denote perfect-match and mismatch probes (Lockhart et al, 1996, Nat. Biotechnol. 14:1675-1680; Liu et al, 2002, Bioinformatics 18:1593-9; Hubbell et al, 2002, Bioinformatics 18(12):1585-92). GCOS-generated detection and change calls were used to identify robust expression changes.'\n",
      " 'Gene expression data from rat embryo testis at day E13 treated with DMSO (vehicle) in utero'\n",
      " 'Gene expression data from rat embryo testis at day E14 treated with DMSO (vehicle) in utero'\n",
      " 'Gene expression data from rat E13 embrio testis treated with Vinclozolin in vitro'\n",
      " 'Gene expression data from rat embryo testis at day E14 treated with Vinclozolin in utero'\n",
      " 'Gene expression data from rat embryo testis at day E16 treated with DMSO (vehicle) in utero'\n",
      " 'Gene expression data from rat E13 embryo testis treated with ethanol (vehicle) in vitro'\n",
      " 'Gene expression data from rat embryo testis at day E13 treated with Vinclozolin in utero'\n",
      " 'Gene expression data from rat E13 embrio testis treated with DMSO (vehicle) in vitro'\n",
      " 'Gene expression data from rat embryo testis at day E16 treated with Vinclozolin in utero'\n",
      " 'Sample_Ailles_1_22_VHLwt_RNA_A' 'Sample_Ailles_10_407_mut_RNA_A'\n",
      " 'Sample_Ailles_16_RCC_407_tumour_RNA'\n",
      " 'Sample_Ailles_17_RCC243_tumour_RNA' 'Sample_Ailles_14_RCC323_tumour_RNA'\n",
      " 'Lung tissue samples obtained from the University of Pittsburgh Lung Cancer SPORE Tissue Bank.'\n",
      " 'X414e' 'X410e' 'X408e' 'P15Ctrl3' 'E16T2MG3' 'P15T2MG1' 'P15T2MG2'\n",
      " 'P15Ctrl1'\n",
      " 'Gene expression data from current smoker nasal epithelial sample'\n",
      " 'Gene expression data from never smoker nasal epithelial sample'\n",
      " 'RB1.1_a' 'RB1_a' 'HCC827_a' 'HCC4006_a' 'RA1_a' 'hSKP-HPC CTL 3'\n",
      " 'hSKP-HPC + Na-VPA 1' 'hSKP-HPC CTL 2' 'hSKP-HPC CTL 1'\n",
      " 'hSKP-HPC + Na-VPA 2' 'replicate 1' 'replicate 2'\n",
      " 'HuPV_072513H_WL12_L.rma-Signal' 'HuPV_072513H_WL11_K.rma-Signal'\n",
      " 'HuPV_072513H_WL1_A.rma-Signal' 'HuPV_072513H_WL2_B.rma-Signal' 'N3A'\n",
      " 'N4A' 'N2A' 'N1A' '5208A' 'Replicate 5' 'Replicate 3' 'Replicate 2'\n",
      " 'Replicate 4' 'Replicate 1' 'Comb-4' 'PBS3' 'Comb-3' 'PBS1' 'PBS2'\n",
      " 'Raw data files were processed as single-end files since the read2 (mate) files contained only cell and molecule barcodes. This information is appended to the fastq entry header.'\n",
      " 'Biological replicate 2 MtbDsigH vaccination group'\n",
      " 'Biological replicate 3 MtbDsigH vaccination group'\n",
      " 'Biological replicate 2 BCG vaccination group'\n",
      " 'Biological replicate 1 MtbDsigH vaccination group'\n",
      " 'Biological replicate 1 BCG vaccination group'\n",
      " 'Gene expression data of human bone marrow derived mesenchymal stem cells.'\n",
      " 'gene expression data of human bone marrow erythroblasts'\n",
      " 'RNA-seq_598-SKD.txt']\n"
     ]
    }
   ],
   "source": [
    "print(train['description'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Total RNA was extracted using the RNeasy Mini Kit (Qiagen) including on-column DNase digestion 48 h after transfection. Poly-A RNA was isolated from total RNA using the NEBNext Poly(A)mRNA Magnetic Isolation Module (E7490).'\n",
      " 'Nasal fibroblasts were exposed to LPS (10 μg/mL) for 12 h. Total RNA was isolated using Trizol reagent (Invitrogen, Carlsbad, CA). For control and test RNAs, synthesis of target cRNA probes and hybridization were performed using the Low RNA Input Linear Amplification kit (Agilent Technology, Santa Clara, CA). Hybridized images were scanned using a DNA microarray scanner and quantified using Feature Extraction Software (Agilent). All data normalization and selection of fold-change of the genes were performed using GeneSpringGX 7.3 (Agilent).'\n",
      " 'RNA  was extracted using the RNeasy Mini Kit (Qiagen). RNA quality was monitored on Agilent RNA Pico LabChips (Agilent Technologies, Palo Alto, CA).'\n",
      " 'Maxell® RSC SimplyRNA Tissue'\n",
      " 'Lungs were cleared of BALF content, the trachea removed and the more distal lung lobe tissue enriched by dissection. The interstitial cells were collected and reacted with anti-CD45 antibody coupled to magnetic beads before depletion of the CD45+ cells using an autoMACS cell separator .The purity was (>95% epCAM+, <1% CD31+ endothelial cells) and viability (<5% 7-AAD (viability stain)'\n",
      " 'Smart-seq2 method described by Picelli et al. (Nature Protocols 2014)'\n",
      " \"3' polyA RNA seq\" 'RNA was extracted with Trizol reagent'\n",
      " 'We collected the cells 3 hours after infection. Total RNA was extracted using TRIzol Reagent.'\n",
      " 'RNA was extracted from flash frozen patient tissue samples (RCC tissue and matched normal kidney tissue) using TRIzol (Invitrogen) and chloroform (Sigma). All tumor samples are from the primary tumor site. RNA extracts were purified on Spin Chroma column (BD bioscience K1332-2). The 18S/28S bands were verified on a 1% agarose gel. RNA samples were shipped to Advanced Genomic Technology Center Gene Expression Core at the Mayo Clinic in Rochester, MN for further analysis. RNA quality was assessed by Agilent Bioanalyzer (Santa Clara, CA). Reverse transcription to second strand cDNA was generated from 100 ng of good quality total RNA.'\n",
      " 'We extracted total RNA utilizing RNeasy Fibrous Tissue (Qiagen) according to manufacturer’s instructions.'\n",
      " \"Total RNA was extracted by Qiagen RNEasy kit according to the manufacturer's instructions.\"\n",
      " \"Trizol extraction of total RNA was performed according to the manufacturer's instructions.\"\n",
      " 'Total RNA was isolated using QIAGEN’s RNeasy kit (Qiagen) and the sample integrity was verified using RNA 6000 Nano Assay RNA chips run in Agilent 2100 Bioanalyzer (Agilent Technologies).'\n",
      " \"Trizol extraction of total RNA was performed according to the manufacturer's instructions\"\n",
      " 'poly(A) selection'\n",
      " 'riefly, total RNA from control and GALNT10 siRNA knockdown A549s cells were extracted in triplicates using GeneJET RNA Purification kit (Thermo Scientific) following the manufacturer’s protocol and quantified by NanoDrop spectrophotometer. The quality of the RNA samples was examined by capillary electrophoresis using the Agilent 2100 Bioanalyzer (Agilent).'\n",
      " 'RNA was extracted with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit.'\n",
      " 'Total RNA including small RNAs from hepatocytes was isolated with the miRNeasy Mini Kit (Qiagen) at different time points'\n",
      " 'The animals were sacrificed 30\\u202fdays post-surgery, and lumbar DRG (L3 to L5) and\\xa0dorsal horn of corresponding spinal cord\\xa0section were extracted and frozen at -80 °C. Tissues from ten animals were pooled per sample and two independent replicates were made for each of the four conditions. Pooled tissue were lysed in lysis buffer containing cyclohexamide and the lysate was divided into two fractions. One fraction was used to purify poly(A) mRNA for RNA-seq and the remaining fraction was used to isolate ribosome footprints. For the former, poly(A) mRNA were purified using magnetic oligo-dT DynaBeads. The purified RNA was then subjected to alkaline fragmentation. Fragmented mRNAs were size-selected on a denaturing 10% polyacrylamide-urea gel and the bands corresponding to 30–50 nucleotides were excised, eluted, and precipitated with Isopropanol. For the later, lysate was treated with RNAse 1 followed by ultracentrifugation through a sucrose cushion. The resulting RNA pellet was resuspended in Tris HCl buffer and purified by double acid Phenol and one Chloroform extraction. Purified RNA was resolved on a 15% polyacrylamide urea gel and bands corresponding to 28–32 nucleotides, containing the desired ribosome footprints (RFPs), was excised and extracted for RNA using Costar Spin-X columns (Sigma).'\n",
      " \"RNA was prepared using TRIzol reagent (Invitrogen) following the manufacturer's recommendations. RNA was quantified using a NanoDrop-1000 spectrophotometer and quality was monitored with the Agilent 2100 Bioanalyzer (Agilent Technologies, Santa Clara, CA).\"\n",
      " 'Trizol + RNeasy' 'RNA were purified with Trizol reagent'\n",
      " 'The cells were collected by adding 1ml 1X trypsin-EDTA (Trypsin 10X, 15400-054 Invitrogen) to the flasks followed by 4-5 min incubation at 37°C. Trypsin was inhibited by adding 1ml MEM+10% FCS and the cells were collected by centrifugation at 9.600x g for 5 min. The pellet was washed with 500µl 1X PBS and RNA was extracted using RNeasy Plus Mini kit (74134, QIAGEN).'\n",
      " 'RNA was purified using RNeasy Mini Kit following the company’s instructions (Qiagen). Total RNA from two independent experiments was pooled.'\n",
      " 'RNA was purified using an Absolutely RNA Purification Kit (Agilent Technologies), and amplified using the WT-Ovation Pico RNA Amplification System (Nugen) to generate cDNA'\n",
      " 'Liver samples were fixed in 10% formalin, embedded in paraffin and sectioned (thickness of 2µm). Slides were stained with hematoxylin and eosin (H&E) and assessed by an expert pathologist (RM).'\n",
      " 'Total RNA was extracted by using Qiagen RNA extraction kit according to manufacturer’s manual, and then 10ug total RNA was used for downstream library construction. For ChIP asssy, approximately 1×107 cells were fixed with 1% formaldehyde and quenched by glycine. The cells were washed three times with PBS and then harvested in ChIP lysis buffer (50 mM Tris-HCl, pH7.6, 1 mM CaCl2, 0.2% Triton X-100). DNA was digested to 150-300 bp by MNase (SIGMA) before extensive centrifugation. Four volume of ChIP dilution buffer (20 mM Tris-HCl, pH 8.0, 150 mM NaCl, 2 mM EDTA, 1% Triton X-100, 0.1% SDS) was added to the supernatant. The resulted lysate was then incubated with protein G beads and antibodies at 4ºC over night. The beads were washed 5 times and DNA was eluted by Chip elution buffer (0.1 M NaHCO3, 1% SDS, 20 µg/ml proteinase K). The elution was incubated at 65ºC over night and DNA was extracted with DNA purification kit (TIANGEN).'\n",
      " 'Total RNAs from either Piz and wild-type livers were purified by QIAshredder (Qiagen spa, Milano, Italy, catalog no. 79656) and extracted with RNeasy Protect Mini Kit (Qiagen, catalog no. 74126) using standard conditions.'\n",
      " \"Total RNA was extracted using the Qiagen RNeasy Mini Kit (#74104) following the manufacturer's instructions.\"\n",
      " 'after immunopreipitation (or for total cell lysate), RNA was extracted using Trizol and purfied using the PureLink kit (Ambion) whole-transcriptome isolation protocol.'\n",
      " 'RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent Bioanalyser.'\n",
      " \"RNA was isolated using RNeasy Midi Kit together with the RNase free DNase kit (both from Qiagen) for DNA-digestion according to manufacturer's instructions. RNA was photometrically quantified on a NanoDrop devive and RNA integrity was checked on an Agilent2100 Bioanalyzer (Agilent).\"\n",
      " 'Total cellular RNA was extracted using  RNAqueous™ phenol-free total RNA isolation kits (Life Technologies, CA) according to the manufacturer’s recommendations'\n",
      " \"Total RNA was extracted using the miRVana total RNA isolation kit (Ambion) using the manufacturer's instructions.\"\n",
      " 'RNA was extracted by Rneasy plus mini kit (Qiagen) following the manufacturer’s instructions.'\n",
      " 'Total RNA was extracted with the Qiagen RNEasy Columns'\n",
      " 'RNA was harvest using RNeasy and miRNeasy Qiagen kits.'\n",
      " \"RNA was extracted using RNeasy Micro kit (Qiagen) according to the manufacturer's instructions.\"\n",
      " 'Using RNA later-soaked tissue specimens, total RNA was extracted using RNeasy Mini kit (Qiagen).'\n",
      " \"Gastric lymph node was excised and immediately homogenized in Trizol Reagent.  RNA was extracted in Trizol and then purified using the RNeasy kit (Qiagen). RNA purity and quality was assessed using the NanoDrop 1000 spectrophotometer (Thermo Scientific), Agilent RNA 6000 Nano Reagents, RNA Nano chips, and the Agilent 2100 bioanalyzer (Agilent Technologies), according to manufacturer's instructions.\"\n",
      " 'RNA were extracted with the RNeasy Micro Kit from Qiagen. RNA was amplified using the Ovation Pico WTA System V2 from NUGEN applied for 40ng total RNA.'\n",
      " 'Dissected liver sections were removed, flash frozen on dry ice, and RNA was harvested using Trizol RNA extraction.'\n",
      " \"Total RNA was isolated using the PicoPure RNA Isolation Kit (Thermo Fisher Scientific) according to the manufacturer's instructions.\"\n",
      " \"Total RNA was isolated using the High Pure FFPE RNA Micro Kit (Roche) according to the manufacturer's instructions.\"\n",
      " \"Total RNA in the samples was isolated using Trizol Reagent according to the manufacture's protocol.\"\n",
      " 'RNEasy' 'RNA from the livers was isolated with the RNeasy Kit (Qiagen)'\n",
      " 'Trizol and RNeasy'\n",
      " 'Total RNA was extracted from cells using the PureLink™ RNA Mini kit (Ambion) following the manufacturer protocol.'\n",
      " 'Tissues were dissected and flushed with ice‐cold HBSS. RNA was isolated using the RNeasy Mini Kit (Qiagen, Hilden, Germany) following the manufacturer′s instructions.'\n",
      " \"RNA from isolatedCD4+ cells was isolated using the RNeasy Mini Kit (Qiagen, Hilden, Germany) according to the manufacturer's instructions. The RNA quality and quantity was determined with Agilent Bioanalyzer chips (Bio-Rad Laboratories).\"\n",
      " 'RNA was extracted using Rneasy mini kit (Qiagene)'\n",
      " 'According to the manufacture instruction. RNA was subjected to Human Genome U133 Plus 2.0 microarray platform (Affymetrix); washing and scanning were performed according to the manufacturer’s protocol'\n",
      " 'Total RNA was isolated from FACS-sorted cells using TRIzol (Thermo Fisher Scientific) and Direct-zol RNA MicroPrep (Zymo Research).'\n",
      " 'Total RNAs were extracted from muscle, adipose tissue, and liver using RNeasy Mini kit (Qiagen, Valencia, CA).'\n",
      " 'trizol phenol extraction followed by Rneasy clean up'\n",
      " 'Total RNA was extracted from 70% confluent cell cultures using the RNeasy (QIAGEN) kit.'\n",
      " \"Total RNA from whole blood collected in PAXgene tubes (BD) was extracted using the Qiagen Rneasy kit according to manufactuer's instructions. Globin RNA contents in the isolated whole blood RNA was reduced by the GLOBINclear kit (Ambion). Purity and integrity of yielded RNA were assessed with the NanoDrop spectrophotometer and Agilent Bioanalyzer, respectively.\"\n",
      " \"Total RNA was extracted from 600 microglia cells isolated by Laser-capture microdissection using RNeasy micro kit according to the manufacturer's instructions.\"\n",
      " 'Extraction with the Ambion Totally RNA kit.'\n",
      " 'RNA was extracted using Trizol (Invitrogen)'\n",
      " \"RNA was extracted from cells using the miRNeasy mini kit (Qiagen) according to manufacturer's instructions.\"\n",
      " 'RNA was extracted with Trizol (thymus, Thyroid) or performed by Miltenyi Biotech following standard procedure'\n",
      " \"24h after infection or second transfection, total RNA was extracted by RNeasy mini kit (QIAGEN) according to manufacturer's protocol. RNA was quantified using the NanoDrop spectrophotometer.\"\n",
      " 'RNA was isolated using TRIzol (Invitrogen) according to the manufacturer’s protocol.'\n",
      " 'PolyA+ RNA was extracted using Dynabead magnetic particles.'\n",
      " 'The cells were harvested and total RNA was isolated using Qiazol miRNAeasy kits (Qiagen) and quantified by spectrophotometry'\n",
      " 'Qiagen RNAeasy'\n",
      " 'RNA from 10A/143B and 468/143B cybrids were extracted in triplicate and 5ug of total RNA was used for first- and second-strand cDNA synthesis.'\n",
      " 'FACS sorted cells were lysed for RNA preparation using QIAGEN RNAeasy micro-kit'\n",
      " 'RRBS: The frozen tissue was cut in 20 µm slices, mounted on precooled glass-slides and dried at room temperature for 30 minutes. In sum, an area of 200.000 µm2 per sample was microdissected using a ZEISS PALM MicroBeam LCM in Auto-LPC mode. Tissue pieces subjected for RRBS were frozen and maintained at -80° C. RNA-seq: The frozen tissue was cut in 20 µm slices and mounted on precooled glass-slides. Immediately after mounting, the tissue was fixed by incubation in 70% EtOH for 2 min and stained by incubation in a cresyl violet/50% EtOH solution for 30 sec with two short subsequent wash steps in 70% EtOH, followed by a 2 min air-drying step. All steps were performed on ice. In sum, an area of 500.000 µm2, from of 5-10 periportal fields and intermediate zones or hepatozytes surrounding 15-20 central veins per sample, were microdissected using a ZEISS PALM MicroBeam LCM in Auto-LPC mode. Tissue pieces subjected for RRBS were frozen and maintained at -80° C. Samples subjected for RNA-Seq were immediately processed.'\n",
      " 'RNeasy Mini kit (QIAGEN, Hilden, Germany)'\n",
      " \"RNA was prepared using the RNAeasy Mini Kit (QIAGEN, Valencia, CA) following the manufacturer's recommendations, with an included on-column DNase digestion. RNA was quantified using a NanoDrop-1000 spectrophotometer and quality was monitored with the Agilent 2100 Bioanalyzer (Agilent Technologies, Santa Clara, CA).\"\n",
      " \"Total RNA was isolated from approximately 50-100 mg of tissue from each animal using RNA STAT-60 according to the manufacturer's instructions (Tel Test Inc) [RNA Stat-60 is the same as Trizol reagent]. \"\n",
      " \"Trizol reagent (Sigma) extraction of total RNA was performed according to the manufacturer's instructions.\"\n",
      " \"Trizol reagent (Sigma) extraction of total RNA from tetstis was performed according to the manufacturer's instructions.\"\n",
      " 'RNA was extracted from frozen tumor tissue and cultured cells using the QiagenRNeasy Kit'\n",
      " 'RNA was extracted withQiazol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent 2100 Bioanalyzer.'\n",
      " 'Cell pellets for mRNA isolation were lysed in Trizol and RNA extracted by guanidium-isothescyanate phenol/chloroform extraction method following the manufacturer’s instructions'\n",
      " 'Colon tumors and adjacent tissues were dissected and immediately maintained in RNA stabilization solution. RNA was isolated with TRIzol. Samples were used with 1 ug of total RNA for the construction of sequencing libraries.'\n",
      " \"Total RNA extracted using Trizol reagent (Invitrogen) as per manufacturers' guidelines.\"\n",
      " 'RNA was extracted using the RNAeasy Plus Micro kit from Qiagen, following manufacturer´s instruction.'\n",
      " \"Trizol extraction of buccal mucosa samples performed based on manufacturer's protocol, with RNA from serial collections from same patient pooled to obtain minimum of 8ug for hybridization to HG-U133A_2 arrays; Nasal epithelial RNA isolated via Qiagen RNeasy Mini Kits per manufacturer’s protocol, with enough RNA from one collection to hybridize to HG-U133A_2 arrays.\"\n",
      " 'Total RNA was isolated using a total RNA purification kit (Norgen Biotek, Thorold, ON, Canada). RNA quantity was determined by absorbance at 260\\u2009nm using a NanoDrop ND-1000 UV-VIS spectrophotometer (Thermo Fisher Scientific, Wilmington, DE, USA). The quality and integrity of each sample were confirmed using a BioAnalyzer. RNA sample replicates for each cell line were labeled and assayed by using SurePrint G3 Human Gene Expression 8x60K v2 (G4851B - 039494) one-color microarray kit of Agilent Technologies. Microarray hybridization was performed according to the Agilent standard protocols.'\n",
      " \"Total mRNA was extracted using the TriPure Isolation ReagentTM (Roche Applied Science) according to the manufacturer's instructions.\"\n",
      " 'Total RNA was isolated using the Qiagen Rneasy Plus Kit'\n",
      " 'RNeasy kit with spin columns (Qiagen)'\n",
      " 'Harvested spleens were frozen in RNA Later (Qiagen) at 1 mg/ml tissue.  Whole spleens were homogenized, and RNA from homogenates were isolated with Rneasy extraction kit (Qiagen).'\n",
      " \"After 12 h of treatment with CLEFMA (1 µM), the cells were washed twice with Dulbecco’s PBS. The total RNA was extracted using RNAeasy kit (Qiagen, CA) as per the manufacturer’s instructions. The quality of RNA was ascertained by Nanodrop 2000. A total of 250 ng of total RNA from three biological replicates of untreated H441 cells and cells treated with CLEFMA were labeled using the Illumina Total Prep RNA Amplification Kit following manufacturer's directions (Ambion, Austin. TX).\"\n",
      " 'Tumors were harvested at day 10 after treatments and immediately immersed in the RNAlater RNA Stabilization Reagent (Qiagen, Hilden, Germany) and total RNA was extracted with the RNeasy Plus Mini Kit (Qiagen) following the manufacturers’ instructions.'\n",
      " 'Each mouse was euthanized and brain or Yolk sac was extracted for proccessing'\n",
      " 'For transcriptomics, RNA was isolated from total BAL obtained at the pre-vaccination and the 3 week post-vaccination time points as previously described (Mehra S et al, AJRCCM, 2015) using an RNAEasy kit (Qiagen), followed by RNA amplification (Ambion MessageAmp).'\n",
      " 'RNA was collected using the RNeasy mini kit (Qiagen)'\n",
      " 'For HA-tag ChIP-seq, stable MCF7 cell lines were induced using 100 ng/ml doxycycline (Sigma) at 0 h, doxycycline media was refreshed at 48 h, and cells were harvested at 72 h by crosslinking in a final concentration of 1% formaldehyde. Crosslinking was stopped after 5 min by adding glycine to a final concentration of 125 mM. Crosslinked cells were washed in cold phosphate buffered saline, lysed using 1 ml low-salt IP buffer (150 mM NaCl, 50 mM Tris-HCl (pH7.5), 5 mM EDTA, NP-40 (0.5%), Triton X-100 (1%) containing protease inhibitors) and aliquoted at 1 × 10∧7 cells/ml.']\n"
     ]
    }
   ],
   "source": [
    "print(train['extract_protocol_ch1'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[     0  34127  28231  48803  47310  54675  47279  33297  45101  34760\n",
      "  25697  31099  41093  45015  45220  31139  22690  28869  47231  28856\n",
      "  41174  38354  28815  45281  32321  53617  49395  41345  53590  29922\n",
      "  35110  26432  31097  45037  44343  43379  42405  39429 165703  22283\n",
      "  28829  15923  26180  41000  29376  42125  22277  58717  35557  20217]\n",
      "                    mean  count\n",
      "data_row_count                 \n",
      "0               0.518182    110\n",
      "15923           0.000000      4\n",
      "20217           0.000000      5\n",
      "22277           0.600000      5\n",
      "22283           0.250000      4\n",
      "22690           0.600000      5\n",
      "25697           0.800000     10\n",
      "26180           0.400000      5\n",
      "26432           0.400000      5\n",
      "28231           0.000000      5\n",
      "28815           0.300000     10\n",
      "28829           0.400000      5\n",
      "28856           0.600000      5\n",
      "28869           0.600000      5\n",
      "29376           0.000000      5\n",
      "29922           0.600000      5\n",
      "31097           0.000000      3\n",
      "31099           0.636364     22\n",
      "31139           1.000000      5\n",
      "32321           0.700000     10\n",
      "33297           0.400000     10\n",
      "34127           0.500000      8\n",
      "34760           0.400000      5\n",
      "35110           0.000000      5\n",
      "35557           0.800000      5\n",
      "38354           1.000000      5\n",
      "39429           0.600000      5\n",
      "41000           0.333333      3\n",
      "41093           0.600000      5\n",
      "41174           0.250000      8\n",
      "41345           0.307692     13\n",
      "42125           0.000000      5\n",
      "42405           0.666667      3\n",
      "43379           0.000000      4\n",
      "44343           0.000000      3\n",
      "45015           0.600000      5\n",
      "45037           0.400000      5\n",
      "45101           0.604651     43\n",
      "45220           0.400000      5\n",
      "45281           1.000000      5\n",
      "47231           0.300000     10\n",
      "47279           0.800000      5\n",
      "47310           0.600000      5\n",
      "48803           0.800000     10\n",
      "49395           0.555556      9\n",
      "53590           0.200000      5\n",
      "53617           0.600000     10\n",
      "54675           0.471910    178\n",
      "58717           0.400000      5\n",
      "165703          0.000000      3\n"
     ]
    }
   ],
   "source": [
    "print(train['data_row_count'].unique()) \n",
    "print(train.groupby('data_row_count').agg(['mean','count'])['ctrl'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "77.0\n",
      "151\n",
      "['nan'\n",
      " 'Amplified and labeled cRNA was purified on cRNA Cleanup Module (Agilent Technology) according to the manufacturer’s protocol. Labeled cRNA target was quantified using ND-1000 spectrophotometer (NanoDrop Technologies, Inc., Wilmington, DE).'\n",
      " '300 ng of total RNA were analysed using the Affymetrix Human Gene1.0 ST Array (exon array), according to the GeneChip whole transcript sense target labeling assay manual, using the GeneChip WT cDNA Synthesis and amplification Kit and WT terminal labeling Kit.'\n",
      " 'Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays'\n",
      " 'Total RNA was reverse-transcribed to cDNA and cRNA using the Ambion TotalPrep cRNA Amplification Kit (Invitrogen, USA). The cRNA concentration was quantified and adjusted to 150 ng/ml using an ND-1000 Spectrophotometer (NanoDrop Technologies, USA)'\n",
      " 'The RNA products were column-purified (Affymetrix) and then in vitro transcribed to generate biotin-labeled cRNA.'\n",
      " 'Using the Ambion Illumina TotalPrep RNA Amplification kit (Ambion #IL1791), we reverse-transcribed 300 ng of total RNA using a T7 oligo-dT primer, used the resulting cDNA to synthesize the complement strand, purified, and in vitro transcribed biotinylated cRNA.'\n",
      " 'Biotinylated cDNA were prepared according to the standard Affymetrix protocol from 200ng total RNA'\n",
      " 'Biotinylated cRNA was prepared according to the standard Affymetrix protocol from 2 ug total RNA (Expression Analysis Technical Manual, 2005, Affymetrix).'\n",
      " 'The total RNA (300 ng) was reverse transcribed using Whole Transcript cDNA Synthesis kit (Affymetrix). The cDNA obtained was used as a template for in vitro transcription using a Whole Transcript cDNA Amplification Kit (Affymetrix). The antisense cRNA obtained was purified using GeneChip Sample Cleanup Module (Affymetrix) and used as a template for reverse transcription (Whole Transcript cDNA Synthesis kit, Affymetrix) to produce single-stranded DNA in the sense orientation. During this step, dUTP was incorporated. Then, the DNA was fragmented using uracil DNA glycosylase (UDG) and apurinic/apyrimidinic endonuclease 1 (APE 1), labeled with DNA Labeling Reagent and covalently linked to biotin using terminal deoxynucleotidyl transferase (TdT, Whole Transcript Terminal Labeling kit, Affymetrix). IVT and cDNA fragmentation quality controls were performed by an mRNA Nano assay in the Agilent 2100 Bioanalyzer.'\n",
      " 'Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 6 ug total RNA (Expression Analysis Technical Manual, 2001, Affymetrix)'\n",
      " 'Fluorescently labeled cDNA targets were generated from 10 μg of total RNA from each corresponding A549s cell clone following user’s manual. Agilent Gene Expression Hybridization Kit was used for hybridization according to the manufacturer’s instruction.'\n",
      " \"Biotinylated cRNA probes were synthesized using the GeneChip Whole Transcript Sense Target Labeling Assay Kit (Affymetrix) following the manufacturer's instructions.\"\n",
      " 'Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 6 ug total RNA (Expression Analysis Technical Manual, 2001, Affymetrix).'\n",
      " \"Cyanine-3 (Cy3) labeled cRNA was prepared from 0.2 ug RNA using the One-Color Quick Amp Labeling kit (Agilent) according to the manufacturer's instructions, followed by RNAeasy column purification (QIAGEN). Dye incorporation and cRNA yield were checked with a NanoDrop ND-1000 Spectrophotometer.\"\n",
      " 'Johns Hopkins Microarray Core Facility'\n",
      " 'DNA was synthesized with the One-cycle cDNA Synthesis kit (Affymetrix). The cDNA was the used as a template for the synthesis of biotin labeled cRNA with IVT MEGAscript® T7 Kit.'\n",
      " 'The labeling was done according to manufacturer´s instructions and using the protocol One-Color Microarray-Based Gene Expression Analysis -protocol version 5.5, February 2007.'\n",
      " '1µg per pool was labeled by fluorescent dye Cy3 and Cy5 using dual-color QuickAmp kit (Agilent) according to the instructions matching a balanced dye-swap design.'\n",
      " 'The purified cDNA (5ug per sample) was fragmented and biotinylated using FL-Ovation cDNA Biotin Module V2 (Nugen, cat# 4200-12)'\n",
      " 'Standard Affymetrix protocol'\n",
      " \"100ng RNA was amplified and labeled using the MessageAmp™ Premier RNA Amplification Kit (Ambion, AM1792) according to the manufacturer's instructions..\"\n",
      " 'Biotinylated cRNA were prepared using the GeneChip IVT labelling kit (Affymetrix).'\n",
      " \"For microarray hybridization, RNA was processed for hybridization to Affymetrix Human Gene 1.0 ST Array through a core facility in the Functional Genomics Laboratory in the Center for Ecogenetics and Environmental Health at the University of Washington according to the manufacturer's instructions.\"\n",
      " \"Synthesis, labeling and fragmentation of cRNA was performed using the GeneChip® 3’ IVT Express Kit (Affymetrix) according to manufacurer's protocols.\"\n",
      " 'The testing RNA samples and universal mouse reference RNA (Agilent) were labeled with CTP-cy5 and CTP-cy3, respectively, Using Agilent Quick Amp Labeling Kit.'\n",
      " 'Agilent low RNA Input Linear Amplification Kit using Cyanine-3-CTP'\n",
      " 'Done by the SCIBLU Micoarray Resourse Center (MARC), Lund University, Sweden'\n",
      " 'Hybridization targets were prepared with MessageAmp™ Premier RNA Amplification Kit (Applied Biosystems/Ambion, Austin, TX) from total RNA.'\n",
      " 'Biotinylated cRNA was prepared according to the standard Affymetrix protocol from 50 ng total RNA (GeneChip Expression Analysis Technical Manual, 2005-2006, Affymetrix).'\n",
      " 'Refer to the Illumina System Manual'\n",
      " 'RNA was amplified, labeled with Cy5 and combined with spike B using the Agilent low RNA input fluorescence linear amplification kit (Agilent Technologies). The amplified cRNA was purified using the RNeasy kit (Qiagen), and specific activity was determined with the NanoDrop 1000 spectrophotometer (Thermo Scientific).'\n",
      " 'Samples were labelled with Encore biotin module (NUGEN) washed and stained using the HWS Kit (Affymetrix).'\n",
      " 'Biotinylated cRNA samples were prepared using the Affymetrix GeneChip WT Pico Reagent Kit from 2 ng total RNA.'\n",
      " \"Biotinylated cRNA were prepared according to the standard Affymetrix protocol(GeneChip® 3' IVT PLUS Reagent Kit, Affymetrix).\"\n",
      " 'Affymetrix two-cycle labelling kit'\n",
      " 'Biotin-labeled cRNA samples were prepared for hybridization using TotalPrep RNA Amplification kit (Ambion Inc., Austin, TX) following the sample labeling procedure: Total RNA (500 ng) was used for cDNA synthesis. Amplification/labeling step (in vitro transcription) was then performed to synthesize biotin-labeled cRNA. Concentration of cRNA was then measured by spectrophotometer (Nanodrop, ND-1000)'\n",
      " 'Total RNA was amplified using the Ovation PicoSL WTA System V2 and further processed using the Encore Biotin Module according to the instructions supplied by the manufacturer (NuGen).'\n",
      " 'Whole Transcript (WT) Sense Target Labeling (Affymetrix)'\n",
      " 'Total RNA (200 ng) was amplified and labeled with biotin using Affymetrix GeneChip Whole Transcript Sense Target Labeling Kit.'\n",
      " '100ng of total RNA was converted to first-strand cDNA using reverse transcriptase primed by a poly(T) oligomer that incorporated the T7 RNA polymerase promoter. Second-strand cDNA synthesis was followed by in vitro transcription (Affymetrix One-Cycle Target Labeling Kit) for linear amplification and biotinylation of each transcript.'\n",
      " '200ng total RNA, Affymetrix protocol'\n",
      " 'Fluorescence-labeled RNA targets were prepared from 5 μg of total RNA using MessageAmpTM aRNA kit (Ambion, Austin, TX, USA) and cyanine (Cy5) dye (Amersham Pharmacia, Piscataway, NJ, USA).'\n",
      " 'PICR Std Label v 2-0 - utalizing the Enzo BioArray High Yield Transcript Labelling Kit'\n",
      " 'Microarray experiments (GEO Accession numbers pending paper acceptance) were performed by the Bionomics Research and Technology Center at EOHSI (UMDNJ, Piscataway, NJ). Forty nanograms of total RNA from each sample were used to generate a high fidelity cDNA for array hybridization.  The cDNA were amplified, fragmented and labeled with biotin using NuGEN Ovation Amplification and Labeling kit V2 (NuGEN, San Diago, CA).'\n",
      " 'Biotinylated cRNA were prepared with the Illumina TotalPrep RNA Amplification kit (Ambion)'\n",
      " 'Total RNA (15ng) was reverse transcripted into the first-strand cDNA using a T7-Oligo (dT) Primer (Two-Cycle Target Labeling and Control Reagent package, Affymetrix, CA, USA). After second-strand cDNA synthesis, the double-stranded cDNA was purified and served as a template in the first cycle of in vitro transcription (IVT) reaction. The unlabeled cRNA was then reverse transcripted into the first-strand cDNA of the second cycle using random primers. Subsequently, the T7-Oligo(dT) Promoter Primer was used in the second-strand cDNA synthesis to generate double-stranded cDNA template containing T7 promoter sequences. Then the double-stranded cDNA was amplified and labeled using a biotinylated nucleotide analog/ribonucleotide mix in the second IVT reaction.'\n",
      " 'Target RNA is reverse transcribed into cDNA and in-vitro transcription is performed to generate biotin-labeled cRNA for subsequent hybridization.'\n",
      " 'Affymetrix GeneChip® WT Sense Target Labeling'\n",
      " '200 nanograms of total RNA per sample was used in conjunction with the Affymetrix recommended protocol for the GeneChip Human 2.0 ST Array (Affymetrix, Inc).'\n",
      " 'Performed by Milteny Biotech following standard procedures'\n",
      " \"Total RNA (100 ng) was labeled using a Low Input Quick Amp Labeling Kit One-Color (Agilent Technologies) according to manufacturer's instruction.\"\n",
      " 'Synthesis of Cy3-labeled cRNA was performed with the ‘Quick Amp Labeling kit, one color’ (#5190-0442, Agilent Technologies) according to the manufacturer’s recommendations.'\n",
      " 'First-strand cDNA synthesis employing a chimeric primer including the T7 promoter sequences and a stretch of 6 random bases. Second-strand cDNA synthesis and in vitro transcription using the T7 RNA polymerase in the presence of a labeled ribonucleotide (Cy3- or Cy5-CTP).'\n",
      " 'This information is not available since the experiments were done more than 10 years ago.'\n",
      " 'Gene expression profiling was assessed using Affymetrix GeneChip microarray assays according to the manufacturer’s protocol (Affymetrix). Briefly, 1 ug of total RNA was reverse transcribed (T-7 dT primers), amplified and biotinylated (Affymetrix IVT Labeling Kit).'\n",
      " 'Affymetrix small sample labeling protocol'\n",
      " 'Cybrids containing mitochondria derived from MCF10A, MDA-MB-468 and 143B cells with 143B p0 cells as nuclear background are named as 10A/143B, 468/143B and 143B/143B cybrids, respectively.'\n",
      " 'Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 100 ng total RNA  amplified with TargetAmp 1-round Aminoallyl-aRNA Amplication Kit 101 (Epicentre, WI) (Expression Analysis Technical Manual, 2001, Affymetrix).'\n",
      " \"Cyanine-3 (Cy3) labeled cRNA was prepared from 0.5 ug RNA using the One-Color Low RNA Input Linear Amplification PLUS kit (Agilent) according to the manufacturer's instructions, followed by RNAeasy column purification (QIAGEN, Valencia, CA).  Dye incorporation and cRNA yield were checked with the NanoDrop ND-1000 Spectrophotometer.\"\n",
      " 'First and second strand cDNA were synthesized from 15 ug of total RNA using the SuperScript Double-Stranded cDNA Synthesis Kit (Invitrogen) and oligo-dT24-T7 (5’-GGC CAG TGA ATT GTA ATA CGA CTC ACT ATA GGG AGG CGG-3’) primer (PrOligo) according to the manufacturer’s instructions. '\n",
      " 'Biotinylated cRNA were prepared according to the standard Affymetrix protocol from at least 3 ug total RNA (Expression Analysis Technical Manual, 2001, Affymetrix).'\n",
      " 'As per Affimetrix specifications'\n",
      " 'The double stranded cDNA was used as a template to generate biotinylated cRNA using Bioarray HighYield RNA Transcription Labeling Kit (Enzo) and the labeled cRNA purified by GeneChip Sample Cleanup Module (Affymetrix)'\n",
      " 'Eight micrograms of total RNA from bronchial epithelial cells were converted into double-stranded cDNA with the SuperScript II reverse transcriptase (Invitrogen) with an oligo-dT primer containing a T7 RNA polymerase promoter (Genset, Boulder, CO). The ENZO Bioarray RNA transcript labeling kit (Affymetrix) was used for in vitro transcription of the purified double-stranded cDNA.'\n",
      " '500ng of Poly A+RNA were labelled, using Cyanine 3-CTP  (Perkin Elmer)  with the Agilent Low Imput Linear Amplification Kit and purified with Qiagen’s RNeasy mini spin columns'\n",
      " 'Total RNA (100 ng) was used for amplification and in vitro transcription using the Genechip 3’ IVT Express Kit following the manufacturer’s instructions (Affymetrix). The amplified RNA was purified with magnetic beads and 15 μg Biotin-aRNA was treated with fragmentation reagent.'\n",
      " 'Biotin labeled complementary RNA (cRNA) was made using the Illumina TotalPrep RNA Amplification Kit'\n",
      " 'Biotinylated cRNA were prepared according to the standard Affymetrix protocol from 500 ng total RNA.'\n",
      " \"Affymetrix's GeneChip WT PLUS Reagent Kit\" 'Ambion WT labeling kit'\n",
      " 'Illumina HumanWG-6 v3.0 arrays containing >48,000 probes were handled according to standard protocols established by the manufacturer. Briefly, 200ng of total RNA from each sample were used to generate biotin-labeled cRNA probes using an Illumina MessageAmp cRNA labeling kit protocol (Ambion).'\n",
      " 'The cDNA derived from the amplified RNA samples from pre-vaccination time points were labeled with Cy3, and samples from the 3-week post-vaccination time point were labeled with Cy5 (Agilent Technologies).'\n",
      " '2µg of total RNA was amplified and labeled using the GeneChip® one-cycle target labeling and control reagents (Affymetrix)']\n"
     ]
    }
   ],
   "source": [
    "train['label_protocol_ch1'].fillna('nan',inplace = True)\n",
    "newser = train['label_protocol_ch1'].map(lambda x : 1 if (\"nan\" in x.lower()) else 0) \n",
    "print (train[newser == 1]['ctrl'].sum()) \n",
    "print (newser.sum())\n",
    "print(train['label_protocol_ch1'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Cell were infected with the indicated doxycycline-inducible constructs and treated with Dox or EtOH as solvent control. HMLE cells were additionally sorted according to CD44 & Cd24 surface markers.',\n",
       "       'Freshly drawn nasal fibroblasts were treated with lipopolysaccharide (10 μg/mL).   The cultures were incubated for 12 hours at 37 ºC in a humidified incubator with 5% CO2.',\n",
       "       'Cells were infected  with either lntA-V5+ or DlntA Listeria strains',\n",
       "       nan,\n",
       "       'Cells HELF- 977 and HAF-1608 synchronized in G0 phase were infected with HCMV at multiplicity of infection of 3 or mock-infected. Viral inoculum was added to the cells and allowed to adsorb for 60 min at 37°C in 5% CO2. Then virus inoculum was removed and replaced with fresh medium.',\n",
       "       'The samples were put immediately in RNALater (Qiagen).',\n",
       "       'RNA was extracted from non-confluent cultures of MCF10A stable cell lines. No treatment was given to the cells.',\n",
       "       'Fibroblasts were grown in 10% FBS DMEM in a 37º incubator with 7% CO2.',\n",
       "       'To knockdown Grhl2 gene expression, a replication incompetent, VSV-G pseudotyped lentiviral vector that contained a GFP tracking reporter gene along with either a non-targeting control shRNA (RHS4346) or the shRNA targeting Grhl2 (V2LMM_20130) was employed (Open Biosystems). Lentiviruses were packaged and concentrated as described previously (19). MLE-15 cell infections with lentiviral vectors were performed for 16 h in the presence of polybrene (5 μg/ml) at a multiplicity of infection of 50. Stable cell lines were generated by puromycin dihydrochloride (5 μg/ml, Sigma) selection for 6 days.',\n",
       "       'siRNA knockdown or lentiviral overexpression',\n",
       "       'Primary human hepatocytes were transduced with SINLV.ET.trBRAF, SINLV.PGK.GFP  at MOI 10 or mock-treated (controls).',\n",
       "       'Mice underwent the bilateral Spared Nerve Injury surgical procedure. The mouse was returned to its home-cage for recovery. Sham animals were used as controls, where the surgical procedure was carried out identically but all three branches of the sciatic nerve were left untouched and unharmed.',\n",
       "       'QHD group were fed with HFD for 4 weeks followed by daily gavage of 0.1ml/kg QHD in addition to HFD for additional 4 weeks; rats in the GC group were fed with HFD for 4 weeks followed by daily gavage of a mixture of geniposide and chlorogenic acid (weight ratio) in addition to HFD for additional 4 weeks.  All rats were sacrificed for tissue collection at the end of 4-week therapy.',\n",
       "       'On day of experiment monolayers were transferred to Hanks Buffered Saline, with 1uM fMLP on the apical surface (bottom chamber), to establish chemotactic gradient. Neutrophils were freshly isolated from healthy volunteer venous blood and used for either direct or indirect migration.',\n",
       "       'Mice were sacrificed at 6 weeks, colonic mucosa was isolated with PBS injection into submucosa.',\n",
       "       'Bovine serum albumin (BSA) was obtained from Sigma Aldrich (Albumin, bovine fraction V powder, A8806-1G Sigma Aldrich) and contained endotoxin at ≤ 0.1ng/mg. Lactalbumin was also obtained from Sigma Aldrich (Alpha-lactalbumin from bovine milk, 61289-50MG Sigma Aldrich), however information on endotoxin level in lactalbumin was not available. Pepsin- and trypsin- digested BSA (PT-BSA), pepsin- and trypsin- digested lactalbumin (PT-L), a blank pepsin- and trypsin- control (PT), where no substrate was used for digestion, and PT-G were prepared as described earlier (Lindfors,K. 2008). An endotoxin test was performed on PT-G and PT-BSA using a Limulus Amebocyte Lysate (LAL), Pyrogent plus single test kit (cat no. N289-06, CAMBREX). No endotoxin was found in PT-G, but PT-BSA tested positive for endotoxin.  Lyophilised PT, PT-BSA, PT-L and PT-G were dissolved in MEM+1% FCS and filter sterilised by passing through 0.22µm pore filter (SLGS033SS, Millipore). Similarly the synthetic peptides p31-43 (LGQQQPFPPQQPY) and the immunogenic peptide p57-68 (QLQPFPQPQLPY) (New England Peptide LLC, USA) were dissolved in MEM+1% FCS to achieve a final concentration of 150µg/ml. PT, PT-BSA and PT-L were used as negative controls for PT-G treatment. The cells were stimulated for six hours with a particular stimulant dissolved in 2ml MEM+1% FCS. The control cells (MED-CTL) were kept in MEM+1% FCS for six hours. In total, four identical but independent experiments were performed with cells differing in 1-2 passages between each experiment.',\n",
       "       'For photon irradiation (X-ray) a 6 MV X-ray beam generated by a clinical linear accelerator (dose rate of 3 Gy/min) was used. Carbon-ion irradiations (12C) using the intensity controlled raster scanning system (25) were performed with a homogeneous 10 mm (for cells) or 40 mm (for tumor) spread-out Bragg peak (SOBP) with an energy of 120.45-135.16 MeV/u (cells) and 122.36-183.74 MeV/u (tumor) on target obtained from the SIS accelerator (GSI Helmholtzzentrum für Schwerionenforschung). Mean dose average of LET was 50 70 keV/µm.',\n",
       "       'To make a total lung cell suspension, lungs were infused with dispase, then 1% agarose and lungs removed from animal; lungs were then minced, further digested with collagenase/dispase, treated with dnase, filtered, and red blood cells lysed.  Cells were then labeled with 7aad, anti-CD31 and anti-CD45 antibodies, and FACS-purified to exclude endothelial, hematopoietic, and dead cells (See Kim et al, 2005).',\n",
       "       'Secondary biliary cirrhosis was induced by common bile duct ligation (CBDL). Briefly, under general anesthesia with midazolam (5mg/kg body weight, i.p., Laboratory Reig Jofre, Barcelona, Spain) and ketamine (100mg/kg body weight, i.p., Imalgene, Barcelona, Spain) the common bile duct was isolated and double ligated with 5-0 silk thread. The first ligature was made below the junction of the hepatic ducts and the second one above the pancreatic ducts. Finally, the portion between the two ligatures was resected. Rats received weekly injections of vitamin K and were studied 4 weeks after the surgery. In the drug treated groups simvastatin was suspended in distilled water and administered by gavage (5mg/kg body weight) once a day from 3 days before the final study. Simvastatin inhibits the HMG-CoA activity in the liver as early as 30 min after oral administration, with a peak of inhibitory activity at one hour. Experiments were performed 1 h after the last dose of the drug.',\n",
       "       '14 days before Pneumocystis infection, the group of dexamethasone-treated PCP mice were given 4mg/l dexamethasone and 1 mg/ml tetracycline in drinking water consecutively until euthanasia. Both dexamethasone-treated and wild-type PCP mice were inoculated intratracheally with 1 × 10^6 Pneumocystis nuclei and control mice were given lung homogenate from uninfected scid mice. 14 days after infection, mice were sacrified and the left lung was excised.',\n",
       "       'Human hepatoma Huh7.5.1 cells were treated with silymarin or DMSO solvent for varying times (4hr, 8hr, 24hr).',\n",
       "       'no additional treatments',\n",
       "       'Sample preparation comprised perfusion of the lung with about 10 ml of sterile PBS and subsequent lysis of the tissue in RLT lysis-buffer (Qiagen) and a tissue homogenizer.',\n",
       "       'sgRNA was induced by Doxycycline for 48H',\n",
       "       'Mice were fed AIN-93G Standard Diet, Ad Lib',\n",
       "       'Adult mice were placed in individual compartments within a chamber and exposed to 500 ppm SO2 in air for 4 hr.',\n",
       "       '129S1/SvImJ;C57Bl/6J (129/B6) mice were fed with high-fat diet (Western Diet) and high fructose-glucose solution (Sugar Water) (WD SW) or chow diet (CD) for 52 weeks, and total RNA samples were isolated from liver and tumor tissues for genome-wide expression profiling.',\n",
       "       'Cells were treated with mock (0.1% ethanol), 10 nM calcitriol or calcipotriol for 12 hours.',\n",
       "       'U87 cells in 6-well plates were infected with lentiviral particles containing either shCON or shPHF20. The virus containing culture medium was replaced with fresh DMEM supplemented with 10% FBS at 12h after infection. 72h post lentivirus infection, two U87 cell groups were set up: KD (knockdown, cells infected with PHF20-shRNA lentivirus) and NC (negative control, cells infected with negative control-shRNA lentivirus).',\n",
       "       '24h after seeding cells were transfected with miR-372-3p mimic or non-targeting siRNA control using RNAiMAX. 30h post transfection cells were harvested and total RNA was extracted.',\n",
       "       'Eight week-old FXR floxed male mice were injected via the tail vein with 100 µl of a mixture of purified AAV serotype 8 (AAV8) containing a liver-specific TBG promoter driving Cre recombinase and either AAV8-TBG-FXR WT or AAV8-TBG-Y67F-FXR at 5 x1010 virus particles per mouse, and four weeks later, mice were fed 0.5% CA chow for 6 h to activate FXR signaling',\n",
       "       \"For VSV-M2 infection mice were injected intravenously with 2 x 106 pfu VSV-M2 or mock infected with PBS. 7 μL of in vivo jetPEI was mixed with 50 μg of nucleic acids (MWG-BIOTECH AG) at a N:P ratio of 7:1 in a volume of 200 μL of 5 % glucose solution (vol/vol), and incubated for 15 min. Subsequently, 50 μg of complexed 5'-triphosphate RNA, the synthetic dsRNA analog poly(I:C) (Invivogen) was injected.\",\n",
       "       'No treatment', 'PBS', 'aqueous extract of MC seeds',\n",
       "       'samples 1 - 18: Mice treated with 3x 80mg/kg Beta-Napthoflavone in one day and culled and dissected at the desiganted day post induction., samples 20-24 collected from untreated mice.',\n",
       "       'The 10,791 base pair COP rat Fry cDNA (GenBank accession number EU563851.1) was cloned in four sections and then reconstructed in the expression vector which was inserted into the pEGFP-N1 plasmid. The cells were transfected and subject to puromycin selection. A stable clone was selected which expressed Fry at levels comparable to the non-transformed normal mammary epithelial cell line, MCF10A. We termed the selected clone, 231wCFry.',\n",
       "       'In order to establish a defined baseline condition prior to dietary zinc depletion, subjects consumed meals composed of a mixed diet providing ~11 mg Zn/d for 7 d. An egg white-based liquid formula which provided <0.5 mg Zn/d was consumed during the following 10 d for acute dietary zinc depletion.',\n",
       "       'Injection of ketamine/xylazine (300mg/kg; 40mg/kg) before sacrifice.',\n",
       "       'To determine gene expression alterations induced by Nkx2-1 knockdown in TnonMet cells  Array analyses were performed on 368T1-control (bc37), 368T1-shNkx2-1 (E1), 394T4-control (bc37), and 394T4-shNkx2-1(E1) cells. The 394T4 samples were run in duplicate using either the Affymetrix GeneChip® WT Sense Target Labeling (A9242, A9244, A9245, A9247) or the Nugene Ovation® Systems Target Preparation kits (A9248 and A9249).  Thus, a pairwise comparison between the three control and their corresponding shNkx2-1 datasets was used to determine the potential Nkx2-1 regulated genes',\n",
       "       'H1299 cells were infected with adenoviruses expressing p53, p63g or p73b at 25 moi. HCT116 cells were transfected with siRNAs targeting AKR1B10 or negative control. 4h after the transfection, these cells were treated with adriamycin (0.5 ug/ml). 24h after the treatment, siRNAs were transfected into cells again.',\n",
       "       'Cells were rinsed with cold PBS and and placed in the Trizol solution.',\n",
       "       'HCAEC were incubated at 37 °C for 16-24 h with SUV (final PLPC concentration 3 mmol/L), lipid-free apoA-I (final concentration 1 mg /ml), rHDL (final apoA-I concentration 1 mg/ml), native HDL (final total protein concentration 1 mg/ml) or PBS.',\n",
       "       'According to Affymetix protocol.', 'none',\n",
       "       'Pregnant rats were injected with 100mg/kg/day of vinclozolin, or DMSO vehicle only, daily on days E8-E14 of pregnancy to expose pups in utero during this time. Animals were euthanized at E13, E14, or E16 of pregnancy, and embryonic gonads were collected for RNA isolation and hystology. Sex was determined by PCR using primers specific for Sry on genomic DNA isolated from embryo tails as previously described [Levine E, et al.,2000]. Therefore, in addition to the histological presence of testis cords in E14 and E16 the sex was confirmed with SRY PCR, in particular for the E13 testis.',\n",
       "       'Rat gonads from E13 embryos were dissected with mesonephros intact and cultured for 3 days with Vinclozolin. Briefly, gonads were placed in drops of medium on Millicell CM filters (Millipore, Bedford, MA, USA) floating on 0.4 ml of CMRL 1066 medium (Gibco BRL) supplemented with penicillin–streptomycin, insulin (10 µg/ml), L-glutamine (350 µM), transferrin (10 µg/ml), and BSA (0.01%). Gonadal pairs were split with one gonad treated with 50µM vinclozolin in DMSO and the contra-lateral gonad with DMSO only as a vehicle control. Media and treatments were changed daily and given fresh vinclozolin. Gonads were maintained in culture for 3 days at which time testis cords formed and testes were used for histological analysis or separated from mesonephros and used in RNA collections. Sex was determined by PCR using primers specific for Sry on genomic DNA isolated from embryo tails as previously described [Levine E, et al.,2000].',\n",
       "       'Rat gonads from E13 embryos were dissected with mesonephros intact and cultured for 3 days with flutamide. Briefly, gonads were placed in drops of medium on Millicell CM filters (Millipore, Bedford, MA, USA) floating on 0.4 ml of CMRL 1066 medium (Gibco BRL) supplemented with penicillin–streptomycin, insulin (10 µg/ml), L-glutamine (350 µM), transferrin (10 µg/ml), and BSA (0.01%). Gonadal pairs were split with one gonad treated with 0.5 µM flutamide in ethanol and the contra-lateral gonad with ethanol only as a vehicle control. Media and treatments were changed daily and given fresh flutamide. Gonads were maintained in culture for 3 days at which time testis cords formed and testes were used for histological analysis or separated from mesonephros and used in RNA collections. Sex was determined by PCR using primers specific for Sry on genomic DNA isolated from embryo tails as previously described [Levine E, et al.,2000].',\n",
       "       'Cells were activated with plate bound CD3 (OKT3) and CD46 (TRA2.10) for 36h in the presence of IL2',\n",
       "       'Virgin Ccnd1-/- and Ccnd1+/+ mice were overiectomized and allowed to recover for 2 after the surgery. Mice from each genotype split into 2 groups and treated with placebo control or estradiol releasing pelles (0.72mg, 60 day release). Mice were treated for 7 days and then sacrifised.',\n",
       "       'Buccal mucosa samples collected from cheek scrapings, stored in RNA later; Nasal epithelial samples collected by brushing inferior nasal turbinate, brushes stored in RNA later.',\n",
       "       'The Erlotinib-resistant NSCLC cell lines were derived by by exposing HCC827 and HCC4006 cell lines (parental) to erlotinib for 5–6 months as previously described (D. Presutti et al,  PLoS One. 2015 Nov 18;10(11):e0143333. doi: 10.1371/journal.pone.0143333. eCollection 2015.)',\n",
       "       'hSKP-HPC were exposed for 24 h to 27.7mM Na-VPA',\n",
       "       'Cells were left untreated, or were treated with either 10ng/ml of recombinant IFN-γ (Peprotech) or Universal Type I Interferon (PBL Assay Science) for 12 hrs',\n",
       "       'Parental cells were transfected using TransIT LT1 (Mirus Bio) according to the manufacturer’s instructions with plasmids containing targets of interest. Cells were clonogenically isolated via antibiotic selection and validated via qRT-PCR and western blotting.',\n",
       "       'Mice were infected intravenously via the retro-orbital sinus with 2x10^6 plaque forming unites of LCMV-Armstrong or LCMV-Clone 13.',\n",
       "       'H441 cells were seeded in 6-well tissue culture plates at a density of 1x106 cells per well. The cells were allowed to attach and grow overnight. CLEFMA was solubilized in dimethyl sulfoxide (DMSO), filtered through 0.2 µm nylon filter and added to the treated wells (n=4) at 1 µM concentration in culture medium supplemented with 5% FBS. The DMSO concentration was maintained at 0.1% per well. Control wells (n=4) received equivalent volume of DMSO without any test compound. The cells were allowed to remain in the treatment medium for 12 hours.',\n",
       "       'Mice received two times intraperitoneal injections 0.25 mg/ Kg-1 CDDP plus 50 mg/ Kg-1 (R)-crizotinib in 200 µL solvent (10 % Tween-80, 10 % PEG400, 4 % DMSO, 76 % physiological saline; v:v), or 200  µL solvent only as controls, at day 0 and day2.',\n",
       "       'no treatment',\n",
       "       'BAL samples were obtained from animals prior to vaccination (-2 weeks) and post-vaccination (weeks 3) previously described (Mehra S et al, AJRCCM, 2015).  For stabilization, 8 mL of 100% fetal bovine serum (Invitrogen, Life Sciences) was immediately added to 80 mL of the BAL sample collected. The sample was centrifuged at 400 rpm for 10 min at 4°C in an Allegra benchtop centrifuge. The pelleted cells were washed with cold RPMI media (Invitrogen, Life Sciences) and stored at -80°C until analysis.',\n",
       "       'None.',\n",
       "       'CD235a+, CD45+ and CD11b+ hematopoietic fractions were isolated from BM samples using MACS (Miltenyi).',\n",
       "       'ATF plasmids were designed and stably integrated into MCF7 cells as described in (Stolzenburg et al. 2012). Stable lines were grown at 30–80% confluency in Dulbecco’s Modified Eagle’s Medium (Corning, Corning, NY) supplemented with 10% heat-inactivated fetal bovine serum (Invitrogen, Life Technologies, Grand Island, NY) and 1% penicillin/streptomycin; cells were selected using 5 µg/ml puromycin (VWR, Radnor, PA) and 200 µg/ml G418 (VWR, Radnor, PA). ATF expression was induced by treatment with media containing 1µg/ml doxycycline (VWR, Radnor, PA) at 0 h, doxycycline media was refreshed at 48 h, and cells were harvested at 72 h. ATF expression was confirmed by hemagglutinin (HA) tag western blot prior to HA ChIP-seq, histone ChIP-seq and RNA-seq analysis.'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### how is treatment done\n",
    "train['treatment_protocol_ch1'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "182.0\n",
      "382\n",
      "['IMEC cell lines were grown in DMEM/F-12 with appropriate supplements. HMLE cells were kept in MEGM medium.'\n",
      " 'Nasal fibroblasts were isolated from surgical tissues and cultured.'\n",
      " 'LoVo cells grown in 6-well plates following ATCC  recommendations, to confluency. All cells were cultured at 37°C in a humidified atmosphere containing 10% CO2.'\n",
      " 'none'\n",
      " 'The cells were propagated in Dulbecco’s modified Eagle’s medium (DMEM, PanEco, Moscow, Russia) supplemented with 10% fetal bovine serum (FBS, PanEco), 2 mM L-glutamine and 50 ug/ml gentamicin at 37°C in 5% CO2. The cells were maintained by regular passages when confluence was reached. To obtain cells in G0 phase of cell cycle the cells of both lines were cultured in DMEM with 0.2% FBS for 48 hours.'\n",
      " 'A 3mm punch skin biopsy was obtained from the arm of study subjects.'\n",
      " 'Stable MCF10A cell lines were grown following the standard growth conditions (Debnath et al 2003).'\n",
      " 'Fibroblasts were isolated from mammary gland tissue using a modified protocol for gravity separation.'\n",
      " 'MLE-15 cells (provided by Dr. J. Whitsett, Children’s Hospital Medical Center, Cincinnati, OH) are murine lung epithelial cells produced from tumors in transgenic mice expressing the SV40 large tumor antigen under transcriptional control of the human surfactant protein C promoter (SP-C) (34). Cells were maintained in Dulbecco’s modified Eagle’s medium (DMEM) (Life Technologies, Inc.), 10% heat-inactivated fetal bovine serum (FBS) (Life Technologies, Inc.), 100 units/ml penicillin G, 100 mg/ml streptomycin sulfate.'\n",
      " 'isolated primary hepatocytes'\n",
      " 'C57BL/6J female mice, at 8\\u202fweeks of age'\n",
      " 'The rats were maintained in a temperature-controlled room (25 ± 1◦C) on a 12 h: 12 h light-dark cycle in the animal center.'\n",
      " 'T84 intestinal epithelia were cultured in DMEM/F12 with 10% FCS at 37oC with 5% CO2. Experimental cells were seeded and grown to confluence (TER>1,000 Ohms.cm2) on the underside of 3cm2 3um-pore Transwell permeable supports (Corning).'\n",
      " 'Mice were maintained at all times on a 12-hour light-dark cycle in a pathogen-free environment with free access to rodent chow and water.'\n",
      " 'Caco-2 cells (American Cell Type Collection, HTB-37, Rockville, MD, USA) were kindly provided by Prof. M. Mäki (University of Tampere, Tampere, Finland). Cells were maintained in minimum essential medium (MEM, 31095-029 GIBCO) supplemented with 1X sodium pyruvate (NaPy 100X, 11360-039 GIBCO), 1% Pen-Strep (Pen-Strep, 15070-063 GIBCO), 1X non-essential amino acids (NEAA 100X, 11140-035 GIBCO) and 10% fetal calf serum (FCS, C-37350 PROMOCELL) in 75cm2 flasks (353136 BD Biosciences) and were routinely split every 5 days. For stimulations, 3x105 cells were seeded in 25 cm2 flasks (156367, nunc) in 3ml MEM+1% FCS and incubated at 37°C with 5% CO2 in the air. On the third day of seeding, before stimulations the undifferentiated Caco-2 cells were washed once with 1X phosphate-buffered saline (PBS).'\n",
      " 'Human NSCLC adenocarcinoma cell lines A549 (ATCC-No.: CCL-185) were obtained from the American Type Culture Collection and maintained in RPMI 1640 (E15-840, PAA Laboratories GmbH) supplemented with 10% fetal bovine serum (Greiner BioOne), at 37 C in humidified atmosphere containing 5% CO2 in air. The cell lines were authenticated by microsatellite profiling (DSMZ, Germany).'\n",
      " 'Primary lung cells were purified from 4 month old female mice; no ex vivo growth of cells prior to RNA extraction'\n",
      " 'After euthanasia, the left lobe of mice lung was excised and then frozen in liquid nitrogen quickly. The tissuse was then transferrd to a -80℃ freezer.'\n",
      " 'Animal procedures were performed in accordance to regulations of the Italian Ministry of Health. PiZ transgenic mice were maintained on a C57/BL6 background.'\n",
      " 'Caco2 cells grown as recommended (MEM supplemented with 10% FBS, non-essential amino-acids, and sodium pyruvate). Cells were grown to confluency.'\n",
      " 'Cells were grown at 37C in 5%CO2 using DMEM media supplemented with 10%FBS + 1mM L-glutamine. Cells were passaged every 3 days and grown for several passages before RNA extraction.'\n",
      " 'Mantle cells were grown in RPMI1640 media plus 20%FBS,1%  penicillin, 1% streptomycin.'\n",
      " 'ES cells were cultured on irradiated MEFs in DMEM, 15% FBS media containing LIF (ESGRO) at 37oC with 5% CO2. For RNA-Seq experiments ES cells were cultured on gelatin-coated dishes in ES cell media containing 1.5 µM CHIR9901 (GSK3 inhibitor) for several passages to remove feeder cells. ES cells were passed by washing with PBS, and dissociating with trypsin. For embryoid body (EB) formation, ES cells were cultured in low attachment binding dishes to promote 3D formation in ES cell media without LIF.'\n",
      " 'The HC11 cells were grown at 37°C in a 5% CO2/95% air atmosphere in RPMI-1640 medium supplemented with 10% FCS, 1% PEST, 5 μg/ml insulin and 10ng/ml EGF.'\n",
      " 'Mice tracheas were cleaned well, cut into pieces and incubated in Dispase at RT for 23min. Digestion was stopped by addition of DMEM with 5% FBS. Epithelium was peeled off with forceps and incubated in 0.1% trypsin, 1.6mM EDTA 20 min at 37 °C.Mouse CD45 MicroBeads (Miltenyi Biotec) were used to deplete CD45+ leukocytes from mouse tracheal epithelial cells.'\n",
      " 'Huh-7.5 cells were grown in complete DMEM at a confluency of 70% prior to treatment.'\n",
      " 'DEN injection and High Fat Diet (HFD)'\n",
      " 'The cells were cultured in DMEM medium containing 10% FBS, 50U/mL penicillin and 50μg/mL streptomycin at 37℃ with 5% CO2 incubator. The cell line tested negative for any mycoplasma contamination.'\n",
      " 'Cells were cultured in RPMI-1640 AQmedium supplemented with 10% FBS and 1% Penicillin Streptomycin'\n",
      " 'Male mice only were used for the experiments. Mice were group housed up to five per cage with 12 h light/dark cycle with lights on at 6 a.m. Food and water were available ad libitum.'\n",
      " 'CD4+ cells were isolated by depleting the non-CD4+ cells (human CD4+CD25high isolation kit, Miltenyi Biotec). The purity of the isolated cells was determined by Flow cytometry analysis and was above 90%.'\n",
      " 'Primary cell isolation'\n",
      " 'Cell cultures were maintained at 37°C in a humidified atmosphere of 5% CO2 and 95% air. MDA-MB-231 and 231wCFry cells were cultured in Advanced DMEM with 10% heat inactivated FBS, 100 μg/ml penicillin-streptomycin solution.'\n",
      " 'Mice were maintained in a pathogen-free environnment.'\n",
      " 'Tumours were initiated by intratracheal infection of KrasLSL-G12D/+;p53flox/flox mice with a lentiviral vector expressing Cre-recombinase.  The MIT Institutional Animal Care and Use Committee approved all animal studies and procedures. Cell lines were created from individual lung tumours and metastases harvested from mice 8-14 months after tumour initiation. Tumours were cut into small pieces and then digested for 30 minutes at 37°C in 2ml of HBSS-free containing Trypsin, Collagenase IV, and Dispase in a 15ml conical tube. Following digestion, 4mL of Quench Solution (L15 media with 400μl of FBS and 15μl of 5mg/ml DNase) was added. Digested tumour samples were then pressed through 40μm cell strainers (BD). Finally, samples were centrifuged at 1,000 r.p.m. for 5 minutes, resuspended in culture media (DMEM, 10% FBS, pen/strep, glutamine), and plated in a 12-well plate. Cells were washed and culture media was changed every day for a week until stable cell lines were formed. In general we were able to derive cell lines from approximately 50% of tumours and metastases without a noticeable difference in the frequency of cell line generation from metastases from different site relative to the primary lung tumours.'\n",
      " 'NCRM-5 and iNC-012 hiPSCs were maintained as feeder-free on matrigel (BD)-coated 6-well plates and split every 3 days. The cells were split 1:6 or 1:12 using EDTA (0.5mM in Ca2+ and Mg2+ free PBS with 1.8mg/ml NaCl). Essential 8 (Invitrogen) medium was utilized for growth and maintenance of NCRM-5 hiPSC, and maintained in humidified 37°C incubator with 5% CO2 and 21% O239. mTESR (StemCell Technologies) media was used for iNC-012 cells.'\n",
      " \"H1299 and HCT116 cells were cultured in DMEM and McCoy's 5A medium with 10% FCS, respectively.\"\n",
      " 'Mouse intestinal tissue was incubated for 10 min at 37°C in 0.3 M EDTA. shaken vigorously and epithelial cells were separated from the underlying tissue and immune cells by repeated sedimentation at 1 x g at 4°C (Lotz et al., J. Exp Med. 2006). Flow cytometric analysis revealed less than 1% of CD45+ cells in IEC preparations.'\n",
      " 'TBE cells were grown and stimutaed with the cytokines indicated.'\n",
      " 'Human coronary artery endothelial cells (HCAEC) and human umbilical vein endothelial cells (HUVEC), passages 3 to 6, were grown in MesoEndo Growth Medium (Cell Applications, Inc. San Diego CA) containing 5% (v/v) serum and plated 24 hours before use in 12 well plates at a density of 1x105 cells per well.'\n",
      " 'For generating cybrids, motochondrial donor cells (MCF10A, MDA-MB-468 and 143B TK-) were enucleated by overnight actinomycin-D treatment and fused with the 143B p0 cells using 45% polyethylene glycol (MW 1450; Sigma cat P-5402) for 60 seconds. One day following fusion, cells were cultured in selection medium with bromodeoxyuridine (BrdU) but without uridine. Unfused control 143 p0 cells could not survive without uridine and mitochondria donor cells were killed by BrdU.'\n",
      " 'Rats in early estrus were bred overnight and mating confirmed by sperm positive smears, denoted as day 0 of pregnancy.'\n",
      " 'Cells were cultured in RPMI-1640 (containing 2mM L-glutamine + 0.1 g/l sodium bicarbonate) supplemented with 10mM Hepes, 100 U/ml Penicillin + 0.1 mg/ml Streptomycin and 10% fetal bovine serum'\n",
      " 'No specific growth protocol' 'Cells collected from human samples.'\n",
      " 'The NSCLC cell lines were grown in RPMI 1640 (BioWhittaker, Lonza, USA) supplemented with 10 mM Hepes pH 6.98-7.30, 1 mM L-glutamine, 100 U/ml penicillin/streptomycin (BioWittaker, Lonza) and heat inactivated 10% fetal bovine serum (FBS) (Sigma-Aldrich). All cells were cultured at 37°C in a 5% CO2 humidified incubator.'\n",
      " 'Approximately two weeks after isolation, hSKP spheroids were broken down to single cells using Liberase DH at 37 °C. They were subsequently cultivated in hSKP growth medium supplemented with 5 % (v/v) FBS to promote adherence of the cells to the plastic. The next day, the attached hSKP were washed and further cultivated in hSKP growth medium without FBS until further use.'\n",
      " 'Astrocytes were isolated from mixed glial cultures derived from the cerebral cortex of 1-3 day old neonatal mice, as previously described (Wilson E.H., et al., 2010, Journal of Clinical Investigation 120: 1368-1379). After 12 days of cultures, astrocytes (>90% GFAP+) were plated at a density of 1x10^5 cells/cm^2 in 12-well plates.'\n",
      " 'Mouse oviductal epithelium was grown in α-modified Eagle’s medium (Mediatech) supplemented with 10% v/v fetal bovine serum (Gibco), 2mM L-glutamine (Gibco), 2 μg/ml epithelial growth factor (Roche), 5 μg/ml insulin (Roche), 5 μg/ml transferrin (Roche), 5ng/ml sodium selenite (Roche), 1mg/ml gentamycin (Mediatech), 50U penicillin (Gibco), 50 μg/ml streptomycin (Gibco), 20ng/ml β-estradiol dissolved in ethanol (Sigma-Aldrich).'\n",
      " 'Six-ten week old mice were used for all experiments, and were housed under pathogen-free conditions.'\n",
      " \"Human lung adenocarcinoma cell line NCI-H441 (ATCC Number: HTB-174) was obtained from the American Type Culture Collection (Manassas, VA).  H441 cells were maintained at 37 °C with 5% CO2 in McCoy's 5A Medium (Invitrogen, Carlsbad, California) supplemented with 5% heat-inactivated fetal bovine serum (FBS).  All media contained gentamicin at 50 µg/ml (GIBCO Laboratories, Grand Island, NY).\"\n",
      " 'Subcutaneous MCA205 tumors were established by injecting 500,000 WT MCA205 cells (in 100 µL PBS) into the right flank of WT C57BL/6 mice, until tumors became palpable'\n",
      " 'Mice housed at the Weizmann Institute animal facility'\n",
      " 'Human BM cells obtained from iliac crest aspirates were amplified by cultivation in alpha-MEM with 10 % fetal calf serum (FCS), 1 ng/mL fibroblast growth factor 2 (FGF2), 20 µmol/L L-glutamine and 100 units/mL penicillin G'\n",
      " 'Thawing Cells:  1. Thaw vial immediately in 37º C water bath. Keep O ring above the water surface to prevent contamination. Thaw content with slight  shake until only small ice is left in vial. It usually takes 1 min. Spray vial with 70% ethanol all over and wipe its surface with clean tissue in the hood.  2. Open the vial and transfer the content to a 15 ml Falcon tube already containing 5 ml of fresh medium.  3. Spin down at 1000 rpm or 200g for 3 to 5 mins at 4º C. Aspirate supernatant.  4. Resuspend cells in fresh medium and transfer to 150mm x 25mm tissue culture dish.  5. Check the cells under microscope.  6. Cells are cultured in CO2 incubator and medium is changed about every 3 days.  7. It usually takes 3 days or more for cells to recovery from freezing. After cell culture reaches  80-85%  confluence, subculture is conducted. Subculture ratio is about 1:3 or 1:4.  Passaging Cells:  1. Observe cells to see how confluent they are, whether the cells are alive, whether the cells are contaminated, and whether the cells have the correct morphology. After cell culture reaches 80 to 85% confluence, subculture is conducted  2. Remove media from dish  3. Wash 1x with 10 ml of PBS  4. Add 5 ml of Trypsin and trypsinize for 3 min at 37C. Whack hard – you should see the cells coming down. (Important: never overtrypsinize the cells, so work quickly)  5. Add 5 ml of media and use it to rinse the dish to detach the cells off (4 to 5 times)  6. Spin down at 1000rpm for 3 to 5 min at room temperature. Aspirate supernatant  7. Add 15 ml of media to 15ml tube containing cell pellet, and pipette up and down to mix  8. Add 15 ml of media to each new 150mm x 25mm tissue culture dish (we split 1 dish into 3 to 4 dishes)  9. Add 5mls of cell culturecontaining media to each tissue culture dish. Make sure the media covers the entire area of the dish. Put the cells into 37C with 5% CO2.']\n"
     ]
    }
   ],
   "source": [
    "train['growth_protocol_ch1'].fillna('none',inplace =True) \n",
    "newser = train['growth_protocol_ch1'].map(lambda x : 1 if (len(x) < 40) else 0) \n",
    "print (train[newser == 1]['ctrl'].sum()) \n",
    "print (newser.sum())\n",
    "print(train['growth_protocol_ch1'].unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Model Training And Testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def combine_text_features(df,df1,columns2): \n",
    "    df['nf'] = df1[columns2].apply(lambda row: ' '.join([str(i) for i in row]) ,axis = 1)\n",
    "# trainc['nf'] = train[cols].apply(lambda row: ' '.join([str(i) for i in row]) ,axis = 1) \n",
    "# testc['nf']  = test[cols].apply(lambda row: ' '.join([str(i) for i in row]), axis = 1) \n",
    "# unlc['nf']  = unl[cols].apply(lambda row: ' '.join([str(i) for i in row]), axis = 1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "trainc['nf'] = train[cols].apply(lambda row: ' '.join([str(i) for i in row]) ,axis = 1) \n",
    "testc['nf']  = test[cols].apply(lambda row: ' '.join([str(i) for i in row]), axis = 1) \n",
    "unlc['nf']  = unl[cols].apply(lambda row: ' '.join([str(i) for i in row]), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def preprocess(data_df):\n",
    "    data_df['cleaned_feature'] = ''\n",
    "    \n",
    "    # Initializing Stopwords and Lemmatization objects\n",
    "    stop_words = set(stopwords.words('english'))\n",
    "    wordnet_lemm = WordNetLemmatizer()\n",
    "    \n",
    "    # Pattern to detect characters which are not alphabets or numbers so they can removed\n",
    "    alpha_or_numeric = \"[^a-zA-Z0-9- ]\"\n",
    "\n",
    "    for index, row in tqdm(data_df.iterrows(), total=data_df.shape[0]):\n",
    "    \n",
    "        sample = row['nf']\n",
    "        \n",
    "        # Replacing characters which are not alphabets or numbers with blank space and changing text to lowercase\n",
    "        # These two steps are for cleaning text data, you can add more on top of this to make your data cleaner.\n",
    "        #pre_txt = re.sub(alpha_or_numeric, \" \", sample)\n",
    "        pre_txt = sample.lower()\n",
    "            \n",
    "        \n",
    "        # Removing stop words and lemmatizing different words in preprocessed text and making the final processed text\n",
    "        sample_words = [wordnet_lemm.lemmatize(w) for w in pre_txt.split() if w not in stop_words and len(w)>1]\n",
    "        pre_proc_ver = ' '.join(sample_words)\n",
    "        \n",
    "        data_df.loc[index, 'cleaned_feature'] = pre_proc_ver\n",
    "        \n",
    "    return data_df\n",
    "def add_feature(df,df1,columns2,word_neg =neg_reduced , word_pos = pos_reduced ):  \n",
    "    sernew = df1[columns2[0]] \n",
    "    for j in range(1,len(columns2)): \n",
    "        sernew =sernew + df1[columns2[j]]\n",
    "    list1 = [] \n",
    "    for i in sernew.values: \n",
    "        flag = 0 \n",
    "        for j in word_neg: \n",
    "            if (j in i): \n",
    "                flag = 1  \n",
    "        if(flag == 1): \n",
    "            list1.append(-1) \n",
    "            continue\n",
    "        for j in word_pos: \n",
    "            if(j in i):\n",
    "                flag = 1 \n",
    "        if (flag == 1): \n",
    "            list1.append(1) \n",
    "            continue \n",
    "        list1.append(0) \n",
    "    df['nf2'] = pd.Series(list1)\n",
    "#clean \n",
    "def add_bools(df,df1,columns2): \n",
    "    words = pos_reduced[:20] + neg_reduced[:20]\n",
    "    sernew = df1[columns2[0]] \n",
    "    for j in range(1,len(columns2)): \n",
    "        sernew =sernew + df1[columns2[j]]\n",
    "    list1 = [] \n",
    "    sz = len(words) \n",
    "    for strings in sernew.values: \n",
    "        templist = [0]*sz \n",
    "        for i in range(sz): \n",
    "            if(words[i] in strings.lower()): \n",
    "                templist[i] = 1 \n",
    "        list1.append(templist) \n",
    "    fset = pd.DataFrame(list1,columns = list(map(lambda x : 'nfi'+str(x),range(sz)))) \n",
    "    df3 = df.copy() \n",
    "    df3 = pd.concat([df3,fset] , axis = 1) \n",
    "    return (fset,df3) \n",
    "\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_and_add_features(trainc,train,columns2): \n",
    "    #print(trainc['nf'].dtype)\n",
    "    cleaned_trainc = preprocess(trainc.copy())  \n",
    "    print(cleaned_trainc.columns) \n",
    "    \n",
    "    add_feature(cleaned_trainc,train,columns2)\n",
    "    print(cleaned_trainc.columns)\n",
    "    X1_train,X2 = add_bools(cleaned_trainc,train,columns2)\n",
    "    cleaned_trainc = X2.copy() \n",
    "    print(cleaned_trainc.columns)\n",
    "    return (cleaned_trainc,X1_train) \n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def tfidf_vectorization(cleaned_trainc,X1_train,vr = None,mode = 1): \n",
    "    stop_words = set(stopwords.words('english'))   \n",
    "    print(cleaned_trainc.values.shape , X1_train.shape)\n",
    "    if(mode == 0):\n",
    "        vect = TfidfVectorizer(analyzer=\"word\", preprocessor=None, stop_words=stop_words, max_features=10000,ngram_range = (1,1) ) \n",
    "        x_train_vect = vect.fit_transform(cleaned_trainc['cleaned_feature']).toarray() \n",
    "        print(x_train_vect.shape)\n",
    "        train_final_1 = np.c_[X1_train.values,x_train_vect] \n",
    "        train_final_2 = np.c_[train_final_1,cleaned_trainc['nf2'].values] \n",
    "        return (train_final_1,train_final_2,vect) \n",
    "    else: \n",
    "        x_train_vect = vr.transform(cleaned_trainc['cleaned_feature']).toarray() \n",
    "        train_final_1 = np.c_[X1_train.values,x_train_vect] \n",
    "        train_final_2 = np.c_[train_final_1,cleaned_trainc['nf2'].values] \n",
    "        return (train_final_1,train_final_2,None) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def extend_dataset(model,X_train,y,X_test): \n",
    "    \n",
    "    model.fit(X_train,y) \n",
    "    y_preds = model.predict(X_test) \n",
    "    print(type(y_preds)) \n",
    "    print(y_preds.dtype) \n",
    "    yn  = np.r_[y,y_preds] \n",
    "    train_final_n = np.r_[train_final_2,unl2] \n",
    "    return (train_final_n,yn) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_features(traindf,testdf,othercols): \n",
    "    temptrain = traindf[othercols].copy() \n",
    "    temptest = testdf[othercols].copy() \n",
    "    temptrain['new'] = temptrain['status'].map(lambda x: int(x[-4:])) \n",
    "    temptest['new'] = temptest['status'].map(lambda x: int(x[-4:])) \n",
    "    return (temptrain.drop(columns = ['status']).values,temptest.drop(columns = ['status']).values)\n",
    "    \n",
    "def Make_Train_Test(columns2 = ['title','characteristics_ch1','source_name_ch1','description'] , othercols = ['status'] , eds = 0):\n",
    "    train1 = pd.read_csv(\"labelled_train_data.csv\") \n",
    "    test1  = pd.read_csv(\"data_only_test.csv\")  \n",
    "      \n",
    "    \n",
    "    for i in range(len(columns2)): \n",
    "        train1[columns2[i]].fillna('NA' , inplace =True) \n",
    "        test1[columns2[i]].fillna('NA' , inplace =True) \n",
    "        ##unl1[columns2[i]].fillna('NA' , inplace = True)\n",
    "    trainc1 = train1.copy() \n",
    "    testc1 = test1.copy() \n",
    "   ## unlc1 = unl1.copy()\n",
    "    \n",
    "    combine_text_features(trainc1,train1,columns2) \n",
    "    combine_text_features(testc1,test1,columns2) \n",
    "    ##combine_text_features(unlc1,unl1,columns2) \n",
    "    print(trainc1['nf'])\n",
    "    ##print(trainc.columns) \n",
    "    train_list = [] \n",
    "    test_list = [] \n",
    "    words = pos_reduced + neg_reduced \n",
    "    for i in trainc1['nf'].values: \n",
    "        boolmask = [0]*len(words)\n",
    "        for ind in range(len(words)): \n",
    "            if(words[ind] in i.lower()): \n",
    "                boolmask[ind] = 1\n",
    "        train_list.append(boolmask) \n",
    "    \n",
    "    \n",
    "    for i in testc1['nf'].values: \n",
    "        boolmask = [0]*len(words)\n",
    "        for ind in range(len(words)): \n",
    "            if(words[ind] in i.lower()): \n",
    "                boolmask[ind] = 1\n",
    "        test_list.append(boolmask)\n",
    "    \n",
    "#     train_f = []\n",
    "#     for i in trainc1['nf'].values: \n",
    "#         flag = 0 \n",
    "#         for j in pos_reduced: \n",
    "#             if (j in i): \n",
    "#                 flag = 1\n",
    "#         if(flag == 1):\n",
    "#             train_f.append([1]) \n",
    "#             continue\n",
    "#         for j in neg_reduced: \n",
    "#             if (j in i): \n",
    "#                 flag = 1\n",
    "#         if(flag==1): \n",
    "#             train_f.append([-1]) \n",
    "#             continue \n",
    "#         train_f.append([0])\n",
    "        \n",
    "    \n",
    "#     test_f = []\n",
    "#     for i in testc1['nf'].values: \n",
    "#         flag = 0 \n",
    "#         for j in pos_reduced: \n",
    "#             if (j in i): \n",
    "#                 flag = 1\n",
    "#         if(flag == 1):\n",
    "#             test_f.append([1])\n",
    "#             continue\n",
    "#         for j in neg_reduced: \n",
    "#             if (j in i): \n",
    "#                 flag = 1\n",
    "#         if(flag==1): \n",
    "#             test_f.append([-1])\n",
    "#             continue \n",
    "#         test_f.append([0])\n",
    "        \n",
    "    \n",
    "    \n",
    "    \n",
    "    train_f , test_f = make_features(train1,test1,othercols) \n",
    "    \n",
    "    x_train = np.array(train_list) \n",
    "    x_train = np.c_[x_train,train_f]\n",
    "    x_test = np.array(test_list) \n",
    "    x_test = np.c_[x_test,test_f]\n",
    "    \n",
    "    cleaned_trainc1,x1_train1 = process_and_add_features(trainc1,train1,columns2) \n",
    "    cleaned_testc1,x1_test1   = process_and_add_features(testc1,test1,columns2) \n",
    "    if(eds == 1):\n",
    "        cleaned_unlc,x1_unl     = process_and_add_features(unlc,unl,columns2) \n",
    "    \n",
    "    train_final_11,train_final_21,vect1 = tfidf_vectorization(cleaned_trainc1,x1_train1,mode = 0)  \n",
    "    if(eds == 1):\n",
    "        unl_final_1,unl_final2,v1 = tfidf_vectorization(cleaned_unlc, x1_unl,mode=1,vr = vect) \n",
    "    test_final_11,test_final_21,v21  = tfidf_vectorization(cleaned_testc1,x1_test1,mode = 1,vr = vect1) \n",
    "    y1 = train1['ctrl'].astype(int).tolist()\n",
    "    \n",
    "    \n",
    " \n",
    "     \n",
    "        \n",
    "    \n",
    "    if(eds == 1): \n",
    "        final_train, final_y = extend_dataset(LogisticRegression(max_iter = 1000) , train_final_2, y , test_final_2) \n",
    "        return (final_train , final_y , test_final_2) \n",
    "    else:  \n",
    "        s1 = StandardScaler() \n",
    "        train_final_31  = s1.fit_transform(train_final_21)\n",
    "        test_final_31 = s1.transform(test_final_21) \n",
    "        s2 = StandardScaler() \n",
    "        x_trains = s2.fit_transform(x_train) \n",
    "        x_tests = s2.transform(x_test)\n",
    "        return (train_final_11,test_final_11,train_final_21,test_final_21,train_final_31,test_final_31,x_train,x_test,x_trains,x_tests,y1) \n",
    "    \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "def print_scores(train_final_1,train_final_2,train_final_3,x_train,x_trains,y):\n",
    "    def get_scores(models ,X,y): \n",
    "        names = [\"LogReg \" , \"SVC \", \"RandomForestClassfier \"]\n",
    "        for model,name in zip(models,names): \n",
    "            scores = cross_val_score(model,X,y,cv=5,scoring=\"f1\") \n",
    "            print(name , scores.mean() , scores.std()) \n",
    "    models = [LogisticRegression(max_iter=1000),SVC(kernel=\"rbf\") , RandomForestClassifier(random_state = 2)] \n",
    "    get_scores(models,train_final_1,y) \n",
    "    get_scores(models,train_final_2,y) \n",
    "    get_scores(models,train_final_3,y)\n",
    "    get_scores(models,x_train,y) \n",
    "    get_scores(models,x_trains,y)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0      rep1_cd44low_dox facs sorting: CD44low/CD24hig...\n",
      "1      rep1_imecs_myc_dox facs sorting: Unsorted IMEC...\n",
      "2      rep2_cd44low_etoh facs sorting: CD44low/CD24hi...\n",
      "3      rep1_cd44high_dox facs sorting: CD44high/CD24l...\n",
      "4      lipopolysaccharide treated sample replication ...\n",
      "                             ...                        \n",
      "618    RNA-seq_598-SKD_3dDox_rep3 cell line: MCF7 Hum...\n",
      "619    RNA-seq_598-SKD_0dDox_rep1 cell line: MCF7 Hum...\n",
      "620    RNA-seq_598-SKD_0dDox_rep3 cell line: MCF7 Hum...\n",
      "621    RNA-seq_598-SKD_3dDox_rep2 cell line: MCF7 Hum...\n",
      "622    RNA-seq_598-SKD_0dDox_rep2 cell line: MCF7 Hum...\n",
      "Name: nf, Length: 623, dtype: object\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████| 623/623 [00:00<00:00, 1750.08it/s]\n",
      "  2%|█▊                                                                           | 139/6070 [00:00<00:04, 1379.35it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'ctrl', 'pert',\n",
      "       'channel_count', 'characteristics_ch1', 'contact_address',\n",
      "       'contact_city', 'contact_country', 'contact_department',\n",
      "       'contact_email', 'contact_institute', 'contact_name', 'contact_state',\n",
      "       'data_processing', 'data_row_count', 'description',\n",
      "       'extract_protocol_ch1', 'growth_protocol_ch1', 'hyb_protocol',\n",
      "       'label_ch1', 'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
      "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
      "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
      "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
      "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
      "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
      "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
      "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
      "       'biomaterial_provider_ch2', 'growth_protocol_ch2', 'nf',\n",
      "       'cleaned_feature'],\n",
      "      dtype='object')\n",
      "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'ctrl', 'pert',\n",
      "       'channel_count', 'characteristics_ch1', 'contact_address',\n",
      "       'contact_city', 'contact_country', 'contact_department',\n",
      "       'contact_email', 'contact_institute', 'contact_name', 'contact_state',\n",
      "       'data_processing', 'data_row_count', 'description',\n",
      "       'extract_protocol_ch1', 'growth_protocol_ch1', 'hyb_protocol',\n",
      "       'label_ch1', 'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
      "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
      "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
      "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
      "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
      "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
      "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
      "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
      "       'biomaterial_provider_ch2', 'growth_protocol_ch2', 'nf',\n",
      "       'cleaned_feature', 'nf2'],\n",
      "      dtype='object')\n",
      "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'ctrl', 'pert',\n",
      "       'channel_count', 'characteristics_ch1', 'contact_address',\n",
      "       'contact_city', 'contact_country', 'contact_department',\n",
      "       'contact_email', 'contact_institute', 'contact_name', 'contact_state',\n",
      "       'data_processing', 'data_row_count', 'description',\n",
      "       'extract_protocol_ch1', 'growth_protocol_ch1', 'hyb_protocol',\n",
      "       'label_ch1', 'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
      "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
      "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
      "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
      "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
      "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
      "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
      "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
      "       'biomaterial_provider_ch2', 'growth_protocol_ch2', 'nf',\n",
      "       'cleaned_feature', 'nf2', 'nfi0', 'nfi1', 'nfi2', 'nfi3', 'nfi4',\n",
      "       'nfi5', 'nfi6', 'nfi7', 'nfi8', 'nfi9', 'nfi10', 'nfi11', 'nfi12',\n",
      "       'nfi13', 'nfi14', 'nfi15', 'nfi16', 'nfi17', 'nfi18', 'nfi19', 'nfi20',\n",
      "       'nfi21', 'nfi22', 'nfi23', 'nfi24', 'nfi25', 'nfi26', 'nfi27', 'nfi28',\n",
      "       'nfi29', 'nfi30', 'nfi31', 'nfi32', 'nfi33', 'nfi34', 'nfi35', 'nfi36',\n",
      "       'nfi37', 'nfi38', 'nfi39'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████| 6070/6070 [00:04<00:00, 1421.73it/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'channel_count',\n",
      "       'characteristics_ch1', 'contact_address', 'contact_city',\n",
      "       'contact_country', 'contact_department', 'contact_email',\n",
      "       'contact_institute', 'contact_name', 'contact_state', 'data_processing',\n",
      "       'data_row_count', 'description', 'extract_protocol_ch1',\n",
      "       'growth_protocol_ch1', 'hyb_protocol', 'label_ch1',\n",
      "       'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
      "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
      "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
      "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
      "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
      "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
      "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
      "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
      "       'biomaterial_provider_ch2', 'growth_protocol_ch2', 'nf',\n",
      "       'cleaned_feature'],\n",
      "      dtype='object')\n",
      "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'channel_count',\n",
      "       'characteristics_ch1', 'contact_address', 'contact_city',\n",
      "       'contact_country', 'contact_department', 'contact_email',\n",
      "       'contact_institute', 'contact_name', 'contact_state', 'data_processing',\n",
      "       'data_row_count', 'description', 'extract_protocol_ch1',\n",
      "       'growth_protocol_ch1', 'hyb_protocol', 'label_ch1',\n",
      "       'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
      "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
      "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
      "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
      "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
      "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
      "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
      "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
      "       'biomaterial_provider_ch2', 'growth_protocol_ch2', 'nf',\n",
      "       'cleaned_feature', 'nf2'],\n",
      "      dtype='object')\n",
      "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'channel_count',\n",
      "       'characteristics_ch1', 'contact_address', 'contact_city',\n",
      "       'contact_country', 'contact_department', 'contact_email',\n",
      "       'contact_institute', 'contact_name', 'contact_state', 'data_processing',\n",
      "       'data_row_count', 'description', 'extract_protocol_ch1',\n",
      "       'growth_protocol_ch1', 'hyb_protocol', 'label_ch1',\n",
      "       'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
      "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
      "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
      "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
      "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
      "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
      "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
      "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
      "       'biomaterial_provider_ch2', 'growth_protocol_ch2', 'nf',\n",
      "       'cleaned_feature', 'nf2', 'nfi0', 'nfi1', 'nfi2', 'nfi3', 'nfi4',\n",
      "       'nfi5', 'nfi6', 'nfi7', 'nfi8', 'nfi9', 'nfi10', 'nfi11', 'nfi12',\n",
      "       'nfi13', 'nfi14', 'nfi15', 'nfi16', 'nfi17', 'nfi18', 'nfi19', 'nfi20',\n",
      "       'nfi21', 'nfi22', 'nfi23', 'nfi24', 'nfi25', 'nfi26', 'nfi27', 'nfi28',\n",
      "       'nfi29', 'nfi30', 'nfi31', 'nfi32', 'nfi33', 'nfi34', 'nfi35', 'nfi36',\n",
      "       'nfi37', 'nfi38', 'nfi39'],\n",
      "      dtype='object')\n",
      "(623, 96) (623, 40)\n",
      "(623, 1294)\n",
      "(6070, 94) (6070, 40)\n",
      "(623, 1335)\n",
      "(623, 1334)\n",
      "LogReg  0.7372575700161907 0.11221605181601797\n",
      "SVC  0.7327104397667031 0.1246956325958747\n",
      "RandomForestClassfier  0.7274679559817108 0.13746423646730085\n",
      "LogReg  0.7879231282711313 0.09543622481403195\n",
      "SVC  0.7691857024129624 0.09964888513416598\n",
      "RandomForestClassfier  0.7341214884142895 0.1187771741182049\n",
      "LogReg  0.726730604288499 0.12064602250318958\n",
      "SVC  0.7304758610718991 0.1238341919909763\n",
      "RandomForestClassfier  0.7341214884142895 0.1187771741182049\n",
      "LogReg  0.3523137213737119 0.3299649581332286\n",
      "SVC  0.08376068376068377 0.11995080200238518\n",
      "RandomForestClassfier  0.7695182640597507 0.10009227720682588\n",
      "LogReg  0.7835472595316321 0.09265018043802269\n",
      "SVC  0.7770698385863868 0.10565213858792252\n",
      "RandomForestClassfier  0.7666089327100524 0.09915771311828983\n"
     ]
    }
   ],
   "source": [
    "train_final_1,test_final_1,train_final_2,test_final_2,train_final_3,test_final_3,x_train,x_test,x_trains,x_tests,y = Make_Train_Test() \n",
    "print(train_final_2.shape) \n",
    "print(train_final_1.shape)\n",
    "print_scores(train_final_1,train_final_2,train_final_3,x_train,x_trains,y) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "def stackemodel(listofmodels,train_final_2,y,finmodel): \n",
    "    fin_score = [] \n",
    "    for i in range(5):\n",
    "        X_train,X_test,y_train,y_test = train_test_split(train_final_2,y,test_size = 0.2)\n",
    "        preds = np.array([[1]]*(X_train.shape[0])) \n",
    "        pred_test = np.array([[1]]*X_test.shape[0])\n",
    "        print(preds.shape) \n",
    "        print(pred_test.shape)\n",
    "        for model in listofmodels: \n",
    "            model.fit(X_train,y_train) \n",
    "            p1 = model.predict(X_train)  \n",
    "            p2 = model.predict(X_test)\n",
    "\n",
    "            preds=np.c_[preds,p1]  \n",
    "            pred_test = np.c_[pred_test,p2] \n",
    "\n",
    "        preds = preds[:,1:] \n",
    "        print(preds.shape) \n",
    "        pred_test = pred_test[:,1:] \n",
    "        print(pred_test.shape)\n",
    "\n",
    "\n",
    "        finmodel.fit(preds,y_train) \n",
    "        yf = finmodel.predict(pred_test)\n",
    "\n",
    "\n",
    "        fin_score.append(f1_score(y_test,yf)) \n",
    "    return np.array(fin_score)\n",
    "\n",
    "    \n",
    "def predictstackemodel(listofmodels,train_final_2,y,test_final_2,finmodel): \n",
    "    \n",
    "    preds = np.array([[1]]*(train_final_2.shape[0])) \n",
    "    pred_test = np.array([[1]]*(test_final_2.shape[0]))\n",
    "    print(preds.shape) \n",
    "    print(pred_test.shape)\n",
    "    for model in listofmodels: \n",
    "        model.fit(train_final_2,y) \n",
    "        p1 = model.predict(train_final_2)  \n",
    "        p2 = model.predict(test_final_2)\n",
    "        \n",
    "        preds=np.c_[preds,p1]  \n",
    "        pred_test = np.c_[pred_test,p2] \n",
    "        \n",
    "    preds = preds[:,1:] \n",
    "    print(preds.shape) \n",
    "    pred_test = pred_test[:,1:] \n",
    "    print(pred_test.shape)\n",
    "    \n",
    "    \n",
    "    finmodel.fit(preds,y) \n",
    "    yf = finmodel.predict(pred_test)\n",
    "    \n",
    "    \n",
    "    check(yf) \n",
    "#     array = np.c_[test['geo_accession'].values,yf]\n",
    "#     df = pd.DataFrame(array,columns = ['geo_accession' , 'ctrl']) \n",
    "#     print(df.shape) \n",
    "#     df['ctrl'] = df['ctrl'].astype(np.float64)  \n",
    "#     name = 'stack2' + ' submission.csv'\n",
    "#     df.to_csv(name , index = False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(498, 1)\n",
      "(125, 1)\n",
      "(498, 4)\n",
      "(125, 4)\n",
      "(498, 1)\n",
      "(125, 1)\n",
      "(498, 4)\n",
      "(125, 4)\n",
      "(498, 1)\n",
      "(125, 1)\n",
      "(498, 4)\n",
      "(125, 4)\n",
      "(498, 1)\n",
      "(125, 1)\n",
      "(498, 4)\n",
      "(125, 4)\n",
      "(498, 1)\n",
      "(125, 1)\n",
      "(498, 4)\n",
      "(125, 4)\n",
      "0.8803038923194337 0.01939895462165149\n"
     ]
    }
   ],
   "source": [
    "scores =stackemodel([LogisticRegression(max_iter = 1000), SVC(kernel = \"rbf\"),GaussianNB(),RandomForestClassifier(random_state = 2)] , train_final_2,y,LogisticRegression(max_iter=1000)) \n",
    "print(scores.mean() , scores.std())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(20079, 814)\n",
      "(20079,)\n"
     ]
    }
   ],
   "source": [
    "print(train_final_2.shape) \n",
    "print(y.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(623, 1)\n",
      "(6070, 1)\n",
      "(623, 2)\n",
      "(6070, 2)\n",
      "1442\n",
      "4628\n"
     ]
    }
   ],
   "source": [
    "predictstackemodel([LogisticRegression(max_iter = 1000), SVC(kernel = \"rbf\")] , train_final_2,y,test_final_2,LogisticRegression(max_iter = 1000))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "def check(test_predictions): \n",
    "    df = pd.read_csv(\"logistic_sub2 submission.csv\") \n",
    "    ap = df['ctrl'].values\n",
    "    print((ap != test_predictions).sum())\n",
    "    print((ap == test_predictions).sum())\n",
    "def predict(model,X_train,y_train,X_test,name = \"logreg\"):\n",
    "    model.fit(X_train,y_train)\n",
    "    test_predictions = model.predict(X_test)   \n",
    "    check(test_predictions)\n",
    "    print(test['geo_accession'].values)\n",
    "    array = np.c_[test['geo_accession'].values,test_predictions]\n",
    "    df = pd.DataFrame(array,columns = ['geo_accession' , 'ctrl']) \n",
    "    print(df.shape) \n",
    "    df['ctrl'] = df['ctrl'].astype(np.float64)  \n",
    "    name = name + ' submission.csv'\n",
    "    df.to_csv(name , index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1766\n",
      "4304\n",
      "['GSM1973514' 'GSM1973517' 'GSM1973515' ... 'GSM590832' 'GSM590835'\n",
      " 'GSM590836']\n",
      "(6070, 2)\n"
     ]
    }
   ],
   "source": [
    "predict(LogisticRegression(max_iter = 1000),x_train,y,x_test,name = \"logreg-0.82\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "557\n",
      "5513\n",
      "(6070, 2)\n"
     ]
    }
   ],
   "source": [
    "predict(LogisticRegression(max_iter=1000),train_final_2,y,test_final_2,name= \"unlabelled_ds_557\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "552\n",
      "(6070, 2)\n",
      "(6070, 2)\n"
     ]
    }
   ],
   "source": [
    "m_svc = SVC(kernel=\"rbf\") \n",
    "m_lr = LogisticRegression(max_iter=1000) \n",
    "p11 = pd.read_csv(\"logistic_sub2 submission.csv\")['ctrl'].values \n",
    "m_svc.fit(train_final_2,y) \n",
    "p_svc = m_svc.predict(test_final_2) \n",
    "m_lr.fit(train_final_2,y) \n",
    "p_lr = m_lr.predict(test_final_2) \n",
    "list1 = []\n",
    "for i in range(6070): \n",
    "    num0 = 0 \n",
    "    if (p11[i] == 1.0):  \n",
    "        num0 = num0+1 \n",
    "    if(p_svc[i] == 1.0): \n",
    "        num0 = num0+1 \n",
    "    if(p_lr[i] == 1.0): \n",
    "        num0 = num0+1 \n",
    "    if(num0>1): \n",
    "        list1.append(1) \n",
    "    else: \n",
    "        list1.append(0) \n",
    "##print(list1) \n",
    "arr = np.array(list1) \n",
    "print((arr != p11).sum()) \n",
    "array = np.c_[test['geo_accession'].values,arr]\n",
    "df = pd.DataFrame(array,columns = ['geo_accession' , 'ctrl']) \n",
    "print(df.shape) \n",
    "df['ctrl'] = df['ctrl'].astype(np.float64)  \n",
    "name = \"majority_voter\" + ' submission.csv'\n",
    "df.to_csv(name , index = False) \n",
    "print(df.shape)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "filename = 'submission_xgb_wordembedding_word2vec_train_6000.csv' \n",
    "pretrained_df = pd.read_csv(filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6070, 2)\n",
      "Index(['geo_accession', 'ctrl'], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(pretrained_df.shape) \n",
    "print(pretrained_df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_additional = test_final.copy() \n",
    "train_total = np.r_[train_final,train_additional]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6693, 811)\n"
     ]
    }
   ],
   "source": [
    "print(train_total.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(6693,)\n"
     ]
    }
   ],
   "source": [
    "y1_train = train['ctrl'].astype(int).values\n",
    "y2_train = pretrained_df['ctrl'].astype(int).values\n",
    "y_total = np.r_[y1_train,y2_train] \n",
    "print(y_total.shape)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def clg(train_total,y_total):\n",
    "    model = LogisticRegression(max_iter = 1000) \n",
    "    model.fit(train_total,y_total) \n",
    "    preds = model.predict(train_final) \n",
    "    print(classification_report(y1_train,preds))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.90      0.99      0.95       319\n",
      "           1       0.99      0.89      0.94       304\n",
      "\n",
      "    accuracy                           0.94       623\n",
      "   macro avg       0.95      0.94      0.94       623\n",
      "weighted avg       0.95      0.94      0.94       623\n",
      "\n"
     ]
    }
   ],
   "source": [
    "clg(train_final,y1_train)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.79      0.98      0.87       319\n",
      "           1       0.98      0.72      0.83       304\n",
      "\n",
      "    accuracy                           0.86       623\n",
      "   macro avg       0.88      0.85      0.85       623\n",
      "weighted avg       0.88      0.86      0.85       623\n",
      "\n"
     ]
    }
   ],
   "source": [
    "clg(train_total,y_total)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "def augment_and_test(rfc): \n",
    "    rfc.fit(train_final,y1_train) \n",
    "    y_else = rfc.predict(test_final) \n",
    "    y_else = y_else.astype(int)\n",
    "    y_ref = np.r_[y1_train,y_else] \n",
    "    clg(train_total,y_ref) \n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.79      0.97      0.87       319\n",
      "           1       0.95      0.73      0.83       304\n",
      "\n",
      "    accuracy                           0.85       623\n",
      "   macro avg       0.87      0.85      0.85       623\n",
      "weighted avg       0.87      0.85      0.85       623\n",
      "\n"
     ]
    }
   ],
   "source": [
    "augment_and_test(RandomForestClassifier())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "              precision    recall  f1-score   support\n",
      "\n",
      "           0       0.79      0.98      0.88       319\n",
      "           1       0.98      0.73      0.83       304\n",
      "\n",
      "    accuracy                           0.86       623\n",
      "   macro avg       0.88      0.86      0.86       623\n",
      "weighted avg       0.88      0.86      0.86       623\n",
      "\n"
     ]
    }
   ],
   "source": [
    "augment_and_test(LogisticRegression(max_iter = 1000))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mean</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>taxid_ch1</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9544</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9555</th>\n",
       "      <td>0.800000</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9606</th>\n",
       "      <td>0.469003</td>\n",
       "      <td>371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10090</th>\n",
       "      <td>0.514423</td>\n",
       "      <td>208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10116</th>\n",
       "      <td>0.558824</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               mean  count\n",
       "taxid_ch1                 \n",
       "9544       0.000000      5\n",
       "9555       0.800000      5\n",
       "9606       0.469003    371\n",
       "10090      0.514423    208\n",
       "10116      0.558824     34"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train.groupby('taxid_ch1').agg(['mean','count'])['ctrl']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "528\n",
      "68\n",
      "['blood', 'control', 'airway', 'frozen', 'stage', '0', 'exposure', 'ethanol', 'mock', 'rep3cell', 'wild', 'diet', 'untreated', 'vector', 'healthy', 'negative', 'empty', 'uninfected', '9', 'control,', 'null', 'fat', 'diet,', 'vehicle', '0,', 'mice,', 'diploid', '3tissue:', 'without', 'wt', 'liverliver', 'condition,', 'reporter', 'rep3strain:', 'rep3tissue:', 'cellsprimary', 'rep4strain:', 'standard', 'transcriptome', 'ad', 'single', 'infection.', 'livernormal', 'cellsreplicate', 'protocol.', 'reads', '3gender:', 'typewild', 'weeksstrain:', 'rep4tissue:', 'deletion', 'incubation', '85', 'rep2,', '12strain:', '1diagnosis:', '3genotype:', '83', 'neutrophils', '2treatment:', '18strain:', '16cell', '23cell', 'rep5tissue:', '2diagnosis:', '3diagnosis:', 'rep3gender:']\n",
      "['blood', 'control', 'airway', 'frozen', 'stage', '0', 'exposure', 'ethanol', 'mock', 'rep3cell', 'wild', 'diet', 'untreated', 'vector', 'healthy', 'negative', 'empty', 'uninfected', '9', 'control,', 'null', 'fat', 'diet,', 'vehicle', '0,', 'mice,', 'diploid', '3tissue:', 'without', 'wt', 'liverliver', 'condition,', 'reporter', 'rep3strain:', 'rep3tissue:', 'cellsprimary', 'rep4strain:', 'standard', 'transcriptome', 'ad', 'single', 'infection.', 'livernormal', 'cellsreplicate', 'protocol.', 'reads', '3gender:', 'typewild', 'weeksstrain:', 'rep4tissue:', 'deletion', 'incubation', '85', 'rep2,', '12strain:', '1diagnosis:', '3genotype:', '83', 'neutrophils', '2treatment:', '18strain:', '16cell', '23cell', 'rep5tissue:', '2diagnosis:', '3diagnosis:', 'rep3gender:']\n",
      "['blood', 'control', 'airway', 'frozen', 'stage', 'exposure', 'ethanol', 'mock', 'wild', 'diet', 'untreated', 'vector', 'healthy', 'negative', 'empty', 'uninfected', 'fat', 'vehicle', 'diploid', 'without', 'wt', 'reporter', 'cellsprimary', 'standard', 'transcriptome', 'ad', 'single', 'livernormal', 'cellsreplicate', 'reads', 'typewild', 'deletion', 'incubation', 'neutrophils']\n"
     ]
    }
   ],
   "source": [
    "pos_words = []\n",
    "interesting_words = ['control' , 'normal' , 'Standard' , 'uninfected']\n",
    "words = dict()  \n",
    "columns3 = ['title','characteristics_ch1','source_name_ch1','description'] \n",
    "for i in columns3: \n",
    "    train[i].fillna('NAN' ,inplace = True)\n",
    "for i in columns3: \n",
    "    test[i].fillna('NAN',inplace = True)\n",
    "d_selected = train[train['ctrl'] == 1] \n",
    "d_not_selected = train[train['ctrl'] == 0]   \n",
    "\n",
    "ser1 = d_selected[columns3[0]] \n",
    "final =test[columns3[0]]\n",
    "for i in range(1,len(columns3)): \n",
    "    ser1 = ser1 + d_selected[columns3[i]] \n",
    "    final =final + test[columns3[i]]\n",
    "stops = set(stopwords.words('english'))\n",
    " \n",
    "\n",
    "for titles in ser1.values: \n",
    "    temp = titles.lower()\n",
    "    for k in temp.split(): \n",
    "        if (k in stops): \n",
    "            continue\n",
    "        if (k not in words): \n",
    "            words[k] = 1 \n",
    "        else: \n",
    "            words[k] = words[k] + 1\n",
    "\n",
    "bow = []\n",
    "for keys in words: \n",
    "    bow.append((words[keys],keys)) \n",
    "bow.sort(key = lambda x: -1*x[0]) \n",
    "for i in range(len(bow)): \n",
    "    b1 = bow[i][1] in words1  \n",
    "    if(not b1): \n",
    "        pos_words.append(bow[i][1])\n",
    "    #print(f\"{bow[i][1]}  --  {b1}\") \n",
    "print(len(pos_words)) \n",
    "pos_reduced = [] \n",
    "for i in pos_words: \n",
    "    if(len(i) <= 6): \n",
    "        pos_reduced.append(i) \n",
    "print(len(pos_reduced)) \n",
    "\n",
    "words_ps = [] \n",
    "for i in pos_words: \n",
    "    count0 = 0 \n",
    "    for records in final.values:  \n",
    "        temp = records.lower()\n",
    "        for elements in records.split(): \n",
    "            if(i == elements): \n",
    "                count0 = count0+1\n",
    "    \n",
    "    words_ps.append((i,count0))\n",
    "\n",
    "words_ps.sort(key = lambda x: -1*x[1]) \n",
    "#print(words_ps) \n",
    "pos_reduced = [] \n",
    "for i in words_ps: \n",
    "    if(i[1] > 0): \n",
    "        pos_reduced.append(i[0]) \n",
    "print(pos_reduced)  \n",
    "ws = pos_reduced.copy() \n",
    "print(ws)\n",
    "pos_reduced = [] \n",
    "for i in ws: \n",
    "    if(i.isalpha() and i != 'null' and i != 'liverliver'): \n",
    "        pos_reduced.append(i) \n",
    "\n",
    "print(pos_reduced) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['whole', 'donor', 'derived', 'obtained', 'spinal', 'transfected', 'information', 'group', 'stem', 'using', 'culture', 'contained', 'molecule', 'appended', 'fastq', 'entry', 'exposed', 'mutant', 'bone', 'vs', 'renal', 'marrow', 'clone', 'squamous', 'transgenic', 'lymph', 'alveolar', 'associated', 'hybridization', 'sorted', 'node', 'spleen', 'lavage', 'kidney', 'additional', 'gastric', 'replicates', 'zinc', 'acid', 'mesenchymal', 'population', 'stably', 'allele', 'labeling', 'metastasis', 'purified', 'files', 'compared', 'deficient', 'tumorlung', 'bronchoalveolar', 'pancreatic', 'activated', 'cytometry', 'isolate', 'oligodendrocyte', 'cellshuman', 'lumbar', 'assessed', 'smoked', 'ko']\n"
     ]
    }
   ],
   "source": [
    "neg_words = []\n",
    "words1 = dict()  \n",
    "ser2 = d_not_selected[columns3[0]] \n",
    "for i in range(1,len(columns3)): \n",
    "    ser2 = ser2 + d_not_selected[columns3[i]]\n",
    "\n",
    "for titles in ser2.values: \n",
    "    temp = titles.lower()\n",
    "    for k in temp.split(): \n",
    "        if (k in stops): \n",
    "            continue\n",
    "        if (k not in words1): \n",
    "            words1[k] = 1 \n",
    "        else: \n",
    "            words1[k] = words1[k] + 1\n",
    "\n",
    "bow = []\n",
    "for keys in words1:   \n",
    "    \n",
    "        bow.append((words1[keys],keys)) \n",
    "bow.sort(key = lambda x: -1*x[0])  \n",
    "\n",
    "for i in range(len(bow)): \n",
    "    b1 = bow[i][1] in words \n",
    "    if(not b1): \n",
    "        neg_words.append(bow[i][1])\n",
    "    \n",
    "    #print(bow[i][1], \" \" , b1) \n",
    "#print(len(neg_words)) \n",
    "neg_reduced = [] \n",
    "for i in neg_words: \n",
    "    if(len(i)<=6): \n",
    "        neg_reduced.append(i) \n",
    "#print(len(neg_reduced)) \n",
    "\n",
    "words_ns = [] \n",
    "for i in neg_words: \n",
    "    count0 = 0 \n",
    "    for records in final.values:  \n",
    "        temp = records.lower()\n",
    "        for elements in records.split(): \n",
    "            if(i == elements): \n",
    "                count0 = count0+1\n",
    "    \n",
    "    words_ns.append((i,count0))\n",
    "\n",
    "words_ns.sort(key = lambda x: -1*x[1]) \n",
    "##print(words_ns) \n",
    "neg_reduced = [] \n",
    "for i in words_ns: \n",
    "    if(i[1] > 0): \n",
    "        neg_reduced.append(i[0]) \n",
    "###print(neg_reduced) \n",
    "wn = neg_reduced.copy() \n",
    "neg_reduced = [] \n",
    "for i in wn: \n",
    "    if(i.isalpha()): \n",
    "        neg_reduced.append(i) \n",
    "print(neg_reduced)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Sorted HMLE cell line', 'IMEC cell line', 'nasal polyp',\n",
       "       'Lovo cells infected with mutant DLntA Listeria, biological rep1',\n",
       "       'Lovo cells infected with LntA-V5+ Listeria, biological rep3',\n",
       "       'Lovo cells infected with LntA-V5+ Listeria, biological rep1',\n",
       "       'Lovo cells infected with LntA-V5+ Listeria, biological rep2',\n",
       "       'Lovo cells infected with mutant DLntA Listeria, biological rep2',\n",
       "       'Lung cancer cell line', 'Distal lung',\n",
       "       'WT Mouse airway (trachea) epithelial cells',\n",
       "       'Pulse-seq reporter mouse tracheal epithelial cells',\n",
       "       'Lung myeloid dendritic cells',\n",
       "       'human embryonic lung fibroblast (line HELF-977)',\n",
       "       'human adult skin fibroblasts (line HAF-1608)', 'Stage 1 ccRCC',\n",
       "       'normal match to Stage 1 ccRCC', 'normal match to Stage 2 ccRCC',\n",
       "       'Stage 2 ccRCC', 'normal match to Stage 3 ccRCC', 'Stage 3 ccRCC',\n",
       "       'Stage 4 ccRCC', 'normal match to Stage 4 ccRCC', 'skin biopsy',\n",
       "       'MCF10A Vector Control pQCXI', 'MCF10A YAP-5SA', 'MCF10A RUNX3',\n",
       "       'Fibroblasts isolated from mammary gland with Ets2 deletion in fibroblasts and no oncogene present in epithelial cells',\n",
       "       'Fibroblasts isolated from mammary gland with Ets2 present in fibroblasts and the ErbB2 oncogene present in epithelial cells',\n",
       "       'Fibroblasts isolated from mammary gland with Ets2 present in fibroblasts and no oncogene present in epithelial cells',\n",
       "       'Fibroblasts isolated from mammary gland with Ets2 deletion in fibroblasts and the ErbB2 oncogene present in epithelial cells',\n",
       "       'Lung Epithelial Cell Line', 'MCF10A cells', 'primary hepatocytes',\n",
       "       'A549 cell line', 'GALNT10 knockdown in  A549 cell line',\n",
       "       'H322 cell line', 'GALNT10 over-expressing in H322 cell line',\n",
       "       'Liver tissue', 'mock-treated hepatocytes',\n",
       "       'hepatocytes transduced with SINLV.PGK.GFP',\n",
       "       'Dorsal section of lumbar spinal cord',\n",
       "       'Lumbar(L3 to L5) dorsal root ganglion',\n",
       "       'liver,normal diet,4 weeks', 'liver,high fat diet,4 weeks',\n",
       "       'T84, 2hrs post-migration, no PMN, replicate 1',\n",
       "       'T84, 2hrs post-migration, with PMN, replicate 1',\n",
       "       'T84, 2hrs post-migration, with PMN, replicate 2',\n",
       "       'T84, 2hrs post-migration, no PMN, replicate 3',\n",
       "       'T84, 2hrs post-migration, no PMN, replicate 2',\n",
       "       'Errb mutant mouse stria vascularis',\n",
       "       'wild type mouse stria vascularis',\n",
       "       'WT C57BL/6 mice,colonic mucosa, sacrificed at 6 weeks',\n",
       "       'Gasko-/-  C57BL/6 mice,colonic mucosa, sacrificed at 6 weeks',\n",
       "       'CaCo2 cells', 'A549 cells, C12 irradiated, 1h',\n",
       "       'A549 cells, C12 irradiated, 4h', 'A549 cells, C12 control, 1h',\n",
       "       'A549 cells, C12 control, 4h', 'Bmi1 wild-type lung cells',\n",
       "       'Bmi1 mutant lung cells',\n",
       "       'left lung of wild-type mice with Pneumocystis infection',\n",
       "       'L.T.R + siKDM3A', 'L.T.R', 'L.T + shKDM3A', 'P6', 'L', 'liver',\n",
       "       '8HR_novirus_DMSO', '24HR_novirus_DMSO', '4HR_novirus_Silymarin',\n",
       "       '24HR_novirus_Silymarin', 'Caco2 cells',\n",
       "       'IPF Rapidly lung fibroblst 1-2', 'IPF Rapidly lung fibroblst 1-1',\n",
       "       'IPF Rapidly lung fibroblst 1-4', 'IPF Slowly lung fibroblast 2-1',\n",
       "       'IPF Rapidly lung fibroblst 1-3', 'SPC-HAxTCR-HA lung',\n",
       "       'SPC-HA lung', 'primary airway epithelial cells',\n",
       "       '4TO7 pMSCV-miR-23b/27b/24 stable cell line',\n",
       "       '4TO7 pMSCV-puro stable cell line',\n",
       "       'mantle cell lymphoma cell line', 'Normal Diet, Ad Lib, Liver',\n",
       "       'EB_d6_shLuc_ctrl', 'EB_24h_shSmyd5', 'EB_d10_shSmyd5',\n",
       "       'mES_shLuc_ctrl', 'EB_d14_shLuc_ctrl',\n",
       "       'Mammary epithelial HC11 cells',\n",
       "       'normal mouse tracheal epithelium',\n",
       "       'mouse tracheal epithelium 48hrs post SO2 injury',\n",
       "       'hepatoma cell line', 'CD_Liver_2L', 'CD_Liver_1L', 'CD_Liver_1R',\n",
       "       'CD_Liver_2N', 'CD_Liver_3N', 'NOD GLN', 'Huh-7.5', 'Liver',\n",
       "       'Fresh frozen human hippocampus; patient 2',\n",
       "       'Fresh frozen human hippocampus; patient 3',\n",
       "       'Formalin-fixed paraffin-embedded human hippocampus; patient 1',\n",
       "       'Fresh frozen human hippocampus; patient 1',\n",
       "       'Formalin-fixed paraffin-embedded human hippocampus; patient 2',\n",
       "       'U87 cells infected with negative control-shRNA lentivirus',\n",
       "       'U87 cells infected with PHF20-shRNA lentivirus',\n",
       "       'U1810 NSCLC cell line', 'FXR floxed mice',\n",
       "       'mouse primary microglial cells', 'mouse primary microglia',\n",
       "       'K562 VEC', 'Splenocytes, 18h after VSV infection',\n",
       "       'Splenocytes, 18h after PBS injection', 'Lung',\n",
       "       'CD4+ cells from peripheral blood_control',\n",
       "       'Cell sort P4 lung SP-C + cells', 'Teratroma from diploid hESC',\n",
       "       'Teratroma from aneuploid hESC (trisomy 21)',\n",
       "       'Teratroma from aneuploid hESC (trisomy 12)',\n",
       "       'Alveolar type II cells', 'Mice_liver', 'MCSE_mice_adipose tissue',\n",
       "       'MSCE_mice_muscle', 'Mice_muscle', 'Mice_adipose tissue',\n",
       "       'Day4 APC flox hom', 'Day3 APC flox hom',\n",
       "       'MDA-MB-231 cells in monolayer culture',\n",
       "       '231wCFry cells in monolayer culture',\n",
       "       'Human peripheral blood, zinc depleted, 6 d',\n",
       "       'Human peripheral blood, baseline, 0 d',\n",
       "       'Human peripheral blood, zinc depleted, 10 d',\n",
       "       'Ramified microglia from 4-week rat brain',\n",
       "       'Amoeboid microglia from 5-day rat brain',\n",
       "       'adult mouse jejunum, hnf1alpha mutant',\n",
       "       'adult mouse jejunum, hnf1alpha control',\n",
       "       'primary tumor derived cell line',\n",
       "       'metastasis derived cell line (from lymph node metastasis)',\n",
       "       'NCRM-5', 'iMG (NCRM-5)', 'IntestineEC_MHCIIlo',\n",
       "       'IntestineEC_MHCIIhi', 'HCT116, siCont', 'HCT116, siAKR1B10+ADR',\n",
       "       'HCT116, siCont+ADR',\n",
       "       'IECs; 8 days post-infection; EPEC WT; replicate 2',\n",
       "       'IECs; 8 days post-infection; EPEC ΔescV; replicate 2',\n",
       "       'IECs; 8 days post-infection; EPEC WT; replicate 1',\n",
       "       'IECs; 8 days post-infection; EPEC ΔescV; replicate 1',\n",
       "       'IECs; 9 days old; uninfected; replicate 1',\n",
       "       'Paired normal tissue',\n",
       "       'Human primary epithelial cell cultures treated with IL-6 (10 ng/ml) for 24 hours',\n",
       "       'Human primary epithelial cell cultures treated with IL-17A (10 ng/ml) for 24 hours',\n",
       "       'Human primary epithelial cell cultures used as basal controls',\n",
       "       'Human primary epithelial cell cultures treated with all-trans-retinoic acid (30 nM)',\n",
       "       'Human plasma', 'PBS', 'A2B5/OTMP+ oligodendrocytes',\n",
       "       'A2B5/OTMP+ purified oligodendrocytes', '10A143B-3', '468-143B-3',\n",
       "       '10A143B-2', '468-143B-2', '10A143B-1', 'normal trachea tissue',\n",
       "       'hepatocytes', 'wild type mouse colon',\n",
       "       'STAT5b deficient mouse colon', 'ANS2 tumor',\n",
       "       'WTHD2 tumor right kidney', 'WTHD1 tumor left kidney',\n",
       "       'precentral gyrus brain, control, no DHA-ARA',\n",
       "       'Brain Tissue,Control 2',\n",
       "       'precentral gyrus brain, L, 1× LCPUFA (0.33%DHA-0.67%ARA)',\n",
       "       'Rat embryo testis at day E13 treted with DMSO (vehicle) in utero',\n",
       "       'Rat embryo testis at day E14 treted with DMSO (vehicle) in utero',\n",
       "       'Rat E13 embryo testis cultured with Vinclozolin in DMSO for 3 days',\n",
       "       'Rat embryo testis at day E14 treted with Vinclozolin in DMSO in utero',\n",
       "       'Rat embryo testis at day E16 treted with DMSO (vehicle) in utero',\n",
       "       'Rat E13 embryo testis cultured with ethanol for 3 days',\n",
       "       'Rat embryo testis at day E13 treted with Vinclozolin in DMSO in utero',\n",
       "       'Rat E13 embryo testis cultured with DMSO for 3 days',\n",
       "       'Rat embryo testis at day E16 treted with Vinclozolin in DMSO in utero',\n",
       "       'Renal cell carcinoma', 'NSCLC tumor',\n",
       "       'Donor 1, sorted population IFNg+IL10+',\n",
       "       'Donor 1, sorted population IFNg-IL10-',\n",
       "       'Donor 1, sorted population IFNg-IL10+',\n",
       "       'Donor 1, sorted population IFNg+IL10-',\n",
       "       'Donor 2, sorted population IFNg-IL10-',\n",
       "       'adjacent colon epithelium', 'mammary gland_Ccnd1+/+_E2',\n",
       "       'mammary gland_Ccnd1-/-_Vehicle', 'mammary gland_Ccnd1-/-_E2',\n",
       "       'mammary gland_Ccnd1+/+_Vehicle', 'Microglial cells',\n",
       "       'Nasal epithelial samples from Current Smoker',\n",
       "       'Nasal epithelial samples from Never Smoker',\n",
       "       'HCC827 derived cell line', 'HCC827 cell line',\n",
       "       'HCC4006 cell line', 'hSKP-HPC at 24h without exposure to Na-VPA',\n",
       "       'hSKP-HPC at 24h exposed to Na-VPA',\n",
       "       'primary mouse astrocytes, untreated',\n",
       "       'primary mouse astrocytes, IFNg stimulated for 12 hrs',\n",
       "       'primary mouse astrocytes, IFNa stimulated for 12 hrs',\n",
       "       'NCI-H1993 Mark2 Knockdown', 'NCI-H1993 pLKO control',\n",
       "       'NCI-H1693 pLKO control', 'NCI-H1693 Mark2 Knockdown',\n",
       "       'empty vector control', 'R273H mutant p53',\n",
       "       'uninfected_spleen harvested on day 0',\n",
       "       'LCMV-Armstrong-infected_spleen harvested on day 5',\n",
       "       'H441 lung adenocarcinoma cells', 'Solid tumor', 'brain_E12.5',\n",
       "       'yolk_E12.5', 'unvaccinated naïve_BAL',\n",
       "       'bone marrow MSC in culture from normal donor 1',\n",
       "       'CD235a+ cells sorted from bone marrow aspirates of normal donor 1',\n",
       "       'bone marrow MSC in culture from normal donor 2',\n",
       "       'CD235a+ cells sorted from bone marrow aspirates of normal donor 2',\n",
       "       'bone marrow MSC in culture from normal donor 3',\n",
       "       'Human breast adenocarcinoma cell-line MCF7'], dtype=object)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train['source_name_ch1'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Basecalling was performed with the real time analysis (RTA) package within the Genome Analyzer Sequencing Control Software (SCS2.10).',\n",
       "       'All data normalization and selection of fold-changed genes were performed using GeneSpringGX 7.3 (Agilent Technology, USA). The averages of normalized ratios were calculated by dividing the average of normalized signal channel intensity by the average of normalized control channel intensity. Functional annotation of genes was performed according to Gene OntologyTM Consortium (http://www.geneontology.org/index.shtml) by GeneSpringGX 7.3. Gene classification was based on searches done by BioCarta (http://www.biocarta.com/), GenMAPP (http://www.genmapp.org/), DAVID (http://david.abcc.ncifcrf.gov/), and Medline databases (http://www.ncbi.nlm.nih.gov/).',\n",
       "       'Cell intensity files were generated with GeneChipOperating Software. Raw data were processed by the Robust Multichip Analysis (RMA) algorithm.',\n",
       "       'RNA-seq reads were mapped with STAR 2.5.3a',\n",
       "       'RNA-Seq reads were aligned to the mouse genome (UCSC Genome; Assembly NCBI37/mm9) using STAR',\n",
       "       'Full-length RNAseq (SMART-Seq2)', \"3' polyA RNA-seq\",\n",
       "       'Data were processed by variance stabilizing transform and quantile normalization',\n",
       "       'The data were normalised using preprocessCore quantile normalisation in R. Each culture (highly and low sensitive to the virus) was normalized separately',\n",
       "       'Raw data was processed by MAS.5 (Affymetrix) and analyzed using GeneSpring GX10',\n",
       "       'The raw data were exported from Illumina GenomeStudio (no background subtraction was performed).  The data were log2 transformed and normalized by quantiles using BRB-Array Tools.',\n",
       "       'Data were normalized by Robust Multichip Average (RMA) using Affymetrix Powertools',\n",
       "       'Gene expression levels were estimated from GeneChip® probe intensities using robust multi-array average (RMA) normalization and background-correction algorithms.',\n",
       "       \"CEL files were summarized using Affymetrix Expression Console (current version 1.1). Expression estimates were derived from the probe level hybridization intensities using the Robust Multi-Array Analysis (RMA) algorithm.  Differential expression between experimental and control samples was examined using Welch's T-test and a False Discovery Rate correction.\",\n",
       "       'Robust Multichip Averages', 'RNA-seq',\n",
       "       'Array hybridization, washing, scanning, data extraction and analyses were performed as previously described (Keita et al., 2013).',\n",
       "       'The data were normalised using quantile normalisation with Genome Studio',\n",
       "       'The data were analyzed with Expression Console, R and Bioconductor.',\n",
       "       'fastx_clipper from FASTX Toolkit v0.0.14 was employed to remove the adapter sequence from raw sequencing read, reads shorter than 18 bp were removed',\n",
       "       'Command Console Software (Affymetrix, Santa Clara, CA, US) was used to control the scanner and summarize probe cell intensity data (CEL file generation) with default settings. The raw data were normalized by Expression Console with MAS 5.0.',\n",
       "       'The scanned images were analyzed with Feature Extraction Software (Agilent) using default parameters to obtain background subtracted and spatially detrended Processed Signal intensities. Features flagged in Feature Extraction as Feature Non-uniform outliers were excluded. Signal intensities were normalized using combined rank consistency filtering with LOWESS intensity normalization in Agilent Feature Extraction software. Data was subsequently analyzed by GeneSpring (Agilent).',\n",
       "       'RMA (bioconductor)',\n",
       "       'Data was normalized using the GCRMA algorithm as implemented in Bioconductor/R.',\n",
       "       'BioConductor Limma package.',\n",
       "       'Data were evaluated using the R software and the limma package from BioConductor. Pathway analyzes were done using the ranks of the test statistics  based on the KEGG.db annotation package.',\n",
       "       \"The raw intensity data were preprocessed by Affymetrix's Expression Console (version 1.1)\",\n",
       "       \"Data processing cel files were normalized with 'rma' function of the Bioconductor environment (v2.9) in R.3.0.0\",\n",
       "       \"Affymetrix® microarray data were collected with Affymetrix® GeneChip® Command Console® Software. Probe values were corrected for background and normalized via quantile normalized by Robust Multichip Average (RMA) of the Bioconductor 'affy' package.In the detection column, calls include P for Present, A for Absent, and M for Marginal.\",\n",
       "       'For RNA-Seq data quality control was performed by Fastqc. Before downstream analysis, low quality reads and adaptor contamination were deleted. RNA-Seq data mapping was performed by Tophat2 and hg19 as reference genome. Gene assemble and expression quantification was performed by Cufflinks.',\n",
       "       'The data were analyzed with the Bioconductor package Affy using RMA method for normalizing and summarizing probe level intensity measurements.',\n",
       "       'RMA normalization log2 at the transcript level using the oligo package of Bioconductor.',\n",
       "       'CASAVA 1.8.4 - configureBclToFastq.pl --input-dir runs/150924_D00198_0427_BC7WFRACXX/Data/Intensities/BaseCalls/ --output-dir process/150924_D00198_0427_BC7WFRACXX/ --sample-sheet process/150924_D00198_0427_BC7WFRACXX/ --ignore-missing-bcl --ignore-missing-stats --use-bases-mask y51,i6n,y51',\n",
       "       'The data were normalised using quantile normalisation with IlluminaGUI in R',\n",
       "       'The microarray data were analyzed using the GeneSpring GX software (Agilent). The data were summarized, log2 transformed and normalized with the Robust Multi-array Analysis (RMA) algorithm. Only probe sets with signal intensities above the 20th percentile in at least one of the 6 microarrays performed were retained, whereas all other probe sets were excluded from further analyses. Subsequently, the signal intensities of the replicate microarrays performed for each mouse group (SPC-HAxTCR-HA, SPC-HA) were averaged and the fold changes were calculated in reference to the data from the SPC-HA group. Only probe sets with a fold change greater than or equal to ±2 were retained. All other probe sets were excluded from further analyses. Finally, 378 transcripts were found to be differentially expressed in SPC-HAxTCR-HA vs. SPC-HA lungs. 326 of those were up-regulated and 52 were down-regulated.',\n",
       "       'Illumina bcl2fastq-1.8.4 used for base calling',\n",
       "       'Agilent Feature Extraction Software (v 9.5.3.1) was used for background subtraction and linear Lowess normalization.',\n",
       "       'RNAseq reads were filtered for average QC score>30. Reads were mapped to hg19 genome.',\n",
       "       'Data from both channels was extracted using the Agilent Feature Extractor Software, Version 7.5.  Due to potential variation of the Cy-5 dye we prefer to analyze the green channel alone and use single channel data.  Results for control probes were first removed, multiple repeats of any probe were averaged, then each array was z-normalized.  Differently expressed genes were identified between the 3 sample groups and pathway enrichment analysis was conducted using Diane 6.0, an in-house microarray analysis program.',\n",
       "       'Basecalls performed using GAPipeline or CASAVA',\n",
       "       'RMA normalization, done by the SCIBLU Micoarray Resourse Center (MARC), Lund University, Sweden',\n",
       "       'Affymetrix Expression Console version 1.1 was used to generate the .cel files. Partek Genomics Suite 6.5 (Partek Inc., St. Louis, MO) was used to perform data analysis. Robust multi-chip analysis (RMA) normalization was done on the entire data set.',\n",
       "       'Affymetrics probeset summaries were computed using RMA algorithm.\\xa0 Data were then imported to BRB-ArrayTools and subsequent data filtering, median normalization and statistical analyses were performed using this software.',\n",
       "       'Raw scan data were normalized with the use of cubic spline algorithm implemented in Illumina Normalizer module of GenePattern software (www.broadinstitute.org/genepattern).',\n",
       "       'Data was processed using GeneSpring GX 10 Software. Data was imported using the enhanced Agilent FE Import setting. Spot information in Feature Extraction was used to flag data, and normalization was performed using \"per spot and per chip: intensity dependent (Lowess) normalization\".  Data obtained for the Cy5-labelled GLN sample were divided by the Cy3-labelled PDLN sample to obtain ratios shown in the \"PRE_VALUE\" column of processed data. Raw data show log2 of ratio.',\n",
       "       'Statistical analysis was done with the statistical computing environment R. Additional software packages (affy, geneplotter, multtest, and vsn) were taken from the Bioconductor project.  Microarray preprocessing: Probe level normalization was conducted using the variance stabilization method (vsn) from Huber et al..',\n",
       "       'Fastq files containing passed-QC reads were uploaded to BaseSpace. Filtering, mapping and abundance measurements (FPKM) were performed with the app RNAexpress (Illumina) within BaseSpace.',\n",
       "       'The data were analyzed with Affymetrix Expression Console software (v1.4.1) using RMA-Sketch algorithm for data normalization. FF, FFPE_3h and FFPE_10h datasets were normalized together. Normalization was done to gene level as well as to exon level.',\n",
       "       'The raw data were normalized by the log scale robust multi-array analysis (RMA) method with the Expression Console software version 1.1 (Affymetrix).',\n",
       "       '1. Mapping: Reads were mapped with Tophat/2.0.4 to the Human genome assembly, build GRCh37.',\n",
       "       'Using a HiSeq SBS sequencing kit version 4, Fastq files were generated and demultiplexed with with the bcl2fastq v2.17.1.14 Conversion Software (Illumina).',\n",
       "       'GCOS scaling to target intensity of 100',\n",
       "       'Data processing of arrays were performed using Illumina BeadStudio software. Gene expression data was normalized using Quantile normalization and log2 transformation. Sample Gene Profile option of Illumina BeadStudio was used to export to a data matrix.',\n",
       "       'Background correction and normalization was performed using the RMA-Sketch implementation in Affymetrix Expression Console',\n",
       "       'GeneSpring software GX (Agilent Technologies), RMA normalization',\n",
       "       'Data was normalized in R (v.2.14.) with the bioconductor package affy(v.1.32.0) and the function RMA.',\n",
       "       'Microarray data were analyzed using the Oligo package available at the Bioconductor website (www.bioconductor.org). The raw data were first background-corrected by the Robust Multichip Average (RMA) method and then normalized by an invariant set method. Probesets were considered expressed if at least one sample had a RMA expression value greater than the mean of the negative control probesets, 18,494 probesets were considered expressed.  Differential gene expression between the control and mutant mice was analyzed by the Limma package available at the Bioconductor website. P-values obtained from the multiple comparison tests were corrected by false discovery rates. Heatmap displays were created using the freely available MeV package ( http://www.tm4.org/mev/).',\n",
       "       'Affymetrix Expression Console RMA',\n",
       "       'Paired-end sequencing (150 bp for each read) was performed using HiSeq X with the depth of 24 million reads for each sample. Quality of sequenced reads were assessed using FastQC (https://www.bioinformatics.babraham.ac.uk/projects/fastqc/). PolyA/T tails were trimmed using PRINSEQ (Schmieder and Edwards, 2011). Adaptor sequences were trimmed and shorter reads than 24 bp were dropped using Trimmomatic (Bolger et al., 2014). Reads were mapped to the mouse reference genome (mm10) using Hisat2 (Kim et al., 2015) with default setting. Duplicate reads were removed using the markdup option of SAMtools (Li et al., 2009). Fragments numbers were counted using the featureCounts option of SUBREAD (Liao et al., 2014).',\n",
       "       'The Cy5 fluorescent intensity of each spot was analyzed by Genepix 4.1 software (Molecular Devices, Sunnyvale, CA). The signal intensity of each spot was corrected by subtracting background signals in the surrounding. We filtered out spots that signal-to-noise ratio was less than 0 or control probes. Spots that passed aforementioned criteria were normalized by the R program in the limma package using quantile normalization (Smyth, 2005).',\n",
       "       'Raw signal intensities were initially screened by using the MaxD/View-Program to remove false positive and false negative signals (using a cut off of at least one P value for each transcript <0.2). The data were then normalized by using the global geometric mean (GGM) normalization option within the MaxD/View-Program',\n",
       "       'The data were generated using the Affymetrix Scanner according to Affymetrix protocol. The intensities are from MAS5.0',\n",
       "       'Probes with detection P-values above 0.05 in all samples were excluded from the dataset by the Illumina GenomeStudio software. Absolute signal intensities of the filtered probes were quantile normalized using EXPANDER 5.2 (Ulitsky I et al. 2010, Nat Protoc. 2010;5:303-22).',\n",
       "       'CEL files were exported to Affymetrix Expression Console Version 1.1 and RMA normalization was carried out. [The data were also  analyzed with Microarray Suite version 5.0 (MAS 5.0) using GeneChip Operating Software (GCOS, Affymetrix, CA, USA). For absolute analysis, each chip was normalized to a target intensity of 500, and probe sets were assigned a signal intensity and detection call of Present, Marginal or Absent (See MAS5_GCOS_analyzed.txt on GSE29885)]',\n",
       "       'The scanned images were analyzed using the GeneChip Analysis Suite 3.3 (Affymetrix) to identify genes differentially expressed among the RNA samples. Data were normalized using the RMA algorithm.',\n",
       "       'Data processing was done using Partek Genomics Suite 6.5 software. RMA was used to summarize and normalize the probe data into probe set expression values. The list of utilized probe sets was defined by custom .ps and .mps files. The RMA procedure used included the options adjust for GC content, probe sequence, RMA background correction, Quantile Normalization and median polish probe set summarization. Expression values for probesets were summarized to a single value per gene based on the custom file MoEx-1_0-st-v1._core+miRNA+RfSq+UCSC_.SH.2916.mps. The outlier excluded mean method was used. This procedure excludes probe set values that are more than 3 standard deviations above the mean or more and 2 standard deviations below the mean.',\n",
       "       'CEL files were imported into the Partek Genomics Suite v2 (www.partek.com) using option \"RMA\" under default settings. RMA background correction and quantile normalization was performed upon import; generating 53,981 log2 median polished gene probe expression values per CEL file. Probe values representing spike-in and hybridization controls were next filter-removed and the remaining probe values (n=53,617) exported for interrogation and analysis in R (https://cran.r-project.org/).',\n",
       "       'The Agilent Feature Extraction Software (FES) was used to read out and process the microarray image files. The software determines feature intensities (including background subtraction), rejects outliers and calculates statistical confidences. For the normalised data, the signal intensities from the single-experiment raw data lists are normalized by dividing the intensity values by their median.',\n",
       "       'Agilent Feature Extraction Software (v 10.7) was used for background subtraction.',\n",
       "       'Data extraction was performed with the ‘Feature Extraction Software V10.7.3.1’ using the extraction protocol file ‘GE1_107_Sep09.xml’. Extracted microarray raw data were imported into Omics Explorer software v3.0 (Qlucore) under default import settings for Agilent One Color mRNA Microarrays. Accordingly, data processing steps were: 1) removal of control measurements, 2) log base 2 transformation, 3) normalization by shifting to 75 Percentile, 4) averaging of values from on-chip replicates, 5) baseline transformation to the median. Note, that six additional samples have been initially imported into one Qlucore Omics Explorer project. These 6 additional samples have not been utilized in the context of this study. However, baseline transformation in Qlucore Omics Explorer software (see step 5) is based on all samples (including the 6 samples that were finally removed and are not contained in this very GEO series).',\n",
       "       'For data processing, a novel algorithm that distinguished between changes in gene expression and splicing variation was developed.',\n",
       "       'This information is not available since the experiments were done more than 10 years ago.',\n",
       "       'High-level and low-level transcriptomics analyses were carried out using GeneSpring GX12.0 (Agilent) (www.genomics.agilent.com). ExonPLIER16 summarization algorithm, quantile normalization, baseline transformation to all samples.',\n",
       "       'Cel files were imported into Genespring for preprocessing using RMA, and normalization using a global per-chip and per-gene normalization strategy',\n",
       "       'The data was analyzed by the statistical core facility at Dan L. Duncan Cancer Center at Baylor College of Mediaine. CEL files were processed in Bioconductor using Quantile method.',\n",
       "       'Image analysis was performed in real time by the HiSeq Control Software (HCS) v2.2.68 and Real Time Analysis (RTA) v1.18.66.3, running on the instrument computer. RTA performs real-time base calling on the HiSeq instrument computer. Then the Illumina bcl2fastq 1.8.4 pipeline was used to generate the sequence data.',\n",
       "       \"RRBS: Reads were trimmed with Trim Galore! (v0.4.2, http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/) for adapter contamination and 3' ends with base quality below 20 in RRBS mode.\",\n",
       "       'The data was analyzed with GeneSpring version 7.3 using RMA normalization with a custom CDF file(Mm430_Mm_REFSEQ_8), then normalized to the REF colon repl2 and the mean of the 3 WT colon samples',\n",
       "       'The data was analyzed with GeneSpring version 7.3 using RMA normalization with a custom CDF file(Mm430_Mm_REFSEQ_8), then normalized to the REF colon repl1 and the mean of the 3 WT colon samples',\n",
       "       'Agilent Feature Extraction V11.5.1.1 was used for background subtraction and LOWESS normalization. Control spots are removed, replicate oligos are averaged.',\n",
       "       'MAS5',\n",
       "       'The data were analyzed with R software and Bioconductor with MAS5.0 algorithm using global scaling as normalization method. Trimmed mean target intensity of each array was determined as 210. R was used for additional quality control and computing statistics with corrected variances.',\n",
       "       'The data were analyzed with GCOS using MAS5.0 algorithm with Affymetrix default analysis settings and global scaling as normalization method. The trimmed mean target intensity of each array was arbitrarily set to 218.',\n",
       "       'The data were analyzed with GCOS using MAS5.0 algorithm with Affymetrix default analysis settings and global scaling as normalization method. The trimmed mean target intensity of each array was arbitrarily set to 107.',\n",
       "       'Aligned to hg19 using tophat v2.0.13 and Gencode V19 transcript models',\n",
       "       'The data were normalized using quantile normalization withIllumina Genome Studio',\n",
       "       'Raw data were summarised and normalised using Robust Multi-array Average (RMA)',\n",
       "       'The raw data was checked by FastQC.',\n",
       "       'The data were analyzed and normalized by Robust Multi-array Average (RMA) algorithm using affylmGUI (1.14.0) running limma version 2.9.16 within the language and environment program R (2.4.1).',\n",
       "       'Grooming of de-multiplexed reads was performed with FASTQ groomer 1.04.',\n",
       "       'A single weighted mean expression level for each gene was derived using MICROARRAY SUITE 5.0 (MAS 5.0) software . The MAS 5.0 software also generated a detection P value [P(detection)] using a one-sided Wilcoxon sign-ranked test, which indicated whether the transcript was reliably detected. The trimmed mean target intensity of each array was arbitrarily set to 100.',\n",
       "       'The raw signal intensity data were background corrected (normexp method) and quantile normalised using limma package of Bioconductor in R program.',\n",
       "       'Background correction, summarization (median polish) and normalization (quantile) were done with Robust Multi-array Analysis using RMAExpress v1.0.5',\n",
       "       'The data were quantile normalized without background subtraction using Illumina GenomeStudio V1.8',\n",
       "       'The data were analyzed with Expression Console using Affymetrix default analysis settings and RMA normalization method.',\n",
       "       'AGCC Scan Control software',\n",
       "       'Expression values were subjected to RMA normalization using Affymetrix CDF probeset definition and GeneSpringGX v7.3.1.',\n",
       "       'The images and raw data from each chip were transferred automatically to the microarray database using one of the specified microarray core servers',\n",
       "       'Sequencing, data quality, reads repartition (e.g., for potential ribosomal contamination), and insert size estimation were performed using FastQC, Picard-Tools, Samtools and rseqc',\n",
       "       'bcl2fastq/2.15.0.4',\n",
       "       'The resulting .gpr files were uploaded to Spotfire DecisionSite for Microarray Analysis running the Splus suite, and normalization performed using the LOWESS method.  The resulting log2 fold changes were calculated and used for uploading to Database for Annotation, Visualization and Integrated Discovery (DAVID) (NCBI).Two-fold up or down regulated genes were uploaded to DAVID. Pathways/terms which were enriched in a statistically significant manner (P <0.05; FDR corrected) were used for further analysis and display.',\n",
       "       'Data were processed with GCOS 1.4 (TGT = 150) and subsequently imported and further analyzed in the online database (www.bioretis.de).',\n",
       "       'Illumina Casava1.7.1a5 software used for basecalling.'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train['data_processing'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Martin,,Eilers', 'Heung-Man,,Lee', 'Helene,,Bierne',\n",
       "       'Anna,,Esteve-Codina', 'Robert,B,Weiss', 'Adam,,Haber',\n",
       "       'Farrah,,Kheradmand', 'Alina,,Artcibasova', 'John,A,Copland',\n",
       "       'Jeffrey,,Chang', 'Tuan Zea,,Tan', 'Julie,,Wallace',\n",
       "       'Maria,Isabel,Ramirez', 'Sandya,,Liyanarachchi', 'Markus,,Stoffel',\n",
       "       'Jian,,Zhang', 'Matthias,Sebastian,Matter', 'Marco,,Ranzani',\n",
       "       'Arkady,,Khoutrosky', 'Qin,,Feng', 'Eric,L,Campbell',\n",
       "       'Jichao,,Chen', 'Timothy,Cragin,Wang', 'Dario,,Greco',\n",
       "       'Jochen,,Wilhelm', 'Carla,,Kim', 'Juanjo,,Lozano', 'Yang,,Hu',\n",
       "       'Pinji,,Lei', 'Annamaria,,Carissimo', 'James,William,MacDonald',\n",
       "       'Antonis,,Kourtidis', 'Hyun Sub,,Cheong', 'Andreas,,Jeron',\n",
       "       'Allan,,Brasier', 'Yong,,Wei', 'Lixin,,Rui', 'Kevin,G,Becker',\n",
       "       'Benjamin,L,Kidder', 'Jeanette,Annika,Nilsson', 'Xia,,Gao',\n",
       "       'Giacomo,,Diaz', 'Yujin,,Hoshida', 'Remi,J,Creusot',\n",
       "       'Christian,M,Lange', 'Jorge,,Moscat', 'Isabella,,Wimmer',\n",
       "       'Tianlong,,Liu', 'Janne,,Lehtiö', 'Young-Chae,,Kim',\n",
       "       'Anna,,Kiialainen', 'Bon,Quy,Trinh', 'Ori,,Staszewski',\n",
       "       'Raman,,Agrawal', 'Olivia,,Prazeres da Costa',\n",
       "       'Michael,Patrick,Morley', 'Yishai,,Avior', 'Yoshihiko,,Kobayashi',\n",
       "       'Chien-Yun,,Hsiang', 'Karen,Ruth,Reed',\n",
       "       'Jessica,Corban Wos,Graham', 'Robert,J.,Cousins',\n",
       "       'S Thameem,,Dheen', 'Francois,,Boudreau',\n",
       "       'Charles,Arthur,Whittaker', 'Kory,R,Johnson', 'Ana,,Izcue',\n",
       "       'Masashi,,Idogawa', 'Mathias,,Hornef', 'David,,Blanco', 'Reen,,Wu',\n",
       "       'Kasey,C,Vickers', 'Joe,A,Nielsen', 'Chad,,Creighton',\n",
       "       'Gordon,K,Smyth', 'Alexander,,Herrmann', 'Xiaonan,,Han',\n",
       "       'Brigitte,,Royer-Pokora', 'Kumar,,Kothapalli', 'Michael,K,Skinner',\n",
       "       'Laurie,,Ailles', 'Marjorie,,Romkes', 'Esperanza,,Perucha',\n",
       "       'Luoyan,,Ai', 'Mathew,Casimiro,Casimiro', 'Thomas,,Arnold',\n",
       "       'Sriram,,Sridhar', 'Giovina,,Ruberti', 'Robim,M,Rodrigues',\n",
       "       'daniel,,beiting', 'Roland,,Hubaux', 'Joanna,,Burdette',\n",
       "       'Rebekah,,Karns', 'Mikhail,,Dozmorov', 'Noémie,,Robil',\n",
       "       'Ido,,Amit', 'Deepak,,Kaushal', 'Thomas,,Häupl', 'Matt,,Grimmer'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train['contact_name'].unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Unlabelled Train Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "unl = pd.read_csv('unlabelled_train_data.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'geo_accession', 'gse_id', 'channel_count',\n",
       "       'characteristics_ch1', 'contact_address', 'contact_city',\n",
       "       'contact_country', 'contact_department', 'contact_email',\n",
       "       'contact_institute', 'contact_name', 'contact_state', 'data_processing',\n",
       "       'data_row_count', 'description', 'extract_protocol_ch1',\n",
       "       'growth_protocol_ch1', 'hyb_protocol', 'label_ch1',\n",
       "       'label_protocol_ch1', 'last_update_date', 'molecule_ch1',\n",
       "       'organism_ch1', 'platform_id', 'scan_protocol', 'source_name_ch1',\n",
       "       'status', 'submission_date', 'supplementary_file', 'taxid_ch1', 'title',\n",
       "       'treatment_protocol_ch1', 'type', 'contact_phone', 'contact_laboratory',\n",
       "       'relation', 'contact_fax', 'biomaterial_provider_ch1',\n",
       "       'contact_web_link', 'characteristics_ch2', 'extract_protocol_ch2',\n",
       "       'label_ch2', 'label_protocol_ch2', 'molecule_ch2', 'organism_ch2',\n",
       "       'source_name_ch2', 'taxid_ch2', 'treatment_protocol_ch2',\n",
       "       'biomaterial_provider_ch2', 'growth_protocol_ch2'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unl.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Unnamed: 0                      0\n",
       "geo_accession                   0\n",
       "gse_id                          0\n",
       "channel_count                   0\n",
       "characteristics_ch1           136\n",
       "contact_address                22\n",
       "contact_city                   27\n",
       "contact_country                26\n",
       "contact_department           6077\n",
       "contact_email                4740\n",
       "contact_institute              22\n",
       "contact_name                   22\n",
       "contact_state                6463\n",
       "data_processing               153\n",
       "data_row_count                 31\n",
       "description                  5318\n",
       "extract_protocol_ch1          348\n",
       "growth_protocol_ch1          7520\n",
       "hyb_protocol                 6066\n",
       "label_ch1                    5870\n",
       "label_protocol_ch1           6066\n",
       "last_update_date                0\n",
       "molecule_ch1                   10\n",
       "organism_ch1                    0\n",
       "platform_id                    22\n",
       "scan_protocol                6068\n",
       "source_name_ch1                 0\n",
       "status                          0\n",
       "submission_date                 0\n",
       "supplementary_file           5736\n",
       "taxid_ch1                      15\n",
       "title                           0\n",
       "treatment_protocol_ch1       7600\n",
       "type                            0\n",
       "contact_phone               13166\n",
       "contact_laboratory          11823\n",
       "relation                    13166\n",
       "contact_fax                 18856\n",
       "biomaterial_provider_ch1    18991\n",
       "contact_web_link            19220\n",
       "characteristics_ch2         18255\n",
       "extract_protocol_ch2        18418\n",
       "label_ch2                   18255\n",
       "label_protocol_ch2          18414\n",
       "molecule_ch2                18255\n",
       "organism_ch2                18255\n",
       "source_name_ch2             18255\n",
       "taxid_ch2                   18255\n",
       "treatment_protocol_ch2      19092\n",
       "biomaterial_provider_ch2    19425\n",
       "growth_protocol_ch2         19116\n",
       "dtype: int64"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unl.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Control 1\n",
      "Control 2\n",
      "Control 2\n",
      "Control 1\n",
      "Control 2\n",
      "Control 2\n",
      "Control 1\n",
      "Control 1\n",
      "Control 2\n",
      "Control_rep1\n",
      "Ctrl_1\n",
      "Ctrl_2\n",
      "Ctrl_3\n",
      "Control 1\n",
      "Ctrl_3\n",
      "Ctrl_1\n",
      "Ctrl_2\n",
      "Control 1\n",
      "Control 2\n",
      "Control_rep1\n",
      "Ctrl_2\n",
      "Ctrl_1\n",
      "Control 2\n",
      "Control 1\n",
      "Control 2\n",
      "Control 1\n",
      "WT rep 2\n",
      "WT rep 3\n",
      "WT rep 1\n"
     ]
    }
   ],
   "source": [
    "\n",
    "for str1 in unl['title'].values: \n",
    "    if (str1 in train['title'].values): \n",
    "        print(str1) \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
